\section{Appendix}

\subsection{Chromatin feature download links} \label{sec:chromFeat_download_links}
The basic download paths for all chromatin feature files in .bam format are\\
\small{\url{https://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeBroadHistone/}}\\
\small{\url{https://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeOpenChromDnase/}}\\
\small{\url{https://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeSydhTfbs/}}\\
for CTCF/Histones, DNase, RAD21/SMC3, respectively. 
The actual files for replicates 1 and 2 are then easily be found by searching for ``$\mathit{CELL}\_\mathit{FEATURE}$'', 
where $\mathit{CELL}$ is the Cell line (e.\,g. GM12878) and $\mathit{FEATURE}$
is the chromatin feature of interest, e.\,g. CTCF, H3K27ac and so on.
For convenience, the pdf version of this document contains clickable links below.

\begin{minipage}{0.23\textwidth}
\textbf{GM12878} \\
\textbf{replicate 1}\\
\href{https://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeBroadHistone/wgEncodeBroadHistoneGm12878CtcfStdAlnRep1.bam}{\nolinkurl{CTCF}}\\
\href{https://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeBroadHistone/wgEncodeBroadHistoneGm12878H3k27acStdAlnRep1.bam}{\nolinkurl{H3k27ac}}\\
\href{https://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeBroadHistone/wgEncodeBroadHistoneGm12878H3k27me3StdAlnRep1.bam}{\nolinkurl{H3k27me3}}\\
\href{https://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeBroadHistone/wgEncodeBroadHistoneGm12878H3k36me3StdAlnRep1.bam}{\nolinkurl{H3k36me3}}\\
\href{https://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeBroadHistone/wgEncodeBroadHistoneGm12878H3k4me1StdAlnRep1.bam}{\nolinkurl{H3k4me1}}\\
\href{https://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeBroadHistone/wgEncodeBroadHistoneGm12878H3k4me2StdAlnRep1.bam}{\nolinkurl{H3k4me2}}\\
\href{https://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeBroadHistone/wgEncodeBroadHistoneGm12878H3k4me3StdAlnRep1.bam}{\nolinkurl{H3k4me3}}\\
\href{https://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeBroadHistone/wgEncodeBroadHistoneGm12878H3k79me2StdAlnRep1.bam}{\nolinkurl{H3k79me2}}\\
\href{https://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeBroadHistone/wgEncodeBroadHistoneGm12878H3k9acStdAlnRep1.bam}{\nolinkurl{H3k9ac}}\\
\href{https://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeBroadHistone/wgEncodeBroadHistoneGm12878H3k9me3StdAlnRep1.bam}{\nolinkurl{H3k9me3}}\\
\href{https://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeBroadHistone/wgEncodeBroadHistoneGm12878H4k20me1StdAlnRep1.bam}{\nolinkurl{H4k20me1}}\\
\href{https://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeOpenChromDnase/wgEncodeOpenChromDnaseGm12878AlnRep1.bam}{\nolinkurl{DNase}}\\
\href{https://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeSydhTfbs/wgEncodeSydhTfbsGm12878Rad21IggrabAlnRep1.bam}{\nolinkurl{Rad21}}\\
\href{https://www.encodeproject.org/files/ENCFF415OZI/@@download/ENCFF415OZI.bam}{\nolinkurl{SMC3}}
\end{minipage}\hfill
\begin{minipage}{0.23\textwidth}
\textbf{GM12878} \\
\textbf{replicate 2}\\
\href{https://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeBroadHistone/wgEncodeBroadHistoneGm12878CtcfStdAlnRep2.bam}{\nolinkurl{CTCF}}\\
\href{https://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeBroadHistone/wgEncodeBroadHistoneGm12878H3k27acStdAlnRep2.bam}{\nolinkurl{H3k27ac}}\\
\href{https://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeBroadHistone/wgEncodeBroadHistoneGm12878H3k27me3StdAlnRep2.bam}{\nolinkurl{H3k27me3}}\\
\href{https://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeBroadHistone/wgEncodeBroadHistoneGm12878H3k36me3StdAlnRep2.bam}{\nolinkurl{H3k36me3}}\\
\href{https://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeBroadHistone/wgEncodeBroadHistoneGm12878H3k4me1StdAlnRep2.bam}{\nolinkurl{H3k4me1}}\\
\href{https://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeBroadHistone/wgEncodeBroadHistoneGm12878H3k4me2StdAlnRep2.bam}{\nolinkurl{H3k4me2}}\\
\href{https://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeBroadHistone/wgEncodeBroadHistoneGm12878H3k4me3StdAlnRep2.bam}{\nolinkurl{H3k4me3}}\\
\href{https://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeBroadHistone/wgEncodeBroadHistoneGm12878H3k79me2StdAlnRep2.bam}{\nolinkurl{H3k79me2}}\\
\href{https://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeBroadHistone/wgEncodeBroadHistoneGm12878H3k9acStdAlnRep2.bam}{\nolinkurl{H3k9ac}}\\
\href{https://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeBroadHistone/wgEncodeBroadHistoneGm12878H3k9me3StdAlnRep2.bam}{\nolinkurl{H3k9me3}}\\
\href{https://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeBroadHistone/wgEncodeBroadHistoneGm12878H4k20me1StdAlnRep2.bam}{\nolinkurl{H4k20me1}}\\
\href{https://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeOpenChromDnase/wgEncodeOpenChromDnaseGm12878AlnRep2.bam}{\nolinkurl{DNase}}\\
\href{https://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeSydhTfbs/wgEncodeSydhTfbsGm12878Rad21IggrabAlnRep2.bam}{\nolinkurl{Rad21}}\\
\href{https://www.encodeproject.org/files/ENCFF797XWM/@@download/ENCFF797XWM.bam}{\nolinkurl{SMC3}}
\end{minipage}\hfill
\begin{minipage}{0.23\textwidth}
 \textbf{K562}\\
 \textbf{replicate 1}\\
\href{https://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeBroadHistone/wgEncodeBroadHistoneK562CtcfStdAlnRep1.bam}{\nolinkurl{CTCF}}\\
\href{https://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeBroadHistone/wgEncodeBroadHistoneK562H3k27acStdAlnRep1.bam}{\nolinkurl{H3k27ac}}\\
\href{https://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeBroadHistone/wgEncodeBroadHistoneK562H3k27me3StdAlnRep1.bam}{\nolinkurl{H3k27me3}}\\
\href{https://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeBroadHistone/wgEncodeBroadHistoneK562H3k36me3StdAlnRep1.bam}{\nolinkurl{H3k36me3}}\\
\href{https://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeBroadHistone/wgEncodeBroadHistoneK562H3k4me1StdAlnRep1.bam}{\nolinkurl{H3k4me1}}\\
\href{https://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeBroadHistone/wgEncodeBroadHistoneK562H3k4me2StdAlnRep1.bam}{\nolinkurl{H3k4me2}}\\
\href{https://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeBroadHistone/wgEncodeBroadHistoneK562H3k4me3StdAlnRep1.bam}{\nolinkurl{H3k4me3}}\\
\href{https://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeBroadHistone/wgEncodeBroadHistoneK562H3k79me2StdAlnRep1.bam}{\nolinkurl{H3k79me2}}\\
\href{https://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeBroadHistone/wgEncodeBroadHistoneK562H3k9acStdAlnRep1.bam}{\nolinkurl{H3k9ac}}\\
\href{https://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeBroadHistone/wgEncodeBroadHistoneK562H3k9me3StdAlnRep1.bam}{\nolinkurl{H3k9me3}}\\
\href{https://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeBroadHistone/wgEncodeBroadHistoneK562H4k20me1StdAlnRep1.bam}{\nolinkurl{H4k20me1}}\\
\href{https://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeOpenChromDnase/wgEncodeOpenChromDnaseK562AlnRep1V2.bam}{\nolinkurl{DNase}}\\
\href{https://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeSydhTfbs/wgEncodeSydhTfbsK562Rad21StdAlnRep1.bam}{\nolinkurl{Rad21}}\\
\href{https://www.encodeproject.org/files/ENCFF359UWD/@@download/ENCFF359UWD.bam}{\nolinkurl{SMC3}}\\
\end{minipage}\hfill
\begin{minipage}{0.23\textwidth}
 \textbf{K562}\\
 \textbf{replicate 2}\\
\href{https://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeBroadHistone/wgEncodeBroadHistoneK562CtcfStdAlnRep2.bam}{\nolinkurl{CTCF}}\\
\href{https://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeBroadHistone/wgEncodeBroadHistoneK562H3k27acStdAlnRep2.bam}{\nolinkurl{H3k27ac}}\\
\href{https://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeBroadHistone/wgEncodeBroadHistoneK562H3k27me3StdAlnRep2.bam}{\nolinkurl{H3k27me3}}\\
\href{https://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeBroadHistone/wgEncodeBroadHistoneK562H3k36me3StdAlnRep2.bam}{\nolinkurl{H3k36me3}}\\
\href{https://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeBroadHistone/wgEncodeBroadHistoneK562H3k4me1StdAlnRep2.bam}{\nolinkurl{H3k4me1}}\\
\href{https://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeBroadHistone/wgEncodeBroadHistoneK562H3k4me2StdAlnRep2.bam}{\nolinkurl{H3k4me2}}\\
\href{https://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeBroadHistone/wgEncodeBroadHistoneK562H3k4me3StdAlnRep2.bam}{\nolinkurl{H3k4me3}}\\
\href{https://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeBroadHistone/wgEncodeBroadHistoneK562H3k79me2StdAlnRep2.bam}{\nolinkurl{H3k79me2}}\\
\href{https://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeBroadHistone/wgEncodeBroadHistoneK562H3k9acStdAlnRep2.bam}{\nolinkurl{H3k9ac}}\\
\href{https://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeBroadHistone/wgEncodeBroadHistoneK562H3k9me3StdAlnRep2.bam}{\nolinkurl{H3k9me3}}\\
\href{https://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeBroadHistone/wgEncodeBroadHistoneK562H4k20me1StdAlnRep2.bam}{\nolinkurl{H4k20me1}}\\
\href{https://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeOpenChromDnase/wgEncodeOpenChromDnaseK562AlnRep2V2.bam}{\nolinkurl{DNase}}\\
\href{https://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeSydhTfbs/wgEncodeSydhTfbsK562Rad21StdAlnRep2.bam}{\nolinkurl{Rad21}}\\
\href{https://www.encodeproject.org/files/ENCFF845WVY/@@download/ENCFF845WVY.bam}{\nolinkurl{SMC3}}\\
\end{minipage}

\clearpage
\subsection{Listings}
\begin{lstlisting}[language=bash, caption=hic to cooler, label=list:methods:hic2cool]
  #bash-style code
  #convert from hic to cooler, single resolution
  #MATRIXHIC is a matrix in .hic format
  hic2cool convert -r 5000 $MATRIXHIC matrix_5k.cool
  #coarsen the matrix from 5k to 25k, for example
  cooler coarsen -k 5 matrix_5k.cool -o matrix_25k.cool
  #versions used for thesis
  #hic2cool 0.8.3, cooler 0.8.10
\end{lstlisting}


\begin{lstlisting}[language=bash, caption=bam to bigwig, label=list:methods:bamtobigwig]
 #bash-style code
 #indexing a bam file
 samtools index ${BAMFILE} ${BAMFILE%bam}.bai
 #creating a bigwig file from the bam file above
 OUTFILE="${BAMFILE%bam}bigwig"
 hg19SIZE="2685511504"
 COMMAND="--numberOfProcessors 10 --bam ${BAMFILE}" 
 COMMAND="${COMMAND} --outFileName ${OUTFILE}"
 COMMAND="${COMMAND} --outFileFormat bigwig" 
 COMMAND="${COMMAND} --binSize 5000 --normalizeUsing RPGC"
 COMMAND="${COMMAND} --effectiveGenomeSize ${hg19SIZE}"
 COMMAND="${COMMAND} --scaleFactor 1.0 --extendReads 200"
 COMMAND="${COMMAND} --minMappingQuality 30"
 bamCoverage ${COMMAND}
 #computing mean from replicate 1 and 2 bigwig files
 REPLICATE1="${FOLDER1}${PROTEIN}.bigwig"
 REPLICATE2="${FOLDER2}${PROTEIN}.bigwig"
 OUTFILE="${OUTFOLDER}${PROTEIN}.bigwig"
 COMMAND="-b1 ${REPLICATE1} -b2 ${REPLICATE2}"
 COMMAND="${COMMAND} -o ${OUTFILE} -of bigwig"
 COMMAND="${COMMAND} --operation mean -bs 5000"
 COMMAND="${COMMAND} -p 10 -v"
 bigwigCompare ${COMMAND}
 #versions used for thesis
 #samtools 1.9, bamCoverage 3.5.0, bigwigCompare 3.5.0
\end{lstlisting}


\clearpage
\begin{landscape}
\subsection{Hardware}
 \begin{table}[hbp]
\small
\centering
\begin{tabular}{c|lllllllccll}
\hline
\textbf{}        &    & \multicolumn{5}{c}{\textbf{CPU}}                               &  & \multicolumn{2}{c}{\textbf{GPU}} &  & \multicolumn{1}{c}{\textbf{RAM}} \\
\textbf{machine} & \# & \multicolumn{1}{c}{make/model} & freq. & L1   & L2    & L3     &  & make/model              & memory &  & in GB                            \\ \hline
1                & 8  & AMD EPYC 7742 64-Core          & 2.25  & 64   & 512   & 16384  &  & NVIDIA Tesla T4 TU104GL & 15109  &  & 20                               \\
2                & 40 & Intel Xeon E5-2630 v4          & 2.20  & 32   & 4096  & 16384  &  & NVIDIA Tesla T4 TU104GL & 15109  &  & 116                              \\
3                & 20 & AMD EPYC 7351P 16-Core         & 2.40  & 1300 & 10000 & 320000 &  & --                      & --     &  & 116                              \\ \hline
\end{tabular}
\caption{key figures of hardware used throughout the thesis}\label{tab:appendix:hardware}
\end{table}
Note: CPU frequency in GHz; L1,L2,L3 cache in KB; GPU memory in MiB; RAM (main memory) in GB
\end{landscape}


\subsection{Results of pre-training the DNN-embedding} \label{sec:appendix:pretraining_results}
\begin{figure}[h!] %DNN64 for weight transfer, pearson and progress
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/DNN_64/}{pearson_chr03.pdf_tex}}
        \caption{chr3}
    \end{subfigure} \hfill
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/DNN_64/}{pearson_chr05.pdf_tex}}
        \caption{chr5}
    \end{subfigure}\\[5mm]
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/DNN_64/}{pearson_chr10.pdf_tex}}
        \caption{chr10}
    \end{subfigure}\hfill
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/DNN_64/}{pearson_chr19.pdf_tex}}
        \caption{chr19}
    \end{subfigure}\\[3mm]
    \centering
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/DNN_64/}{pearson_chr21.pdf_tex}}
        \caption{chr21}
    \end{subfigure} \hfill
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/DNN_64/}{lossOverEpochs.pdf_tex}}
        \caption{learning progress} \label{fig:results:DNN64_lossEpochs}
    \end{subfigure}
    \caption{results\,/\,metrics DNN, windowsize 64, test chromosomes}   \label{fig:results:DNN64_pearson}
\end{figure}
\begin{figure}[p] %DNN64, for weight transfer, matrices
    \begin{subfigure}{\textwidth}
        \centering
        \scriptsize
        \import{figures/DNN_64/}{pred00250_chr21_030-040.pdf_tex}
        \caption{example  region 1} \label{fig:results:DNN64_r1}
    \end{subfigure}\\[6mm]
    \begin{subfigure}{\textwidth}
        \centering
        \scriptsize
        \import{figures/DNN_64/}{pred00250_chr19_030-040.pdf_tex}
        \caption{example region 2} \label{fig:results:DNN64_r2}
    \end{subfigure}\\[6mm]
    \begin{subfigure}{\textwidth}
        \centering
        \scriptsize
        \import{figures/DNN_64/}{pred00250_chr3_030-040.pdf_tex}
        \caption{example region 3} \label{fig:results:DNN64_r3}
    \end{subfigure}
    \caption{example predictions, DNN, $w=64$, 250 epochs} \label{fig:results:DNN_matrices}
\end{figure}

\subsection{cGAN trained on single chromosomes predicting usual test chromosomes}
\begin{figure}[h!] %cGAN single chrom 14, pearson and progress, predicting usual test chroms 3,5,10,19,21
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/GAN_64_14only/}{pearson_chr03.pdf_tex}}
        \caption{chr3}
    \end{subfigure} \hfill
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/GAN_64_14only/}{pearson_chr05.pdf_tex}}
        \caption{chr5}
    \end{subfigure}\\[5mm]
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/GAN_64_14only/}{pearson_chr10.pdf_tex}}
        \caption{chr10}
    \end{subfigure}\hfill
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/GAN_64_14only/}{pearson_chr19.pdf_tex}}
        \caption{chr19}
    \end{subfigure}\\[3mm]
    \centering
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/GAN_64_14only/}{pearson_chr21.pdf_tex}}
        \caption{chr21}
    \end{subfigure} \hfill
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/GAN_64_14only/}{lossOverEpochs.pdf_tex}}
        \caption{learning progress} \label{fig:appendix:GAN64-14single_lossEpochs}
    \end{subfigure}
    \caption{results\,/\,metrics cGAN, $w=64$, trained on chr14 only, typical test chromosomes}   \label{fig:appendix:GAN64-14single_pearson}
\end{figure}
\begin{figure}[p] %cGAN single chrom 14, matrices
    \begin{subfigure}{\textwidth}
        \centering
        \resizebox{0.9\textwidth}{!}{
        \scriptsize
        \import{figures/GAN_64_14only/}{pred00100_chr21_030-040.pdf_tex}}
        \caption{example  region 1} \label{fig:results:GAN64-14single_r1}
    \end{subfigure}\\[3mm]
    \begin{subfigure}{\textwidth}
        \centering
        \resizebox{0.9\textwidth}{!}{
        \scriptsize
        \import{figures/GAN_64_14only/}{pred00100_chr19_030-040.pdf_tex}}
        \caption{example region 2} \label{fig:results:GAN64-14single_r2}
    \end{subfigure}\\[3mm]
    \begin{subfigure}{\textwidth}
        \centering
        \resizebox{0.9\textwidth}{!}{
        \scriptsize
        \import{figures/GAN_64_14only/}{pred00100_chr3_030-040.pdf_tex}}
        \caption{example region 3} \label{fig:results:GAN64-14single_r3}
    \end{subfigure}
    \caption{example predictions cGAN, trained on chr14 only, $w=64$, 100 epochs} 
     \label{fig:results:GAN64-14single_matrices}
\end{figure}
\begin{figure}[p] %cGAN single chrom 17, pearson and progress, predicting usual test chroms 3,5,10,19,21
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/GAN_64_17only/}{pearson_chr03.pdf_tex}}
        \caption{chr3}
    \end{subfigure} \hfill
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/GAN_64_17only/}{pearson_chr05.pdf_tex}}
        \caption{chr5}
    \end{subfigure}\\[5mm]
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/GAN_64_17only/}{pearson_chr10.pdf_tex}}
        \caption{chr10}
    \end{subfigure}\hfill
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/GAN_64_17only/}{pearson_chr19.pdf_tex}}
        \caption{chr19}
    \end{subfigure}\\[3mm]
    \centering
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/GAN_64_17only/}{pearson_chr21.pdf_tex}}
        \caption{chr21}
    \end{subfigure} \hfill
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/GAN_64_17only/}{lossOverEpochs.pdf_tex}}
        \caption{learning progress} \label{fig:appendix:GAN64-17single_lossEpochs}
    \end{subfigure}
    \caption{results\,/\,metrics cGAN, $w=64$, trained on chr17 only, typical test chromosomes}   \label{fig:appendix:GAN64-17single_pearson}
\end{figure}
\begin{figure}[p] %cGAN single chrom 14, matrices
    \begin{subfigure}{\textwidth}
        \centering
        \resizebox{0.9\textwidth}{!}{
        \scriptsize
        \import{figures/GAN_64_17only/}{pred00100_chr21_030-040.pdf_tex}}
        \caption{example  region 1} \label{fig:results:GAN64-17single_r1}
    \end{subfigure}\\[3mm]
    \begin{subfigure}{\textwidth}
        \centering
        \resizebox{0.9\textwidth}{!}{
        \scriptsize
        \import{figures/GAN_64_17only/}{pred00100_chr19_030-040.pdf_tex}}
        \caption{example region 2} \label{fig:results:GAN64-17single_r2}
    \end{subfigure}\\[3mm]
    \begin{subfigure}{\textwidth}
        \centering
        \resizebox{0.9\textwidth}{!}{
        \scriptsize
        \import{figures/GAN_64_17only/}{pred00100_chr3_030-040.pdf_tex}}
        \caption{example region 3} \label{fig:results:GAN64-17single_r3}
    \end{subfigure}
    \caption{example predictions cGAN, trained on chr17 only, $w=64$, 100 epochs} 
     \label{fig:results:GAN64-17single_matrices}
\end{figure}
