\section{Appendix}

\subsection{Chromatin feature download details} \label{sec:chromFeat_download_links}
The basic download paths for all chromatin feature files in .bam format are\\
\small{\url{https://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeBroadHistone/}}\\
\small{\url{https://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeOpenChromDnase/}}\\
\small{\url{https://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeSydhTfbs/}}\\
for CTCF/Histones, DNase, RAD21/SMC3, respectively. 
The actual files for replicates 1 and 2 are then easily be found by searching for ``$\mathit{CELL}\_\mathit{FEATURE}$'', 
where $\mathit{CELL}$ is the Cell line (e.\,g. GM12878) and $\mathit{FEATURE}$
is the chromatin feature of interest, e.\,g. CTCF, H3K27ac and so on.
For convenience, the pdf version of this document contains clickable links below.

\begin{minipage}{0.23\textwidth}
\textbf{GM12878} \\
\textbf{replicate 1}\\
\href{https://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeBroadHistone/wgEncodeBroadHistoneGm12878CtcfStdAlnRep1.bam}{\nolinkurl{CTCF}}\\
\href{https://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeBroadHistone/wgEncodeBroadHistoneGm12878H3k27acStdAlnRep1.bam}{\nolinkurl{H3k27ac}}\\
\href{https://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeBroadHistone/wgEncodeBroadHistoneGm12878H3k27me3StdAlnRep1.bam}{\nolinkurl{H3k27me3}}\\
\href{https://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeBroadHistone/wgEncodeBroadHistoneGm12878H3k36me3StdAlnRep1.bam}{\nolinkurl{H3k36me3}}\\
\href{https://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeBroadHistone/wgEncodeBroadHistoneGm12878H3k4me1StdAlnRep1.bam}{\nolinkurl{H3k4me1}}\\
\href{https://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeBroadHistone/wgEncodeBroadHistoneGm12878H3k4me2StdAlnRep1.bam}{\nolinkurl{H3k4me2}}\\
\href{https://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeBroadHistone/wgEncodeBroadHistoneGm12878H3k4me3StdAlnRep1.bam}{\nolinkurl{H3k4me3}}\\
\href{https://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeBroadHistone/wgEncodeBroadHistoneGm12878H3k79me2StdAlnRep1.bam}{\nolinkurl{H3k79me2}}\\
\href{https://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeBroadHistone/wgEncodeBroadHistoneGm12878H3k9acStdAlnRep1.bam}{\nolinkurl{H3k9ac}}\\
\href{https://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeBroadHistone/wgEncodeBroadHistoneGm12878H3k9me3StdAlnRep1.bam}{\nolinkurl{H3k9me3}}\\
\href{https://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeBroadHistone/wgEncodeBroadHistoneGm12878H4k20me1StdAlnRep1.bam}{\nolinkurl{H4k20me1}}\\
\href{https://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeOpenChromDnase/wgEncodeOpenChromDnaseGm12878AlnRep1.bam}{\nolinkurl{DNase}}\\
\href{https://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeSydhTfbs/wgEncodeSydhTfbsGm12878Rad21IggrabAlnRep1.bam}{\nolinkurl{Rad21}}\\
\href{https://www.encodeproject.org/files/ENCFF415OZI/@@download/ENCFF415OZI.bam}{\nolinkurl{SMC3}}
\end{minipage}\hfill
\begin{minipage}{0.23\textwidth}
\textbf{GM12878} \\
\textbf{replicate 2}\\
\href{https://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeBroadHistone/wgEncodeBroadHistoneGm12878CtcfStdAlnRep2.bam}{\nolinkurl{CTCF}}\\
\href{https://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeBroadHistone/wgEncodeBroadHistoneGm12878H3k27acStdAlnRep2.bam}{\nolinkurl{H3k27ac}}\\
\href{https://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeBroadHistone/wgEncodeBroadHistoneGm12878H3k27me3StdAlnRep2.bam}{\nolinkurl{H3k27me3}}\\
\href{https://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeBroadHistone/wgEncodeBroadHistoneGm12878H3k36me3StdAlnRep2.bam}{\nolinkurl{H3k36me3}}\\
\href{https://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeBroadHistone/wgEncodeBroadHistoneGm12878H3k4me1StdAlnRep2.bam}{\nolinkurl{H3k4me1}}\\
\href{https://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeBroadHistone/wgEncodeBroadHistoneGm12878H3k4me2StdAlnRep2.bam}{\nolinkurl{H3k4me2}}\\
\href{https://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeBroadHistone/wgEncodeBroadHistoneGm12878H3k4me3StdAlnRep2.bam}{\nolinkurl{H3k4me3}}\\
\href{https://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeBroadHistone/wgEncodeBroadHistoneGm12878H3k79me2StdAlnRep2.bam}{\nolinkurl{H3k79me2}}\\
\href{https://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeBroadHistone/wgEncodeBroadHistoneGm12878H3k9acStdAlnRep2.bam}{\nolinkurl{H3k9ac}}\\
\href{https://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeBroadHistone/wgEncodeBroadHistoneGm12878H3k9me3StdAlnRep2.bam}{\nolinkurl{H3k9me3}}\\
\href{https://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeBroadHistone/wgEncodeBroadHistoneGm12878H4k20me1StdAlnRep2.bam}{\nolinkurl{H4k20me1}}\\
\href{https://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeOpenChromDnase/wgEncodeOpenChromDnaseGm12878AlnRep2.bam}{\nolinkurl{DNase}}\\
\href{https://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeSydhTfbs/wgEncodeSydhTfbsGm12878Rad21IggrabAlnRep2.bam}{\nolinkurl{Rad21}}\\
\href{https://www.encodeproject.org/files/ENCFF797XWM/@@download/ENCFF797XWM.bam}{\nolinkurl{SMC3}}
\end{minipage}\hfill
\begin{minipage}{0.23\textwidth}
 \textbf{K562}\\
 \textbf{replicate 1}\\
\href{https://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeBroadHistone/wgEncodeBroadHistoneK562CtcfStdAlnRep1.bam}{\nolinkurl{CTCF}}\\
\href{https://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeBroadHistone/wgEncodeBroadHistoneK562H3k27acStdAlnRep1.bam}{\nolinkurl{H3k27ac}}\\
\href{https://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeBroadHistone/wgEncodeBroadHistoneK562H3k27me3StdAlnRep1.bam}{\nolinkurl{H3k27me3}}\\
\href{https://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeBroadHistone/wgEncodeBroadHistoneK562H3k36me3StdAlnRep1.bam}{\nolinkurl{H3k36me3}}\\
\href{https://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeBroadHistone/wgEncodeBroadHistoneK562H3k4me1StdAlnRep1.bam}{\nolinkurl{H3k4me1}}\\
\href{https://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeBroadHistone/wgEncodeBroadHistoneK562H3k4me2StdAlnRep1.bam}{\nolinkurl{H3k4me2}}\\
\href{https://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeBroadHistone/wgEncodeBroadHistoneK562H3k4me3StdAlnRep1.bam}{\nolinkurl{H3k4me3}}\\
\href{https://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeBroadHistone/wgEncodeBroadHistoneK562H3k79me2StdAlnRep1.bam}{\nolinkurl{H3k79me2}}\\
\href{https://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeBroadHistone/wgEncodeBroadHistoneK562H3k9acStdAlnRep1.bam}{\nolinkurl{H3k9ac}}\\
\href{https://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeBroadHistone/wgEncodeBroadHistoneK562H3k9me3StdAlnRep1.bam}{\nolinkurl{H3k9me3}}\\
\href{https://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeBroadHistone/wgEncodeBroadHistoneK562H4k20me1StdAlnRep1.bam}{\nolinkurl{H4k20me1}}\\
\href{https://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeOpenChromDnase/wgEncodeOpenChromDnaseK562AlnRep1V2.bam}{\nolinkurl{DNase}}\\
\href{https://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeSydhTfbs/wgEncodeSydhTfbsK562Rad21StdAlnRep1.bam}{\nolinkurl{Rad21}}\\
\href{https://www.encodeproject.org/files/ENCFF359UWD/@@download/ENCFF359UWD.bam}{\nolinkurl{SMC3}}\\
\end{minipage}\hfill
\begin{minipage}{0.23\textwidth}
 \textbf{K562}\\
 \textbf{replicate 2}\\
\href{https://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeBroadHistone/wgEncodeBroadHistoneK562CtcfStdAlnRep2.bam}{\nolinkurl{CTCF}}\\
\href{https://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeBroadHistone/wgEncodeBroadHistoneK562H3k27acStdAlnRep2.bam}{\nolinkurl{H3k27ac}}\\
\href{https://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeBroadHistone/wgEncodeBroadHistoneK562H3k27me3StdAlnRep2.bam}{\nolinkurl{H3k27me3}}\\
\href{https://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeBroadHistone/wgEncodeBroadHistoneK562H3k36me3StdAlnRep2.bam}{\nolinkurl{H3k36me3}}\\
\href{https://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeBroadHistone/wgEncodeBroadHistoneK562H3k4me1StdAlnRep2.bam}{\nolinkurl{H3k4me1}}\\
\href{https://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeBroadHistone/wgEncodeBroadHistoneK562H3k4me2StdAlnRep2.bam}{\nolinkurl{H3k4me2}}\\
\href{https://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeBroadHistone/wgEncodeBroadHistoneK562H3k4me3StdAlnRep2.bam}{\nolinkurl{H3k4me3}}\\
\href{https://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeBroadHistone/wgEncodeBroadHistoneK562H3k79me2StdAlnRep2.bam}{\nolinkurl{H3k79me2}}\\
\href{https://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeBroadHistone/wgEncodeBroadHistoneK562H3k9acStdAlnRep2.bam}{\nolinkurl{H3k9ac}}\\
\href{https://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeBroadHistone/wgEncodeBroadHistoneK562H3k9me3StdAlnRep2.bam}{\nolinkurl{H3k9me3}}\\
\href{https://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeBroadHistone/wgEncodeBroadHistoneK562H4k20me1StdAlnRep2.bam}{\nolinkurl{H4k20me1}}\\
\href{https://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeOpenChromDnase/wgEncodeOpenChromDnaseK562AlnRep2V2.bam}{\nolinkurl{DNase}}\\
\href{https://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeSydhTfbs/wgEncodeSydhTfbsK562Rad21StdAlnRep2.bam}{\nolinkurl{Rad21}}\\
\href{https://www.encodeproject.org/files/ENCFF845WVY/@@download/ENCFF845WVY.bam}{\nolinkurl{SMC3}}\\
\end{minipage}
\clearpage

\begin{table}[htbp]
\centering
\begin{tabular}{lllllll}
\cline{1-3} \cline{5-7}
                 & \multicolumn{2}{c}{\textbf{accession}} &  & & \multicolumn{2}{c}{\textbf{accession}}  \\
\textbf{feature} & \multicolumn{1}{c}{reads}           & \multicolumn{1}{c}{peaks}            &  & \textbf{feature} & \multicolumn{1}{c}{reads} & \multicolumn{1}{c}{peaks}    \\ \cline{1-3} \cline{5-7} 
BEAF             & SRR1164521          & GSE51986         &  & H3K9me1          & SRR869989 & GSE47240 \\
Chro Chriz WR    & SRR869929           & GSE47263         &  & H3K9me2          & SRR869864 & GSE47247 \\
CP190-HB         & SRR869732           & GSE47234         &  & H3K9me3          & SRR869860 & GSE47246 \\
CTCF             & SRR869933           & GSE47264         &  & H4               & SRR870049 & GSE47291 \\
dMi              & SRR869937           & GSE47265         &  & H4K16acM         & SRR869856 & GSE47245 \\
dRING            & SRR869941           & GSE47266         &  & H4K20me1         & SRR870057 & GSE47293 \\
GAF              & SRR869740           & GSE47236         &  & HP1              & SRR869718 & GSE47231 \\
H1               & SRR870123           & GSE47309         &  & HP1b             & SRR870061 & GSE47294 \\
H2AV             & SRR869950           & GSE47268         &  & HP1c             & SRR870183 & GSE47323 \\
H2B-ubiq         & SRR869956           & GSE47269         &  & HP2              & SRR870138 & GSE47312 \\
H3               & SRR869965           & GSE47271         &  & HP4              & SRR869880 & GSE47251 \\
H3K18ac          & SRR869977           & GSE47274         &  & JHDM1            & SRR870069 & GSE47296 \\
H3K23ac          & SRR869981           & GSE47275         &  & LSD1             & SRR870073 & GSE47297 \\
H3K27ac          & SRR869744           & GSE47237         &  & MBD-R2           & SRR870077 & GSE47298 \\
H3K27me2         & SRR869993           & GSE47277         &  & MOF              & SRR870081 & GSE47299 \\
H3K27me3         & SRR869712           & GSE47230         &  & NURF301          & SRR870085 & GSE47300 \\
H3K36me1         & SRR869839           & GSE47241         &  & Pc               & SRR869724 & GSE47232 \\
H3K36me2         & SRR870231           & GSE47335         &  & POF              & SRR870089 & GSE47301 \\
H3K36me3         & SRR869900           & GSE47256         &  & Psc              & SRR869736 & GSE47235 \\
H3K4me1          & SRR870009           & GSE47281         &  & RNA Pol II       & SRR870093 & GSE47302 \\
H3K4me3          & SRR870025           & GSE47285         &  & RPD3             & SRR870105 & GSE47305 \\
H3K79me1         & SRR869748           & GSE47239         &  & SU(HW)-HB        & SRR869728 & GSE47233 \\
H3K79me2         & SRR869884           & GSE47252         &  & Su(var)3         & SRR869876 & GSE47250 \\
H3K79me3         & SRR870191           & GSE47325         &  & ZW5              & SRR870117 & GSE47308 \\
H3K9acS10P       & SRR870033           & GSE47287         &  &                  &           &          \\ \hline
\end{tabular}
\caption[Chromatin features from D. Melanogaster used for the comparison DNN / cGAN / Farré et al. \cite{Farre2018a}]{Chromatin features from D. Melanogaster used for the comparison\\DNN / cGAN / Farré et al. \cite{Farre2018a}}
\label{tab:appendix:features_farre}
\end{table}
\clearpage
\subsection{Listings}
\begin{lstlisting}[language=bash, caption=Hic to cooler, label=list:methods:hic2cool]
  #bash-style code
  #convert from hic to cooler, single resolution
  #MATRIXHIC is a matrix in .hic format
  hic2cool convert -r 5000 $MATRIXHIC matrix_5k.cool
  #coarsen the matrix from 5k to 25k, for example
  cooler coarsen -k 5 matrix_5k.cool -o matrix_25k.cool
  #versions used for thesis
  #hic2cool 0.8.3, cooler 0.8.10
\end{lstlisting}


\begin{lstlisting}[language=bash, caption=Bam to bigwig, label=list:methods:bamtobigwig]
 #bash-style code
 #indexing a bam file
 samtools index ${BAMFILE} ${BAMFILE%bam}.bai
 #creating a bigwig file from the bam file above
 OUTFILE="${BAMFILE%bam}bigwig"
 hg19SIZE="2685511504"
 COMMAND="--numberOfProcessors 10 --bam ${BAMFILE}" 
 COMMAND="${COMMAND} --outFileName ${OUTFILE}"
 COMMAND="${COMMAND} --outFileFormat bigwig" 
 COMMAND="${COMMAND} --binSize 5000 --normalizeUsing RPGC"
 COMMAND="${COMMAND} --effectiveGenomeSize ${hg19SIZE}"
 COMMAND="${COMMAND} --scaleFactor 1.0 --extendReads 200"
 COMMAND="${COMMAND} --minMappingQuality 30"
 bamCoverage ${COMMAND}
 #computing mean from replicate 1 and 2 bigwig files
 REPLICATE1="${FOLDER1}${PROTEIN}.bigwig"
 REPLICATE2="${FOLDER2}${PROTEIN}.bigwig"
 OUTFILE="${OUTFOLDER}${PROTEIN}.bigwig"
 COMMAND="-b1 ${REPLICATE1} -b2 ${REPLICATE2}"
 COMMAND="${COMMAND} -o ${OUTFILE} -of bigwig"
 COMMAND="${COMMAND} --operation mean -bs 5000"
 COMMAND="${COMMAND} -p 10 -v"
 bigwigCompare ${COMMAND}
 #versions used for thesis
 #samtools 1.9, bamCoverage 3.5.0, bigwigCompare 3.5.0
\end{lstlisting}

\begin{lstlisting}[language=bash, caption=Bam to bedgraph, label=list:methods:bam2bedgraph]
chromosome="${2}"
bamfolder="${1}"

shopt -s nullglob
for i in ${1}*.bam
do
 echo "converting: ${i} - chrom ${2}"
 tmpfile="${i}.tmp"
 countfile="${i%.bam}_${chromosome}.count"
 samtools view -b ${i} ${chromosome} > ${tmpfile}
 cmd="-ibam ${tmpfile} -bg"
 bedtools genomecov ${cmd} > ${countfile}
 rm ${tmpfile}
done
\end{lstlisting}
\begin{lstlisting}[language=bash, caption=Bedgraph to HiC-Reg input format, label=list:methods:bedgraph2hicreg]
inputfolder="${1}"
#following example files are provided in HiC-Reg github
region="HiC-Reg/Scripts/aggregateSignalInRegion/hg19_5kbp_chr17.txt"
sizesfile="HiC-Reg/Scripts/aggregateSignalInRegion/hg19.fa.fai"

shopt -s nullglob
for i in ${1}*.count
do
 echo "converting: ${i}"
 outfile="${i%.count}.txt"
 cmd="${region} ${sizesfile} ${i} ${outfile}"
 ./aggregateSignal_v1 ${cmd} #provided by HiC-Reg
done
\end{lstlisting}

\begin{lstlisting}[language=bash, caption=Cooler matrices to HiC-Reg input format, label=list:methods:convertForHicReg_bash]
#"combined_30" GM12878 or K562 matrix from Rao et al. at 5kb resolution
inMatrix=${1} 
#filename for cooler matrix (*.cool) with VC_SQRT correction
outMatrix=${2} 
chrom=${3}

cmd="-m ${inMatrix}"
cmd="${cmd} -o ${outMatrix}"
cmd="${cmd} --inputFormat cool --outputFormat cool" 
cmd="${cmd} --correction_name VC_SQRT --store_applied_correction"
cmd="${cmd} --correction_division"
cmd="${cmd} --chromosome ${chrom}" 
hicConvertFormat ${cmd}

cooler dump ${outMatrix} --join -o ${outMatrix%.cool}.txt
params="--infile ${outMatrix%.cool}.txt"
params="${params} --outfile ${outMatrix%.cool}_converted.txt"
python matrix_conversion.py ${params} #see below for details
\end{lstlisting}

\begin{lstlisting}[language=python, caption=Custom text file conversion for HiC-Reg's matrix input, label=list:methods:convertForHicReg_python]
import click
import pandas as pd

@click.option("--infile", required=True, type=click.Path(exists=True, readable=True, dir_okay=False), help="text file created by cooler dump with --join")
@click.option("--outfile", required=False, type=click.Path(writable=True, dir_okay=False))
@click.command()
def convert(infile, outfile):
	if outfile is None:
		outfile = infile
	try:
		df = pd.read_csv(infile, header=None, sep="\t", index_col=False)
	except Exception as e:
		msg = str(e)
		msg += "\ncould not read csv file, wrong format?"
		raise SystemExit(msg)
	if df.shape[1] != 7:
		msg = "Read file with wrong format, number of columns should be 7 but is {:d}"
		msg = msg.format(df.shape[1])
		raise SystemExit(msg)
	#check if the matrix contains nans
	if df.isnull().values.any():
		msg = "WARNING: there are nans in the dataframe"
		print(msg)
	#create the text-encoded matrix format for HiC-Reg
	df["start"] = "chr" + df.loc[:,0].astype(str) + "_" + df.loc[:,1].astype(str) + "_" + df.loc[:,2].astype(str)
	df["end"] = "chr" + df.loc[:,3].astype(str) + "_" + df.loc[:,4].astype(str) + "_" + df.loc[:,5].astype(str)
	df["count"] = df.loc[:,6].astype("float32")
	df.drop(columns=[0,1,2,3,4,5,6], inplace=True)
	df["count"].fillna(value=0.0, inplace=True)
	#remove matrix diagonal since it is missing in the HiC-Reg examples, too
	diagFilter = df["start"] == df["end"]
	df = df.loc[~diagFilter]
	
	df.to_csv(outfile, sep="\t", columns=["start","end","count"], header=False, index=False, float_format="%.4f")
	print("first few lines of matrix txt file:\n", df.head())

if __name__ == "__main__":
	convert()
\end{lstlisting}

\begin{lstlisting}[language=python, caption=Custom script to convert HiC-Reg's text output to cooler format, label=list:methods:hicreg2cool]
import click
import pandas as pd
import cooler
import os
import numpy as np

@click.option("--filenames", "-f", required=True,
				type=click.Path(exists=True, readable=True, dir_okay=False),
				multiple=True,
				help="testset_error file(s) from HiC-Reg") 
@click.option("--chromsize", "-cs", required=False,
				type=click.IntRange(min=1), 
				help="Size of chromosome. Will be derived from data, if not specified")
@click.option("--outfolder", "-o", required=True,
				type=click.Path(writable=True, file_okay=False, exists=True),
				help="Folder where the coolers will be placed")
@click.option("--exponentiate", "-e", required=False, type=bool, default=False, show_default=True, help="exponentiate count values")
@click.command()
def reconstruct(filenames, chromsize, outfolder, exponentiate):
	#read the the text files from HiC-Reg
	#skip the first row, since it contains the incomplete header with missing "Distance"
	try:
		dataframes = [pd.read_csv(filename, names=["Pair", "TrueValue", "PredictedValue", "SquaredErr", "Distance"], 
									sep="\t", 
									dtype={"Pair": str, "TrueValue": float, "PredictedValue": float, "SquaredErr": float, "Distance": int}, 
									skiprows=[0]) for filename in filenames]
	except Exception as e:
		msg = str(e) + "\nCould not read one of the files, wrong format?"
		raise SystemExit(msg)
		
	#print number of pairs in each file
	for i, df in enumerate(dataframes):
		print("file {:d} - {:d} pairs ({:s})".format(i, df.shape[0], filenames[i]))
	
	for df in dataframes:
		#extract the start- and end positions of all pairs and the chromosome
		df["Pair"] = df["Pair"].str.replace(pat="-", repl="_")
		splitDf = df["Pair"].str.split(pat="_", expand=True)
		df["chrom"] = splitDf.iloc[:,0]
		df["start_L"] = splitDf.iloc[:,1].astype("int32")
		df["end_L"] = splitDf.iloc[:,2].astype("int32")
		#df["chr_R"] = splitDf.iloc[:,3] #not needed here, always the same as chr_L
		df["start_R"] = splitDf.iloc[:,4].astype("int32")
		df["end_R"] = splitDf.iloc[:,5].astype("int32")
		#compute binsizes
		df["binsizes"] = df["end_L"] - df["start_L"]
		#compute the bin ids
		binsize = df["binsizes"].iloc[0]
		df["bin1_id"] = df["start_L"] // binsize
		df["bin2_id"] = df["start_R"] // binsize
		#if desired, take the values to the power of e to get counts from log-scale values
		if exponentiate == True:
			df["TrueValue"] = np.exp(df["TrueValue"])
			df["PredictedValue"] = np.exp(df["PredictedValue"])
		#drop unnecessary columns
		df.drop(columns=["Pair", "Distance", "SquaredErr", "end_L", "end_R", "start_L", "start_R"], inplace=True)	
	
	#check if the binsizes are equal in the single dataframes before merging the dataframes
	binsizes = [df["binsizes"].iloc[0] for df in dataframes]
	binsizes = list(set(binsizes))
	if len(binsizes) != 1:
		msg = "Binsizes not equal: {:s}".format(", ".join([str(b) for b in binsizes]))
		raise SystemExit(msg)
	else:
		print("detected binsize: {:d}".format(binsizes[0])) 
	
	#concat the dataframes from the files
	joinDf = pd.concat(dataframes, ignore_index=True)
	joinDf.drop(columns=["binsizes"], inplace=True)
	
	#When using train_i and test_i, there should be no duplicate pairs. 
	#Otherwise, e.g. in x-validation settings, take the mean
	chrom = joinDf["chrom"].iloc[0]
	elem_number_before = joinDf.shape[0]
	joinDf = joinDf.groupby(["bin1_id", "bin2_id"])[["TrueValue", "PredictedValue"]].mean().reset_index()
	elem_number_after = joinDf.shape[0]
	if elem_number_after != elem_number_before:
		msg = "Aggregated {:d} duplicates by taking the mean".format(elem_number_before - elem_number_after)
		print(msg)
	
	print("first few elems of joint dataset:\n", joinDf.head())
	print("dataset contains {:d} pairs".format(joinDf.shape[0]))
	
	#if no data given, set chromsize to max(bin id) * bin size + bin size
	maxbin = joinDf[["bin1_id","bin2_id"]].max().max()
	if chromsize is None:
		chromsize = maxbin * binsizes[0] + binsizes[0]
	
	#prepare the bins for cooler
	bins = pd.DataFrame(columns=['chrom','start','end'])
	binStartList = list(range(0, chromsize, binsizes[0]))
	binEndList = list(range(binsizes[0], chromsize, binsizes[0])) + [chromsize]
	bins['start'] = np.uint32(binStartList)
	bins['end'] = np.uint32(binEndList)
	bins["chrom"] = str(chrom)

	#prepare the pixels for cooler
	joinDf.sort_values(by=["bin1_id", "bin2_id"], inplace=True)
	pixels = joinDf[["bin1_id","bin2_id"]].copy()
	pixels["count"] = joinDf["PredictedValue"]
	outfilename = os.path.join(outfolder, "predValues_{:s}.cool".format(str(chrom)))
	#build the coolers
	cooler.create_cooler(outfilename, 
						bins=bins, 
						pixels=pixels, 
						dtypes={'count': np.float64}, 
						ordered=True, 
						metadata={"fromFilenames": filenames})
	pixels["count"] = joinDf["TrueValue"]
	outfilename = os.path.join(outfolder, "trueValues_{:s}.cool".format(str(chrom)))
	cooler.create_cooler(outfilename, 
						bins=bins, 
						pixels=pixels, 
						dtypes={'count': np.float64}, 
						ordered=True, 
						metadata={"fromFilenames": filenames})
	#check and report sparsity
	max_bin_dist = (pixels["bin2_id"] - pixels["bin1_id"]).max()
	full_nr_elements = ((bins.shape[0])**2 + bins.shape[0])//2
	cut_nr = bins.shape[0] - max_bin_dist
	cut_nr_elements = (cut_nr**2 + cut_nr)//2
	exp_nr_elements = full_nr_elements - cut_nr_elements
	sparsity_percent = (pixels.shape[0] / exp_nr_elements)*100
	print("number of bins: {:d}".format(bins.shape[0]))
	print("maxdist: {:d}".format(max_bin_dist * binsizes[0]))
	print("expected interacting pairs: {:d}, available interacting pairs: {:d}".format(exp_nr_elements, pixels.shape[0]))
	print("sparsity: {:.2f}%".format(sparsity_percent))

if __name__ == "__main__":
	reconstruct()
\end{lstlisting}

\begin{lstlisting}[language=bash, caption=Mapping ChIP-seq reads to D. melanogaster ref. genome, label=list:methods:map_drosophila]
fqdir=${1}                  #directory with fastq files
procs="8"                   #number of threads
indexdir="./BDGP5/BDGP5"    #bowtie2 index for BDGP5/dm3

shopt -s nullglob
for fqfile in ${fqdir}*.fastq; do
	samfile="${fqfile%fastq}sam"
	bamfile="${fqfile%fastq}bam"
	echo "processing ${fqfile} => ${samfile} => ${bamfile}"
	bowtie2 -x ${indexdir} -U ${fqfile} -S ${samfile} --no-unal -p ${procs}
	samtools view -S -b ${samfile} > ${bamfile}
	rm ${samfile}
	gzip ${fqfile}
	samtools sort ${bamfile} -o ${bamfile}
	samtools index ${bamfile}
done
\end{lstlisting}



\clearpage
\begin{landscape}
\subsection{Hardware}\label{sec:appendix:hardware}
For the thesis, three virtual machines were used to train the neural networks, see \cref{tab:appendix:hardware}.
All training for \cref{sec:results:cgan} with window size $w=256$ was done on machine 2, 
while computations for  \cref{sec:results:cgan} with window sizes $w \in \{64,128\}$ were done on machine 1.
Computations for sections \ref{sec:initialDNNresults}, \ref{sec:results:wider-longer-etc} and \ref{sec:results:binsize_winsize}
were done on machine 3 without GPU and computations for sections \ref{sec:results:loss_functions} and \ref{sec:results:scorebased} were done on machine 2.

Training the DNN with perception- or score-based losses as well as the \acrshort{cgan} was not reasonably possible without GPU.
For the \acrshort{cgan}, it was found that GPU memory should not fall short of the given values (\cref{tab:appendix:hardware})
to avoid undue limitations on batch sizes and/or window sizes.
\SI{20}{\giga\byte} of main memory, as in machine 2, were not enough to train the \acrshort{cgan} at window size 64 for more than about 150 epochs.
It could not be clarified throughout the thesis whether this was due to a memory leak in tensorflow or due to the chosen implementation.

Training samples were stored as tensorflow tfrecords (on the fly at run time) and a custom pipeline including a shuffle buffer and prefetching 
was employed to balance workload between CPUs and GPU.
This approach was found to be faster than generator-based approaches by a large margin.
 \begin{table}[hbp]
\small
\centering
\begin{tabular}{c|lllllllccll}
\hline
\textbf{}        &    & \multicolumn{5}{c}{\textbf{CPU}}                               &  & \multicolumn{2}{c}{\textbf{GPU}} &  & \multicolumn{1}{c}{\textbf{RAM}} \\
\textbf{machine} & \# & \multicolumn{1}{c}{make/model} & freq. & L1   & L2    & L3     &  & make/model              & memory &  & in GB                            \\ \hline
1                & 8  & AMD EPYC 7742 64-Core          & 2.25  & 64   & 512   & 16384  &  & NVIDIA Tesla T4 TU104GL & 15109  &  & 20                               \\
2                & 40 & Intel Xeon E5-2630 v4          & 2.20  & 32   & 4096  & 16384  &  & NVIDIA Tesla T4 TU104GL & 15109  &  & 116                              \\
3                & 20 & AMD EPYC 7351P 16-Core         & 2.40  & 1300 & 10000 & 320000 &  & --                      & --     &  & 116                              \\ \hline
\end{tabular}
\caption{Key figures of hardware used throughout the thesis}\label{tab:appendix:hardware}
\end{table}
Note: CPU frequency in GHz; L1,L2,L3 cache in KB; GPU memory in MiB; RAM (main memory) in GB
\end{landscape}

\subsection{Further figures}
\subsubsection{Combined loss function}
\begin{figure}[htbp]
    \centering
    \scriptsize
    \raggedright{The figure below shows the effect of choosing relatively large loss weights for the TV loss\\in combined loss functions according to \cref{sec:methods:combined_loss}.\\[2mm]}
    \import{figures/DNN_combiloss/}{pred01000_chr19_030-040_failedTvLoss.pdf_tex}
    \caption{Combined loss with parameters $\lambda_\mathit{MSE}=10^{-5}$, $\lambda_\mathit{VGG}=0.0$, $\lambda_\mathit{TV}=1.0$}
    \label{fig:appendix:failed_tv_loss}
\end{figure}
\clearpage

\subsubsection{Results of pre-training the DNN-embedding} \label{sec:appendix:pretraining_results}
\begin{figure}[h!] %DNN64 for weight transfer, pearson and progress
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/DNN_64/}{pearson_chr03.pdf_tex}}
        \caption{chr3}
    \end{subfigure} \hfill
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/DNN_64/}{pearson_chr05.pdf_tex}}
        \caption{chr5}
    \end{subfigure}\\[5mm]
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/DNN_64/}{pearson_chr10.pdf_tex}}
        \caption{chr10}
    \end{subfigure}\hfill
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/DNN_64/}{pearson_chr19.pdf_tex}}
        \caption{chr19}
    \end{subfigure}\\[3mm]
    \centering
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/DNN_64/}{pearson_chr21.pdf_tex}}
        \caption{chr21}
    \end{subfigure} \hfill
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/DNN_64/}{lossOverEpochs.pdf_tex}}
        \caption{Learning progress} \label{fig:results:DNN64_lossEpochs}
    \end{subfigure}
    \caption{Results\,/\,metrics DNN, $w=64$, test chromosomes}   \label{fig:results:DNN64_pearson}
\end{figure}
\begin{figure}[p] %DNN64, for weight transfer, matrices
    \begin{subfigure}{\textwidth}
        \centering
        \scriptsize
        \import{figures/DNN_64/}{pred00250_chr21_030-040.pdf_tex}
        \caption{Example  region 1} \label{fig:results:DNN64_r1}
    \end{subfigure}\\[2mm]
    \centering{\scriptsize{see \cref{sec:methods:matrix_plots} for track explanations}}\\[3mm]
    \begin{subfigure}{\textwidth}
        \centering
        \scriptsize
        \import{figures/DNN_64/}{pred00250_chr19_030-040.pdf_tex}
        \caption{Example region 2} \label{fig:results:DNN64_r2}
    \end{subfigure}\\[2mm]
    \centering{\scriptsize{see \cref{sec:methods:matrix_plots} for track explanations}}\\[3mm]
    \begin{subfigure}{\textwidth}
        \centering
        \scriptsize
        \import{figures/DNN_64/}{pred00250_chr3_030-040.pdf_tex}
        \caption{Example region 3} \label{fig:results:DNN64_r3}
    \end{subfigure}
    \caption{Example predictions GM12878 $\rightarrow$ K562, DNN, $w=64$, 250~epochs} \label{fig:results:DNN_matrices}
\end{figure}
\clearpage
\subsubsection{cGAN trained on K562, predicting GM12878}\label{sec:appendix:reverseTrainTest}
\begin{figure}[h!] %cGAN 256 K-on-G, pearson and progress
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/GAN_256_K-on-G/}{pearson_chr03.pdf_tex}}
        \caption{chr3}
    \end{subfigure} \hfill
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/GAN_256_K-on-G/}{pearson_chr05.pdf_tex}}
        \caption{chr5}
    \end{subfigure}\\[5mm]
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/GAN_256_K-on-G/}{pearson_chr10.pdf_tex}}
        \caption{chr10}
    \end{subfigure}\hfill
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/GAN_256_K-on-G/}{pearson_chr19.pdf_tex}}
        \caption{chr19}
    \end{subfigure}\\[3mm]
    \centering
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/GAN_256_K-on-G/}{pearson_chr21.pdf_tex}}
        \caption{chr21}
    \end{subfigure} \hfill
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/GAN_256_K-on-G/}{lossOverEpochs.pdf_tex}}
        \caption{Learning progress} \label{fig:appendix:GAN256_K-on-G_lossEpochs}
    \end{subfigure}
    \caption{Results\,/\,metrics cGAN, CNN embedding, $w=256$, test chromosomes}   \label{fig:appendix:GAN256_K-on-G_pearson}
\end{figure}
\begin{figure}[p] %cgan mixed 64, pretrained, matrices
    \begin{subfigure}{\textwidth}
        \centering
        \resizebox{0.9\textwidth}{!}{
        \scriptsize
        \import{figures/GAN_256_K-on-G/}{pred00060_chr21_030-040.pdf_tex}}
        \caption{Example  region 1} \label{fig:appendix:GAN256_K-on-G_r1}
    \end{subfigure}\\[3mm]
    \begin{subfigure}{\textwidth}
        \centering
        \resizebox{0.9\textwidth}{!}{
        \scriptsize
        \import{figures/GAN_256_K-on-G/}{pred00060_chr19_030-040.pdf_tex}}
        \caption{Example region 2} \label{fig:appendix:GAN256_K-on-Gr2}
    \end{subfigure}\\[3mm]
    \caption{Example predictions K562 $\rightarrow$ GM12878, cGAN, CNN embedding, $w=256$, 60~epochs} 
\end{figure}
\begin{figure}\ContinuedFloat
    \begin{subfigure}{\textwidth}
        \centering
        \resizebox{0.9\textwidth}{!}{
        \scriptsize
        \import{figures/GAN_256_K-on-G/}{pred00060_chr3_030-040.pdf_tex}}
        \caption{Example region 3} \label{fig:appendix:GAN256_K-on-G_r3}
    \end{subfigure}
    \caption{Example predictions K562 $\rightarrow$ GM12878, cGAN, CNN embedding, $w=256$, 60~epochs} 
     \label{fig:appendix:GAN256_K-on-G_matrices}
\end{figure}
\clearpage
\subsubsection[cGAN trained on single chromosomes]{cGAN trained on single chromosomes predicting usual test chromosomes}
\begin{figure}[h!] %cGAN single chrom 14, pearson and progress, predicting usual test chroms 3,5,10,19,21
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/GAN_64_14only/}{pearson_chr03.pdf_tex}}
        \caption{chr3}
    \end{subfigure} \hfill
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/GAN_64_14only/}{pearson_chr05.pdf_tex}}
        \caption{chr5}
    \end{subfigure}\\[5mm]
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/GAN_64_14only/}{pearson_chr10.pdf_tex}}
        \caption{chr10}
    \end{subfigure}\hfill
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/GAN_64_14only/}{pearson_chr19.pdf_tex}}
        \caption{chr19}
    \end{subfigure}
    \caption{Results\,/\,metrics, cGAN, $w=64$, trained on GM12878 chr14 only; prediction on K562, typical test chromosomes}
    \end{figure}
    \begin{figure}\ContinuedFloat
     \centering
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/GAN_64_14only/}{pearson_chr21.pdf_tex}}
        \caption{chr21}
    \end{subfigure} \hfill
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/GAN_64_14only/}{lossOverEpochs.pdf_tex}}
        \caption{Learning progress} \label{fig:appendix:GAN64-14single_lossEpochs}
    \end{subfigure}
    \caption{Results\,/\,metrics, cGAN, $w=64$, trained on GM12878 chr14 only; prediction on K562, typical test chromosomes}   \label{fig:appendix:GAN64-14single_pearson}
\end{figure}
\begin{figure}[p] %cGAN single chrom 14, matrices
    \begin{subfigure}{\textwidth}
        \centering
        \resizebox{0.9\textwidth}{!}{
        \scriptsize
        \import{figures/GAN_64_14only/}{pred00100_chr21_030-040.pdf_tex}}
        \caption{Example  region 1} \label{fig:results:GAN64-14single_r1}
    \end{subfigure}\\[3mm]
    \begin{subfigure}{\textwidth}
        \centering
        \resizebox{0.9\textwidth}{!}{
        \scriptsize
        \import{figures/GAN_64_14only/}{pred00100_chr19_030-040.pdf_tex}}
        \caption{Example region 2} \label{fig:results:GAN64-14single_r2}
    \end{subfigure}\\[3mm]
    \begin{subfigure}{\textwidth}
        \centering
        \resizebox{0.9\textwidth}{!}{
        \scriptsize
        \import{figures/GAN_64_14only/}{pred00100_chr3_030-040.pdf_tex}}
        \caption{Example region 3} \label{fig:results:GAN64-14single_r3}
    \end{subfigure}
    \caption{Example predictions, cGAN, $w=64$, trained on GM12878 chr14 only; prediction on K562, 100~epochs} 
     \label{fig:results:GAN64-14single_matrices}
\end{figure}
\begin{figure}[p] %cGAN single chrom 17, pearson and progress, predicting usual test chroms 3,5,10,19,21
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/GAN_64_17only/}{pearson_chr03.pdf_tex}}
        \caption{chr3}
    \end{subfigure} \hfill
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/GAN_64_17only/}{pearson_chr05.pdf_tex}}
        \caption{chr5}
    \end{subfigure}\\[5mm]
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/GAN_64_17only/}{pearson_chr10.pdf_tex}}
        \caption{chr10}
    \end{subfigure}\hfill
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/GAN_64_17only/}{pearson_chr19.pdf_tex}}
        \caption{chr19}
    \end{subfigure}\\[3mm]
    \centering
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/GAN_64_17only/}{pearson_chr21.pdf_tex}}
        \caption{chr21}
    \end{subfigure} \hfill
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/GAN_64_17only/}{lossOverEpochs.pdf_tex}}
        \caption{learning progress} \label{fig:appendix:GAN64-17single_lossEpochs}
    \end{subfigure}
    \caption{Results\,/\,metrics cGAN, $w=64$, trained on GM12878 chr17 only; prediction on K562, typical test chromosomes}   \label{fig:appendix:GAN64-17single_pearson}
\end{figure}
\begin{figure}[p] %cGAN single chrom 14, matrices
    \begin{subfigure}{\textwidth}
        \centering
        \resizebox{0.9\textwidth}{!}{
        \scriptsize
        \import{figures/GAN_64_17only/}{pred00160_chr21_030-040.pdf_tex}}
        \caption{Example  region 1} \label{fig:results:GAN64-17single_r1}
    \end{subfigure}\\[3mm]
    \begin{subfigure}{\textwidth}
        \centering
        \resizebox{0.9\textwidth}{!}{
        \scriptsize
        \import{figures/GAN_64_17only/}{pred00160_chr19_030-040.pdf_tex}}
        \caption{Example region 2} \label{fig:results:GAN64-17single_r2}
    \end{subfigure}\\[3mm]
    \begin{subfigure}{\textwidth}
        \centering
        \resizebox{0.9\textwidth}{!}{
        \scriptsize
        \import{figures/GAN_64_17only/}{pred00160_chr3_030-040.pdf_tex}}
        \caption{Example region 3} \label{fig:results:GAN64-17single_r3}
    \end{subfigure}
    \caption{Example predictions cGAN, , $w=64$, trained on GM12878 chr17 only; prediction on K562, 160~epochs} 
     \label{fig:results:GAN64-17single_matrices}
\end{figure}
\clearpage

\subsubsection{Further results for D. Melanogaster} \label{sec:appendix:cgan_drosophila}
\begin{figure}[h!] %DNN trained on drosophila 2L, 2R 
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/DNN_drosophila2/}{pearson_chr3L.pdf_tex}}
        \caption{chr3L (train)}
    \end{subfigure} \hfill
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/DNN_drosophila2/}{pearson_chr3R.pdf_tex}}
        \caption{chr3R (test)}
    \end{subfigure}\\[5mm]
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/DNN_drosophila2/}{pearson_chrX.pdf_tex}}
        \caption{chrX (validation)}
    \end{subfigure}\hfill
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/DNN_drosophila2/}{lossOverEpochs.pdf_tex}}
        \caption{learning progress} \label{fig:appendix:DNN_drosophila_progress}
    \end{subfigure}
    \caption{Results\,/\,metrics DNN, $w=80$, trained on D. Melanogaster chr2L, 2R, 3L}   \label{fig:appendix:DNN_drosophila_pearson}
\end{figure}

\begin{figure}[p] %cGAN trained on drosophila 2L, 2R 
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/GAN_64_drosophila/}{pearson_chr3L.pdf_tex}}
        \caption{chr3L (validation)}
    \end{subfigure} \hfill
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/GAN_64_drosophila/}{pearson_chr3R.pdf_tex}}
        \caption{chr3R (test)}
    \end{subfigure}\\[5mm]
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/GAN_64_drosophila/}{pearson_chrX.pdf_tex}}
        \caption{chrX (test)}
    \end{subfigure}\hfill
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/GAN_64_drosophila/}{lossOverEpochs.pdf_tex}}
        \caption{learning progress} \label{fig:appendix:GAN64_drosophila_progress}
    \end{subfigure}
    \caption{Results\,/\,metrics cGAN, $w=64$, trained on D. Melanogaster chr2L and 2R}   \label{fig:appendix:GAN64_drosophila_pearson}
\end{figure}
\begin{figure}[htbp]
\begin{subfigure}{\textwidth}
 \centering
 \scriptsize
 \import{figures/GAN_64_drosophila/}{pred00100_chr3L_015-020.pdf_tex}
 \caption{chr3L (validation), 100 epochs}
\end{subfigure}\\[5mm]
\begin{subfigure}{\textwidth}
 \centering
 \scriptsize
 \import{figures/GAN_64_drosophila/}{pred00100_chrX_015-020.pdf_tex}
 \caption{chrX (test), 100 epochs}
\end{subfigure}
\caption{Further matrix plots, cGAN with DNN embedding, Drosophila Melanogaster} \label{fig:appendix:GAN64_drosophila_matrices}
\end{figure}

\begin{figure}[htbp]
\begin{subfigure}{\textwidth}
 \centering
 \scriptsize
 \import{figures/GAN_64_drosophila/}{pred00c_chr3R_1675-2175.pdf_tex}
 \caption{Matrix from Hi-cGAN, 100 epochs} \label{fig:appendix:farre-vs-ours_cgan-matrix}
\end{subfigure}\\[5mm]
\centering{\scriptsize{\cref{fig:appendix:farre-vs-ours_dnn-matrix}, \ref{fig:appendix:farre-vs-ours_cgan-matrix}: predicted matrices on top, true matrices from Schuettengruber et al. \cite{Schuettengruber2014} on bottom}}\\[2mm]
\begin{subfigure}{\textwidth}
 \centering
 \scriptsize
 \import{figures/DNN_drosophila2/}{pred00c_chr3R_1675-2175.pdf_tex}
 \caption{Matrix from DNN, 40 epochs} \label{fig:appendix:farre-vs-ours_dnn-matrix}
\end{subfigure}\\[3mm]
\begin{subfigure}{\textwidth}
 \centering
 \scriptsize
 \import{figures/FarreEtAl_drosophila/}{pred00c_chr3R_1675-2175.pdf_tex}
 \caption{Reproduced matrix from Farr\'e et al. \cite{Farre2018a}}, 70 epochs \label{fig:appendix:farre-vs-ours_farre-matrix_recreated}
\end{subfigure}\\[5mm]
\begin{subfigure}{\textwidth}
 \centering{\tiny{Image courtesy Farr\'e, Heurteau, Cuvier, Emberly \cite{Farre2018a}, licensed under Creative Commons Attribution 4.0 License \cite{ccby4}.\\
                                Approximated X-axis not part of original image. Scaled to same size as matrices above.\\
                                predicted matrix on top, target matrix on bottom}}
 \centering
 \scriptsize
 \import{figures/GAN_64_drosophila/}{predFarre_chr3R_168-218.pdf_tex}
 \caption{Original matrix extracted from Farr\'e et al. \cite{Farre2018a}} \label{fig:appendix:farre-vs-ours_farre-matrix_original}
\end{subfigure}\\[5mm]
\caption{Comparison between matrix plots from DNN / Hi-cGAN / Farr\'e et al. \cite{Farre2018a}\\using red/blue color maps} \label{fig:appendix:farre-vs-ours_matrices}
\end{figure}



\clearpage

