\section{Introduction}
In recent years, the three-dimensional organization of DNA has been shown to 
be a key factor for important processes in molecular biology.
However, even with the most advanced experimental methods, 
it remains comparatively expensive to determine the spatial folding of DNA directly,
so that current knowledge of three-dimensional DNA organization is still sketchy.
In the last five years, several methods have thus been proposed to improve on this situation
by determining DNA-DNA interactions \emph{in-silico}
All of these are using existing experimental data which is easier to obtain than spatial data, but correlates with spatial structure in certain ways.
However, most current in-silico approaches have disadvantages and shortcomings,
and the current thesis attempts to improve on these.

\subsection{Spatial structure of DNA}
In the late 1970s, Watson and Crick \xxx discovered the now well-known double helix structure of DNA molecules.
However, this is not the only relevant spatial structure of genomes. 
At larger scales, DNA molecules can be wound around certain proteins, so-called histones, forming DNA-protein complexes named nucleosomes.
Several of these can further be compacted into fibres, which in turn can be ``supercoiled'' into the also well-known chromosomes.
\xxx hier muss das bild hin \xxx

While the spatial structure of chromatin outlined above brings along compaction (fig. \xxx from left to right), 
e.\,g. to make chromosomes fit into eucaryotic cell nuclei,
it also has other functional implications.
One well known effect of spatial structure is the regulation of gene transcription by establishing or releasing contacts
between gene enhancers and the relevant promoters \cite{Smallwood2013,Gorkin2014}.
Since enhancers can be up to a million basepairs away from the promoters, 
the two can only interact by means of spatial DNA structure.
Many effects of spatial structure are still under investigation, 
two recent studies have for example found dependencies between chromatin conformation 
and cell differentiation in Drosophila Melanogaster \cite{Chathoth2019}
and investigated spatial structure dynamics during phase transitions in murine cells \cite{Zhang2019b}.

While the driving mechanisms behind the formation of spatial chromatin structures are partially still under research,
certain proteins like CTCF or modified histones are already well known to mediate or prevent DNA-DNA contacts \cite{Phillips2009,PhillipsCremins2013,Dixon2015}.

In the last two decades, DNA-sequencing-based techniques have increasingly been utilized to capture the spatial structure of DNA experimentally.
The method of choice within this thesis is called Hi-C and shall be explained in the following section.

\subsection{The Hi-C process for determining spatial DNA structure} \label{sec:intro:hic}
\begin{wrapfigure}[27]{r}{0.3\textwidth}
 \vspace{-19mm}
 \resizebox{0.3\textwidth}{!}{
 \small
 \import{figures/}{HiC-lab.pdf_tex}}
 \caption{Hi-C lab process}
 \label{fig:intro:HiC}
\end{wrapfigure}
The Hi-C process is an elaborate biochemical procedure for investigating the 
spatial structure of DNA by detecting DNA-DNA interactions within and 
across chromosomes.
The original Hi-C workflow has been developed by Lieberman-Aiden et al. in 2009  \cite{LiebermanAiden2009}
and is depicted in simplified form in \cref{fig:intro:HiC}.

The typical input (In) to Hi-C consists of several millions of cells,
which are treated chemically to fix existing DNA-DNA contacts, 
commoly using formaldehyde, before they are lysed.
Next, the DNA is extracted and cut into fragments by certain restriction enzymes (1),
usually HindIII or DpnII, 
and the cut ends are repaired with nucleotides, some of which are marked by biotin (2).
The free ends are then joined (3) under conditions which prefer
ligations among open ends over ligations between different fragments.
Originally, such conditions were achieved by high dilution of the fragments in
solvents, but especially this part of the protocol has been replaced by 
more efficient methods in later works \cite{Rao2014,Belaghzal2017}.
The ligated fragments are then purified and cut into shorter sequences,
some of which contain biotinylated nucleotides and some not (4).
The fragments of interest -- the ones containing biotinylated nucleotides -- 
are then selected by pulling down biotin (5), for example using magnetic tags,  
and subjected to paired-end DNA-sequencing (6).
In the end, the output of the Hi-C lab process is a large number of short genomic ``reads'',
which are subsequently processed in the bioinformatics part of the Hi-C protocol 
outlined in the following section.

On the software side of the protocol, the reads first need to be mapped 
to the corresponding reference genome.
Here, only reads are kept where the ``left'' sequence (6)(a)
uniquely maps to a different region of the reference genome than the ``right'' sequence (6)(b).
These so-called chimeric reads are subjected to quality control, and those passing are counted as an interaction
between the two genomic positions (1)(a) and (1)(b) to which the two ends belong.
However, at reasonable read coverages, interactions cannot be counted per base pair. 
Instead, the reference genome is split into equally sized bins (or regions), 
and the reads are counted for those regions where they belong. 
The final outcome of a Hi-C experiment is then a sparse matrix, henceforth referred to as ``Hi-C matrix'', 
which records the interaction count for all possible pairs of regions in the reference genome.

For the thesis at hand, Hi-C matrices can abstractly be considered as sets of sparse, square matrices; one matrix for each chromosome.
The individual elements $m_{i,j}$ of the matrices are counts of interactions between the bins with indices $i,j \in \{0,1,...,l\}$,
where the last index $l$ is given by the chromosome size $cs$ and the binsize $b$ in basepairs as $l=   \left \lceil{\frac{cs}{b}}\right \rceil -1$.
Because interaction counts do not have a ``direction'', it holds that $m_{i,j} = m_{j,i}$, so Hi-C matrices are always symmetric.

A bin with index $n$ then corresponds to the genomic region $p_n = [\,n\cdot b, \; (n+1)\cdot b\,)$, 
except the last bin $l$, where $p_{l} = [\,l \cdot b, \; cs \,]$, with $l,b,cs$ as above.
As an example, if $m_{0,10}=22$ for a matrix with binsize $b=\SI{25}{\kilo\bp}$, this means that in the underlying experiment, 
22 interactions have been counted between region $p_0$, ranging from 0 to \SI{24999}{\bp} and region $p_{10}$, 
ranging from 250000 to \SI{274999}{\bp}. 
Due to insufficient read coverage, $b=\SI{1}{\bp}$ is impossible in practice. 
Instead, common values include $b \in \{1, 5, 10, 25, 50, 100, 1000\}$\,kbp.

In the bioinformatics part of the Hi-C protocol, often just a small fraction of all reads 
fulfill the selection criteria outlined above, for example due to reads not being chimeric or uniquely mappable.
This makes Hi-C a comparatively inefficient, slow and thus expensive process.
For example, the well-known dataset by Rao et al. \cite{Rao2014} with matrix bin sizes down to \SI{1}{\kilo\bp}, required several \emph{billions} of reads being made. 

Parts of \cref{fig:intro:HiC} and the process description above 
have been adapted from the preceding study project \cite{Krauth2020}.



\xxx mod input steps in graph
\xxx mark regions a, b throughout the graph

\subsection{The ChIP-seq process for determining protein binding sites} \label{sec:intro:chipseq}
The ChIP-seq process is a combination of Chromatin-Immunoprecipitation (ChIP) and DNA-sequencing, 
designed for investigating DNA-protein interactions \cite{Johnson2007,Robertson2007}.
As with Hi-C, the input consists of a sufficient number of cells, which are first fixed with formaldehyde.
The cells are then lysed, the DNA-protein structure is extracted and cut into fragments, 
usually by sonication (2).
Next, specific antibodies are added, designed to bind only to a certain protein of interest (3).
These antibodies are additionally equipped with a tag, for example a magnetic one, so that 
the DNA-protein-antibody structures can be precipitated, while fragments without antibodies are discarded (4).
The proteins and antibodies are then removed (5), 
the DNA is purified and finally sequenced (6).
Typically, a control experiment is performed together with the ChIP-seq process, 
which comprises all steps described above, except the immunoprecipitation (3),(4).

The outcome of the ChIP-seq lab process is again a bunch of short genomic sequences, 
which are then fed into the bioinformatics part of the pipeline.

On the software side of the process, the reads are filtered for quality and mapped to an appropriate reference genome.
The number of mapped reads per genomic position can then be simply be counted.
It is common to process reads from the control experiment in the same way as reads from the ChIP-seq experiment
and then use special software to call ``peaks'', i.\,e. protein binding sites,
at those genomic positions where the read count from ChIP-seq
is statistically significant compared to the read count from the control group.

For the thesis at hand, the reads from the ChIP-seq experiment have been used directly
and binned by taking the mean over all reads inside each bin.
Contrary to Hi-C matrices, choosing a binsize of one is in principle possible for ChIP-seq experiments, but yields very sparse vectors.
Thus, binsizes that evenly divide the binsizes of the corresponding matrices have been chosen for this thesis, namely \SI{5}{\kilo\bp} and \SI{25}{\kilo\bp}.

\subsection{Motivation and goal of the thesis}
\begin{wraptable}{r}{5.5cm}
\vspace{-10mm}
\centering
\small
\begin{tabular}{lcccc}
cell line & \rotatebox[origin=l]{90}{TF ChIP-seq} & \rotatebox[origin=l]{90}{Histone ChIP-seq} & \rotatebox[origin=l]{90}{Dnase-seq} & \rotatebox[origin=l]{90}{Hi-C} \\ \hline
K562     & 680 & 20 & 10 & 1 \\
HepG2    & 668 & 15 & 3  & 1 \\
A549     & 251 & 87 & 14 & 6 \\
HEK293   & 231 & 6  &    &   \\
GM12878  & 211 & 15 & 3  & 7 \\
MCF-7    & 155 & 18 & 8  &   \\
H1       & 88  & 55 & 3  &   \\
HeLa-S3  & 77  & 15 & 4  & 1 \\
SK-N-SH  & 62  & 21 & 2  &   \\
IMR-90   & 16  & 34 & 2  & 2 \\
HCT116   & 27  & 17 & 1  & 8 \\
H9       &     & 35 & 7  &   \\
GM23338  & 15  & 13 & 1  &   \\
Ishikawa & 25  &    & 4  &   \\
HEK293T  & 17  &    & 1  &   \\
GM23248  & 2   & 13 & 1  & 1 \\ \hline
\end{tabular}
\caption{availability of selected assays in ENCODE (extract)} \label{tab:intro:assaysInEncode}
\vspace{-10mm}
\end{wraptable}
Due to the high effort, Hi-C experiments have been conducted only by few labs and for comparatively few organisms and cell lines so far.
For example, as of January 2021, Hi-C assays are available for just 26 of more than 150 human cell lines listed in the Encyclopedia of DNA elements \cite{Encode2012,Davis2017},
and these have been produced by only five university labs.

However, investigations and experiments with the available Hi-C data have shown correlations
between the spatial DNA structure and the binding sites of certain transcription factors and histone modifications \cite{Rao2014, Bonev2016}.
Since these binding sites can be detected with the less costly ChIP-seq method, 
a computational method able to predict contact matrices from DNA-protein bindings would be very helpful.

The goal of this master thesis is thus to predict missing Hi-C data from existing ChIP-seq data,
making use of the known correlations between the binding sites of transcription factors and histone modifications of the one hand 
and Hi-C interaction counts on the other hand. 
Because this is a wide field and lead time for a master thesis is limited, 
the present work will focus on machine learning techniques,
which have proven a good choice for exploiting complex patterns
and relationships in various kinds of data. \xxx citations needed
More specifically, the goal of this thesis is to develop an easy-to-use machine learning
approach for predicting Hi-C interaction matrices from ChIP-seq data, 
using standard in- and output formats with minimal pre- and postprocessing.

To underscore the usefulness of such an approach, \cref{tab:intro:assaysInEncode} to the side shows the first 16 human cell lines in ENCODE, 
sorted by total number of selected assays available.
It is obvious that even otherwise well-investigated cell lines like HEK293, MCF-7 and H1 lack Hi-C data, while ChIP-seq data is mostly available.
\xxx data for H1 is available in 


