\section{Results}
In the following two sections, 
the results of the DNN- and cGAN-based approaches for predicting Hi-C contact matrices will be presented.
While the various modifications of the dense neural network described in \cref{sec:improve:DNNapproach} 
did  not really help improve the status quo, the novel conditional generative adversarial network laid out in \cref{sec:improve:Hi-cGAN}
showed interesting properties.

\subsection{Dense Neural Network approaches} \label{sec:results:DNN}
In the following subsections, the resulting predictions for modifications of the dense neural network originally conceived 
by Farr\'e, Heurteau, Cuvier and Emberly \cite{Farre2018a} will be shown. 
This includes variations of the convolutional layer(s), custom loss functions with the intent of reducing blurriness in the predictions
and tuning window- and binsize, cf. \cref{sec:improve:DNNapproach}.
As a start, however, the results of the initial network without any modifications will be shown for comparison.

\subsubsection{Initial results for comparison} \label{sec:initialDNNresults}
The basic network was setup and trained as explained in \cref{sec:methods:basicSetup}.
Here, the validation error (MSE) reached its minimum of about \SI{150000}{} 
after approximately 500 epochs for \SI{25}{\kilo\bp} binsize and around 400 epochs for \SI{10}{\kilo\bp} binsize, \cref{fig:results:basicDNN_lossEpochs_25} and \ref{fig:results:basicDNN_lossEpochs_10}.
Beyond that, the learning curve indicated overfitting, but the resulting test matrices often still looked fairly similar, 
compare e.\,g. the matrix plots after 500 and 1000 epochs in figures \ref{fig:results:basic500} and \ref{fig:results:basic1000}.

\Cref{fig:results:basicDNN_pearson} and \cref{fig:results:basicDNN_10k_pearson} 
show the distance stratified Pearson correlations (cf. \cref{sec:methods:metrics}) 
alongside area under the correlation curve (AUC) for the five test chromosomes at binsizes 25 and \SI{10}{\kilo\bp}, respectively.
The red curves in each correlation plot show the correlation between the corresponding training chromosome from GM12878 and the target chromosome from K562.
It is obvious that all predicted test matrices have a strictly positive Pearson correlation with respect to the experimentally derived matrices by Rao et al. \cite{Rao2014},
but are not better than simply taking data from the training cell line as prediction for the target cell line.

The predicted matrices themselves looked modest when plotted with pygenometracks. 
While the DNN generally produced high interaction counts in regions with many true interactions
and low interaction counts in regions with few true interactions, the (TAD-)boundaries between different interacting domains 
were mostly not discernible, \cref{fig:results:basic500} and \ref{fig:results:basic1000}.
This finding is in line with the clearly positive, but medium-valued Pearson correlations.
Exceptions with more disctinct boundaries existed in all of the five test chromosomes,
for example chr19, 34 to \SI{35}{\kilo\bp} (\cref{fig:results:basic_r2}), but were rare. 
Interestingly, medium-sized interacting structures, for example chr21, 31 to \SI{32.5}{\kilo\bp} 
or between chr19, 31.2 to \SI{32.7}{\kilo\bp} often seemed to be missing altogether -- 
while structures larger than the windowsize, for example chr3, 34 and \SI{36.7}{\kilo\bp} and 36.7 and \SI{39.5}{\kilo\bp}
sometimes were at least indicated.

Reducing the binsize to $b_\mathit{feat}=b_\mathit{mat}=\SI{10}{\kilo\bp}$ as in the paper by Farr\'e et al. \cite{Farre2018a} 
led to somewhat different results.
The area under the correlation curves was approximately the same for test chromosomes 3 and 5, slightly better for chromosome 10, but worse for chromosome 19 and 21,
cf.~\cref{fig:results:basicDNN_pearson} and \ref{fig:results:basicDNN_10k_pearson}.
However, the ability to predict larger structures was lost, and the matrix plots thus did not look better than the ones for binsize \SI{25}{\kilo\bp}.
The comparatively bad result for test chromosome 21 might result from the low chromatin feature coverage of this particular chromosome.

No obvious correlation between comparatively ``good'' and ``bad'' predictions with open and closed states of the chromatin was observed.
However, formally computing such a correlation is challenging, because no adequate objective measure for ``good'' and ``bad'' is known, 
especially considering the rather blurry results obtained so far.
Furthermore, even if suchlike correlations existed, exploiting them for improving predictions would still be, at best, not straightforward.

\begin{figure}[p]
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/DNN_basic/}{pearson_chr03.pdf_tex}}
        \caption{chr3}
    \end{subfigure} \hfill
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/DNN_basic/}{pearson_chr05.pdf_tex}}
        \caption{chr5}
    \end{subfigure}\\[5mm]
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/DNN_basic/}{pearson_chr10.pdf_tex}}
        \caption{chr10}
    \end{subfigure}\hfill
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/DNN_basic/}{pearson_chr19.pdf_tex}}
        \caption{chr19}
    \end{subfigure}\\[3mm]
    \centering
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/DNN_basic/}{pearson_chr21.pdf_tex}}
        \caption{chr21}
    \end{subfigure}\hfill
    \begin{subfigure}{0.45\textwidth}
        \resizebox{\textwidth}{!}{
        \scriptsize
        \import{figures/DNN_basic/}{lossOverEpochs.pdf_tex}}
        \caption{learning progress}\label{fig:results:basicDNN_lossEpochs_25}
    \end{subfigure}
    \caption{results\,/\,metrics, basic DNN, \SI{25}{\kilo\bp}, test chromosomes}
    \label{fig:results:basicDNN_pearson}
\end{figure}

%10k Pearson and progress
\begin{figure}[p]
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/DNN_basic10k/}{pearson_chr03.pdf_tex}}
        \caption{chr3}
    \end{subfigure} \hfill
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/DNN_basic10k/}{pearson_chr05.pdf_tex}}
        \caption{chr5}
    \end{subfigure}\\[5mm]
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/DNN_basic10k/}{pearson_chr10.pdf_tex}}
        \caption{chr10}
    \end{subfigure}\hfill
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/DNN_basic10k/}{pearson_chr19.pdf_tex}}
        \caption{chr19}
    \end{subfigure}\\[3mm]
    \centering
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/DNN_basic10k/}{pearson_chr21.pdf_tex}}
        \caption{chr21}
    \end{subfigure}\hfill
     \begin{subfigure}{0.45\textwidth}
        \resizebox{\textwidth}{!}{
        \scriptsize
        \import{figures/DNN_basic10k/}{lossOverEpochs.pdf_tex}}
        \caption{learning progress}\label{fig:results:basicDNN_lossEpochs_10}
    \end{subfigure}
    \caption{results\,/\,metrics, basic DNN, \SI{10}{\kilo\bp}, test chromosomes}
    \label{fig:results:basicDNN_10k_pearson}
\end{figure}
%25k matrices, after 500 epochs
\begin{figure}[p]
    \begin{subfigure}{\textwidth}
        \centering
        \scriptsize
        \import{figures/DNN_basic/}{pred00500_chr21_030-040.pdf_tex}
        \caption{example  region 1, 500 epochs} \label{fig:results:basic_r1}
    \end{subfigure}\\[6mm]
    \begin{subfigure}{\textwidth}
        \centering
        \scriptsize
        \import{figures/DNN_basic/}{pred00500_chr19_030-040.pdf_tex}
        \caption{example region 2, 500 epochs} \label{fig:results:basic_r2}
    \end{subfigure}\\[6mm]
    \begin{subfigure}{\textwidth}
        \centering
        \scriptsize
        \import{figures/DNN_basic/}{pred00500_chr3_030-040.pdf_tex}
        \caption{example region 2, 500 epochs} \label{fig:results:basic_r3}
    \end{subfigure}
    \caption{example predictions, basic DNN, \SI{25}{\kilo\bp}, 500 epochs} \label{fig:results:basic500}
\end{figure}
%25k matrices, after 1000 epochs
\begin{figure}[p]
    \begin{subfigure}{\textwidth}
        \centering
        \scriptsize
        \import{figures/DNN_basic/}{pred01000_chr21_030-040.pdf_tex}
        \caption{example  region 1, 1000 epochs} \label{fig:results:basic_r1_1000}
    \end{subfigure}\\[6mm]
    \begin{subfigure}{\textwidth}
        \centering
        \scriptsize
        \import{figures/DNN_basic/}{pred01000_chr19_030-040.pdf_tex}
        \caption{example region 2, 1000 epochs} \label{fig:results:basic_r2_1000}
    \end{subfigure}\\[6mm]
    \begin{subfigure}{\textwidth}
        \centering
        \scriptsize
        \import{figures/DNN_basic/}{pred01000_chr3_030-040.pdf_tex}
        \caption{example region 2, 1000 epochs} \label{fig:results:basic_r3_1000}
    \end{subfigure}
    \caption{example predictions, basic DNN, \SI{25}{\kilo\bp}, 1000 epochs} \label{fig:results:basic1000}
\end{figure}
%10k matrices
\begin{figure}[p]
    \begin{subfigure}{\textwidth}
        \centering
        \scriptsize
        \import{figures/DNN_basic10k/}{pred01000_chr21_030-040.pdf_tex}
        \caption{example  region 1} \label{fig:results:basic10k_r1}
    \end{subfigure}\\[6mm]
    \begin{subfigure}{\textwidth}
        \centering
        \scriptsize
        \import{figures/DNN_basic10k/}{pred01000_chr19_030-040.pdf_tex}
        \caption{example region 2} \label{fig:results:basic10k_r2}
    \end{subfigure}\\[6mm]
    \begin{subfigure}{\textwidth}
        \centering
        \scriptsize
        \import{figures/DNN_basic10k/}{pred01000_chr3_030-040.pdf_tex}
        \caption{example region 3} \label{fig:results:basic10k_r3}
    \end{subfigure}
    \caption{example predictions, basic DNN \SI{10}{\kilo\bp}, 1000 epochs}
\end{figure}

\subsubsection{Results for variations of the convolutional part} \label{sec:results:wider-longer-etc}
The results for the ``wider'' network, which featured a wider convolutional filter in the first network layer, 
cf. \cref{sec:improve:convolution_extensions} and \ref{sec:methods:variants}, 
were generally similar to the initial results, both in terms of Pearson correlations and in terms of matrix plots, 
\cref{fig:results:widerDNN_pearson} and \ref{fig:results:wider_matrices}.
Given the small increase in the number of trainable parameters and overall similar network topology, this is not surprising.
Overfitting was less obvious than with the initial setup and the training process looked more smooth, 
but the remaining validation error was slightly higher than for the initial approach, \cref{fig:results:widerDNN_lossEpochs}.
\begin{figure}[p] %wider variant Pearson
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/DNN_wider/}{pearson_chr03.pdf_tex}}
        \caption{chr3}
    \end{subfigure} \hfill
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/DNN_wider/}{pearson_chr05.pdf_tex}}
        \caption{chr5}
    \end{subfigure}\\[5mm]
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/DNN_wider/}{pearson_chr10.pdf_tex}}
        \caption{chr10}
    \end{subfigure}\hfill
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/DNN_wider/}{pearson_chr19.pdf_tex}}
        \caption{chr19}
    \end{subfigure}\\[3mm]
    \centering
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/DNN_wider/}{pearson_chr21.pdf_tex}}
        \caption{chr21}
    \end{subfigure}\hfill
    \begin{subfigure}{0.45\textwidth}
        \resizebox{\textwidth}{!}{
        \scriptsize
        \import{figures/DNN_wider/}{lossOverEpochs.pdf_tex}}
        \caption{learning progress for wider DNN} \label{fig:results:widerDNN_lossEpochs}
    \end{subfigure}
    \caption{Pearson correlations, ``wider'' variant of DNN,  test chromosomes}
    \label{fig:results:widerDNN_pearson}
\end{figure}
%wider variant matrices
\begin{figure}[p]
    \begin{subfigure}{\textwidth}
        \centering
        \scriptsize
        \import{figures/DNN_wider/}{pred01000_chr21_030-040.pdf_tex}
        \caption{example  region 1} \label{fig:results:wider_r1}
    \end{subfigure}\\[6mm]
    \begin{subfigure}{\textwidth}
        \centering
        \scriptsize
        \import{figures/DNN_wider/}{pred01000_chr19_030-040.pdf_tex}
        \caption{example region 2} \label{fig:results:wider_r2}
    \end{subfigure}\\[6mm]
    \begin{subfigure}{\textwidth}
        \centering
        \scriptsize
        \import{figures/DNN_wider/}{pred01000_chr3_030-040.pdf_tex}
        \caption{example region 3} \label{fig:results:wider_r3}
    \end{subfigure}
    \caption{example predictions, ``wider'' variant of DNN \SI{25}{\kilo\bp}, 1000 epochs}\label{fig:results:wider_matrices}
\end{figure}

The predictions from the ``longer'' variant with three convolutinal filter layers instead of just one were better than the initial ones in terms of
Pearson correlations for test chromosomes 10, 19 and 21, but worse for test chromosomes 3 and 5, \cref{fig:results:longerDNN_pearson}.
Interestingly, correlations for some of the larger distances could not be computed after 250 and 500 epochs, 
which generally means that the same values were predicted for these distances, cf. \cref{sec:methods:metrics}.
The reason for this behavior is not fully understood yet, but comparatively few neurons in the outermost layer are responsible for predictions at 
longer distances due to the chosen network setup, cf. \cref{sec:methods:sample_gen}, \cref{fig:methods:prediction}.
Since the longer network variant has a considerably larger number of trainable parameters,
it is assumed that 500 epochs might not have been enough to activate some of the outer neurons.
Apart from that, the learning process for the ``longer'' variant in general looked more smooth and reached a lower validation error than before, \cref{fig:results:longerDNN_lossEpochs},
but the matrix plots did not show any obvious improvement over the initial ones, \cref{fig:results:longer_matrices}.
\begin{figure}[p] %longer variant pearson
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/DNN_longer/}{pearson_chr03.pdf_tex}}
        \caption{chr3}
    \end{subfigure} \hfill
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/DNN_longer/}{pearson_chr05.pdf_tex}}
        \caption{chr5}
    \end{subfigure}\\[5mm]
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/DNN_longer/}{pearson_chr10.pdf_tex}}
        \caption{chr10}
    \end{subfigure}\hfill
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/DNN_longer/}{pearson_chr19.pdf_tex}}
        \caption{chr19}
    \end{subfigure}\\[3mm]
    \centering
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/DNN_longer/}{pearson_chr21.pdf_tex}}
        \caption{chr21}
    \end{subfigure}\hfill
    \begin{subfigure}{0.45\textwidth}
        \resizebox{\textwidth}{!}{
        \scriptsize
        \import{figures/DNN_longer/}{lossOverEpochs.pdf_tex}}
        \caption{learning progress for longer DNN} \label{fig:results:longerDNN_lossEpochs}
    \end{subfigure}
    \caption{results\,/\,metrics, ``longer'' variant of DNN,  test chromosomes}
    \label{fig:results:longerDNN_pearson}
\end{figure}
%longer variant matrices
\begin{figure}[p]
    \begin{subfigure}{\textwidth}
        \centering
        \scriptsize
        \import{figures/DNN_longer/}{pred01000_chr21_030-040.pdf_tex}
        \caption{example  region 1} \label{fig:results:longer_r1}
    \end{subfigure}\\[6mm]
    \begin{subfigure}{\textwidth}
        \centering
        \scriptsize
        \import{figures/DNN_longer/}{pred01000_chr19_030-040.pdf_tex}
        \caption{example region 2} \label{fig:results:longer_r2}
    \end{subfigure}\\[6mm]
    \begin{subfigure}{\textwidth}
        \centering
        \scriptsize
        \import{figures/DNN_longer/}{pred01000_chr3_030-040.pdf_tex}
        \caption{example region 3} \label{fig:results:longer_r3}
    \end{subfigure}
    \caption{example predictions, ``longer'' variant of DNN \SI{25}{\kilo\bp}, 1000 epochs} \label{fig:results:longer_matrices}
\end{figure}

Combining the ``longer'' and ``wider'' variants in the ``wider-longer'' setup with more convolutional layers and wider 
filter kernels also did not perform as expected.
While improvements in the Pearson correlations could again be seen for 3 of 5 test chromosomes compared to the initial network, \cref{fig:results:wider-longerDNN_pearson},
the observed correlations were worse than the ones from the highly similar ``longer''-variant alone. 
Like with the similar ``longer''-approach predictions at longer distances were partially missing. 
Compared to the other variants, the validation error was generally higher and stopped decreasing after very few epochs, \cref{fig:results:wider-longerDNN_lossEpochs}.
In terms of matrix plots, the predictions surprisingly were still quite similar to the initial ones, but seemed a bit more blurry, \ref{fig:results:wider-longer_matrices}.
\begin{figure}[p]%wider-longer Pearson
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/DNN_wider-longer/}{pearson_chr03.pdf_tex}}
        \caption{chr3}
    \end{subfigure} \hfill
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/DNN_wider-longer/}{pearson_chr05.pdf_tex}}
        \caption{chr5}
    \end{subfigure}\\[5mm]
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/DNN_wider-longer/}{pearson_chr10.pdf_tex}}
        \caption{chr10}
    \end{subfigure}\hfill
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/DNN_wider-longer/}{pearson_chr19.pdf_tex}}
        \caption{chr19}
    \end{subfigure}\\[3mm]
    \centering
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/DNN_wider-longer/}{pearson_chr21.pdf_tex}}
        \caption{chr21}
    \end{subfigure}\hfill
    \begin{subfigure}{0.45\textwidth}
        \resizebox{\textwidth}{!}{
        \scriptsize
        \import{figures/DNN_wider-longer/}{lossOverEpochs.pdf_tex}}
        \caption{learning progress} \label{fig:results:wider-longerDNN_lossEpochs}
    \end{subfigure}
    \caption{results\,/\,metrics, ``wider-longer'' variant of DNN,  test chromosomes}
    \label{fig:results:wider-longerDNN_pearson}
\end{figure}
%wider-longer variant matrices
\begin{figure}[p]
    \begin{subfigure}{\textwidth}
        \centering
        \scriptsize
        \import{figures/DNN_wider-longer/}{pred01000_chr21_030-040.pdf_tex}
        \caption{example  region 1} \label{fig:results:wider-longer_r1}
    \end{subfigure}\\[6mm]
    \begin{subfigure}{\textwidth}
        \centering
        \scriptsize
        \import{figures/DNN_wider-longer/}{pred01000_chr19_030-040.pdf_tex}
        \caption{example region 2} \label{fig:results:wider-longer_r2}
    \end{subfigure}\\[6mm]
    \begin{subfigure}{\textwidth}
        \centering
        \scriptsize
        \import{figures/DNN_wider-longer/}{pred01000_chr3_030-040.pdf_tex}
        \caption{example region 3} \label{fig:results:wider-longer_r3}
    \end{subfigure}
    \caption{example predictions, ``wider-longer'' variant of DNN \SI{25}{\kilo\bp}, 1000 epochs} \label{fig:results:wider-longer_matrices}
\end{figure}

Predictions and metrics from the generalized DNN approach with feature binsize \SI{5}{\kilo\bp} and matrix binsize \SI{25}{\kilo\bp}
are shown in \cref{fig:results:25k5DNN_pearson} and \ref{fig:results:25k5_matrices}.
Unfortunately, the results did again not improve compared to the initial predictions.
While the learning curve was smooth and showed signs of slight overfitting beyond 300 epochs, \cref{fig:results:25k5DNN_lossEpochs},
the matrix plots seemed worse than the initial ones. 
For example, the large structure at chromosome 3, 34 to \SI{36.7}{\kilo\bp}, which had been detected by the previous approaches, was now completely missing.
\begin{figure}[p]%25k5 Pearson
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/DNN_25k5/}{pearson_chr03.pdf_tex}}
        \caption{chr3}
    \end{subfigure} \hfill
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/DNN_25k5/}{pearson_chr05.pdf_tex}}
        \caption{chr5}
    \end{subfigure}\\[5mm]
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/DNN_25k5/}{pearson_chr10.pdf_tex}}
        \caption{chr10}
    \end{subfigure}\hfill
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/DNN_25k5/}{pearson_chr19.pdf_tex}}
        \caption{chr19}
    \end{subfigure}\\[3mm]
    \centering
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/DNN_25k5/}{pearson_chr21.pdf_tex}}
        \caption{chr21}
    \end{subfigure}\hfill
    \begin{subfigure}{0.45\textwidth}
        \resizebox{\textwidth}{!}{
        \scriptsize
        \import{figures/DNN_25k5/}{lossOverEpochs.pdf_tex}}
        \caption{learning progress} \label{fig:results:25k5DNN_lossEpochs}
    \end{subfigure}
    \caption{results\,/\,metrics, ``5k -- 25k'' variant of DNN with $b_\mathit{feat}=\SI{5}{\kilo\bp}$ and $b_\mathit{mat}=\SI{25}{\kilo\bp}$,  test chromosomes}
    \label{fig:results:25k5DNN_pearson}
\end{figure}
%25k5 matrices
\begin{figure}[p]
    \begin{subfigure}{\textwidth}
        \centering
        \scriptsize
        \import{figures/DNN_25k5/}{pred01000_chr21_030-040.pdf_tex}
        \caption{example  region 1} \label{fig:results:25k5_r1}
    \end{subfigure}\\[6mm]
    \begin{subfigure}{\textwidth}
        \centering
        \scriptsize
        \import{figures/DNN_25k5/}{pred01000_chr19_030-040.pdf_tex}
        \caption{example region 2} \label{fig:results:25k5_r2}
    \end{subfigure}\\[6mm]
    \begin{subfigure}{\textwidth}
        \centering
        \scriptsize
        \import{figures/DNN_25k5/}{pred01000_chr3_030-040.pdf_tex}
        \caption{example region 3} \label{fig:results:25k5_r3}
    \end{subfigure}
    \caption{example predictions, ``5k -- 25k'' variant of DNN, 1000 epochs} \label{fig:results:25k5_matrices}
\end{figure}

\subsubsection{Results for combined loss function} \label{sec:results:loss_functions}
Exchanging the mean squared error for a combined loss function consinsting of MSE, TV loss and perceptual loss did not improve the results in the given setting.
The results for optimizing a loss function according to \cref{eq:methods:combined_loss} with weighting parameters 
are shown in \cref{fig:results:combilossDNN_pearson} and \ref{fig:results:combiloss_matrices}.

For all test chromosomes, the correlations were highly similar to the inital network's, \cref{fig:results:combilossDNN_pearson},
and the matrix plots also looked similar, chromosome 21 probably being the most different, \cref{fig:results:combiloss_matrices}.
The results plotted are the best ones obtained by manual tuning of the multiplicative parameters $\lambda$. 
Guided parameter tuning was unfortunately infeasible within the thesis at hand due to the training times required for optimizing the combined loss function.
Other options which where not explored for the same reason include truncating the VGG-16 network at a different layer, using a loss function based on
more than just one of the intermediate VGG-16 layers \cite{Johnson2016} or taking another loss network.
However, the results obtained thus far were also not encouraging towards such investigations.
\begin{figure}[p] %combiloss pearson and progress
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/DNN_combiloss/}{pearson_chr03.pdf_tex}}
        \caption{chr3}
    \end{subfigure} \hfill
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/DNN_combiloss/}{pearson_chr05.pdf_tex}}
        \caption{chr5}
    \end{subfigure}\\[5mm]
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/DNN_combiloss/}{pearson_chr10.pdf_tex}}
        \caption{chr10}
    \end{subfigure}\hfill
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/DNN_combiloss/}{pearson_chr19.pdf_tex}}
        \caption{chr19}
    \end{subfigure}\\[3mm]
    \centering
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/DNN_combiloss/}{pearson_chr21.pdf_tex}}
        \caption{chr21}
    \end{subfigure}\hfill
    \begin{subfigure}{0.45\textwidth}
        \resizebox{\textwidth}{!}{
        \scriptsize
        \import{figures/DNN_combiloss/}{lossOverEpochs.pdf_tex}}
        \caption{learning progress} \label{fig:results:combilossDNN_lossEpochs}
    \end{subfigure}
    \caption{results\,/\,metrics, DNN with combined loss function (MSE, TV, VGG-16),  test chromosomes}
    \label{fig:results:combilossDNN_pearson}
\end{figure}
%combiloss matrices
\begin{figure}[p]
    \begin{subfigure}{\textwidth}
        \centering
        \scriptsize
        \import{figures/DNN_combiloss/}{pred00500_chr21_030-040.pdf_tex}
        \caption{example  region 1} \label{fig:results:combiloss_r1}
    \end{subfigure}\\[6mm]
    \begin{subfigure}{\textwidth}
        \centering
        \scriptsize
        \import{figures/DNN_combiloss/}{pred00500_chr19_030-040.pdf_tex}
        \caption{example region 2} \label{fig:results:combiloss_r2}
    \end{subfigure}\\[6mm]
    \begin{subfigure}{\textwidth}
        \centering
        \scriptsize
        \import{figures/DNN_combiloss/}{pred00500_chr3_030-040.pdf_tex}
        \caption{example region 3} \label{fig:results:combiloss_r3}
    \end{subfigure}
    \caption{example predictions, DNN with combined loss function (MSE, TV, VGG-16), 500 epochs} \label{fig:results:combiloss_matrices}
\end{figure}
While manually searching better parameters $\lambda$ was not successful,
it was found that the TV loss weight $\lambda_\mathit{TV}$ needed to be much smaller than the two other weights.
Otherwise, many true interactions off the matrix diagonals were considered as noise and optimized away early in the training process, cf.
\cref{fig:appendix:failed_tv_loss} (p. \pageref{fig:appendix:failed_tv_loss}).

\subsubsection{Results for score-based loss function} \label{sec:results:scorebased}
Exchanging the MSE- loss function by a combination between score-based and MSE loss
allowed for a smooth learning process and a slightly lower validation error compared to the initial approach.
However, at around \SI{7}{\min} per epoch on a GPU, the training process also was about seven times slower than the initial approach on CPU.
Unfortunately, the higher effort did not lead to obvious improvements.

The Pearson correlations for a score-based loss function with parameters $\lambda_\mathit{MSE}=1.0,\; \lambda_\mathit{score}=100,\; ds=12$ 
are shown in figure \ref{fig:results:scoreLossDNN_pearson}.
While a slight improvement was achieved for test chromosome 21, the correlations of the others remained widely unchanged.
The matrix plots also looked fairly similar to the initial ones, \cref{fig:results:scoreloss_matrices}, chromosome 21 again being the 
most different compared to the intial predictions.

In the matrix plots, the true- and predicted scores have been added as a second track, replacing the PCA track. 
Indeed, the score curve computed from the true matrices showed local minima at putative TAD boundaries, as set forth in \ref{sec:improve:TAD_loss},
so score computation with the chosen diamondsize seemed sound.
However, despite the optimization term in the loss function, the score curve of the predicted matrices compared to the true curve 
somewhat like the predicted matrices compared to the true ones:
The predicted score was generally high, when the true score was high, and low when the true score was also low,
but high peaks (local maxima) and steep valleys (local minima) in the plots were usually averaged out.

Long training times forbade a targeted parameter tuning by grid- or tree-search,
so the results presented in this section should not be interpreted as the optimal ones achievable by a score-based loss function.
\begin{figure}[p]%score loss pearson and progress
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/DNN_scoreLoss/}{pearson_chr03.pdf_tex}}
        \caption{chr3}
    \end{subfigure} \hfill
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/DNN_scoreLoss/}{pearson_chr05.pdf_tex}}
        \caption{chr5}
    \end{subfigure}\\[5mm]
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/DNN_scoreLoss/}{pearson_chr10.pdf_tex}}
        \caption{chr10}
    \end{subfigure}\hfill
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/DNN_scoreLoss/}{pearson_chr19.pdf_tex}}
        \caption{chr19}
    \end{subfigure}\\[3mm]
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/DNN_scoreLoss/}{pearson_chr21.pdf_tex}}
        \caption{chr21}
    \end{subfigure}\hfill
    \begin{subfigure}{0.45\textwidth}
        \resizebox{\textwidth}{!}{
        \scriptsize
        \import{figures/DNN_scoreLoss/}{lossOverEpochs.pdf_tex}}
        \caption{learning progress} \label{fig:results:scoreLossDNN_lossEpochs}
    \end{subfigure}
    \caption{results\,/\,metrics, DNN with score-based loss function, test chromosomes\\ ($\lambda_\mathit{MSE}=1.0,\; \lambda_\mathit{score}=100,\; ds=12$)} \label{fig:results:scoreLossDNN_pearson}
\end{figure}
\begin{figure}[p] %score loss matrices
    \begin{subfigure}{\textwidth}
        \centering
        \scriptsize
        \import{figures/DNN_scoreLoss/}{pred00500_chr21_030-040.pdf_tex}
        \caption{example  region 1} \label{fig:results:scoreloss_r1}
    \end{subfigure}\\[6mm]
    \begin{subfigure}{\textwidth}
        \centering
        \scriptsize
        \import{figures/DNN_scoreLoss/}{pred00500_chr19_030-040.pdf_tex}
        \caption{example region 2} \label{fig:results:scoreloss_r2}
    \end{subfigure}\\[6mm]
    \begin{subfigure}{\textwidth}
        \centering
        \scriptsize
        \import{figures/DNN_scoreLoss/}{pred00500_chr3_030-040.pdf_tex}
        \caption{example region 3} \label{fig:results:scoreloss_r3}
    \end{subfigure}
    \caption{example predictions,  DNN with score-based loss function, 500 epochs} \label{fig:results:scoreloss_matrices}
\end{figure}

\subsubsection{Results for different binsizes and windowsizes} \label{sec:results:binsize_winsize}
To assess predictions at larger binsizes, five different approaches were compared, cf. \cref{sec:improve:binsize_winsize}:
\begin{enumerate}
 \item ``50k direct'': \\directly training a network at binsize \SI{50}{\kilo\bp} and predicting at that same binsize
 \item ``initial 25k coarsened'': \\coarsening the results of the initial network discussed above in \cref{sec:initialDNNresults}
                    by summarizing bins via \texttt{cooler coarsen}, cf. \cref{sec:methods:hicMatrices}
 \item ``initial 25k$\rightarrow$50k'': \\using the initial network trained at \SI{25}{\kilo\bp} (cf. \cref{sec:initialDNNresults}) to predict at \SI{50}{\kilo\bp}
 \item ``25k+50k$\rightarrow$50k'': \\predicting at \SI{50}{\kilo\bp} from a network \emph{simultaneously} trained with binsizes 25k and 50k
 \item ``25k+50k$\rightarrow$25k'': \\predicting at \SI{25}{\kilo\bp} from a network \emph{simultaneously} trained with binsizes 25k and 50k
\end{enumerate}

For all test chromosomes except 21, the best Pearson correlations were obtained either by coarsening the initial results to \SI{50}{\kilo\bp} 
(method b) or by taking the network trained at \SI{25}{\kilo\bp} for predicting at \SI{50}{\kilo\bp} (method c), \cref{fig:results:DNN50k_pearson}.
Compared to coarsening, the latter approach had the advantage of doubling the windowsize (in basepairs) and it worked better for test chromosome 21.

Looking at the matrix plots, the desired effect of making larger structures more prominent by increasing the binsize was only partially achieved, \cref{fig:results:50k_from25k_matrices}.
While all larger structures in the example cutout of test chromosome 3 indeed looked more prominent,
no obvious improvement was observed for the medium-sized structures in the example regions of chromosome 19 and 21.
Here, too, the predicted matrices from the network trained at binsize \SI{25}{\kilo\bp} seemed better than the direct predictions 
at \SI{50}{\kilo\bp} (method a), \cref{fig:results:50k_matrices} and \ref{fig:results:50k_from25k_matrices}.

Notably, the training process for the direct prediction at binsize \SI{50}{\kilo\bp} (method a) collapsed after about 420 epochs.
One possible reason for this could be too high a learning rate, which could have been avoided by decreasing the learning rate over time.
However, no further investigations were made into the case, because the divergence occurred after the optimum validation error 
had been reached between 150 and 250 epochs, \cref{fig:results:50k_lossEpochs}, and was thus not seen as too problematic here.
Faster convergence in itself would not be surprising, since there are only about half as many training samples at \SI{50}{\kilo\bp} compared to \SI{25}{\kilo\bp}, 
cf. \cref{tab:methods:samples} (p. \pageref{tab:methods:samples}).

\begin{figure}[p]%50k direct AND from 25k, pearson and progress
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/DNN_50k/}{pearson_chr03.pdf_tex}}
        \caption{chr3}
    \end{subfigure} \hfill
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/DNN_50k/}{pearson_chr05.pdf_tex}}
        \caption{chr5}
    \end{subfigure}\\[5mm]
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/DNN_50k/}{pearson_chr10.pdf_tex}}
        \caption{chr10}
    \end{subfigure}\hfill
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/DNN_50k/}{pearson_chr19.pdf_tex}}
        \caption{chr19}
    \end{subfigure}\\[3mm]
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/DNN_50k/}{pearson_chr21.pdf_tex}}
        \caption{chr21}
    \end{subfigure}\hfill
    \begin{subfigure}{0.45\textwidth}
        \resizebox{\textwidth}{!}{
        \scriptsize
        \import{figures/DNN_50k/}{lossOverEpochs.pdf_tex}}
        \caption{learning progress 50k direct} \label{fig:results:50k_lossEpochs}
    \end{subfigure}
    \caption{results\,/\,metrics, various DNNs at \SI{50}{\kilo\bp}} \label{fig:results:DNN50k_pearson}
\end{figure}
\begin{figure}[p] %50k direct, matrices
    \begin{subfigure}{\textwidth}
        \centering
        \resizebox{0.77\textwidth}{!}{
        \scriptsize
        \import{figures/DNN_50k/}{pred00250_chr21_030-040.pdf_tex}}
        \caption{example  region 1} \label{fig:results:50k_r1}
    \end{subfigure}\\[3mm]
    \begin{subfigure}{\textwidth}
        \centering
        \resizebox{0.77\textwidth}{!}{
        \scriptsize
        \import{figures/DNN_50k/}{pred00250_chr19_030-040.pdf_tex}}
        \caption{example region 2} \label{fig:results:50k_r2}
    \end{subfigure}\\[3mm]
    \begin{subfigure}{\textwidth}
        \centering
        \resizebox{0.77\textwidth}{!}{
        \scriptsize
        \import{figures/DNN_50k/}{pred00250_chr3_030-040.pdf_tex}}
        \caption{example region 3} \label{fig:results:50k_r3}
    \end{subfigure}
    \caption{example predictions,  DNN at \SI{50}{\kilo\bp} direct, 250 epochs} \label{fig:results:50k_matrices}
\end{figure}
\begin{figure}[p] %50k from 25k, matrices
    \begin{subfigure}{\textwidth}
        \centering
        \resizebox{0.77\textwidth}{!}{
        \scriptsize
        \import{figures/DNN_50k/}{pred00500_50k_chr21_030-040.pdf_tex}}
        \caption{example  region 1} \label{fig:results:50k_from25k_r1}
    \end{subfigure}\\[3mm]
    \begin{subfigure}{\textwidth}
        \centering
        \resizebox{0.77\textwidth}{!}{
        \scriptsize
        \import{figures/DNN_50k/}{pred00500_50k_chr19_030-040.pdf_tex}}
        \caption{example region 2} \label{fig:results:50k_from25k_r2}
    \end{subfigure}\\[3mm]
    \begin{subfigure}{\textwidth}
        \centering
        \resizebox{0.77\textwidth}{!}{
        \scriptsize
        \import{figures/DNN_50k/}{pred00500_50k_chr3_030-040.pdf_tex}}
        \caption{example region 3} \label{fig:results:50k_from25k_r3}
    \end{subfigure}
    \caption{example predictions,  DNN trained at \SI{25}{\kilo\bp} predicting at \SI{50}{\kilo\bp}, 500 epochs} \label{fig:results:50k_from25k_matrices}
\end{figure}

Simultaneously training a network with matrix- and feature binsizes of \SI{25}{\kilo\bp} and \SI{50}{\kilo\bp} (methods d, e)
turned out unproblematic with regard to convergence, \cref{fig:results:25plus50_lossEpochs}, 
but the Pearson correlations when predicting at both \SI{25}{\kilo\bp} and \SI{50}{\kilo\bp} were -- often significantly -- worse
than the initial predictions at that binsize, \cref{fig:results:DNN50k_pearson} (``25k+50k$\rightarrow$50k'') and \cref{fig:results:DNN25plus50_pearson} (``25k+50k$\rightarrow$25k'').
Looking into the matrix plots, it could not be clarified what caused the improvement in the Pearson correlations for test chromosome 21, \cref{fig:results:25plus50_matrices}.
Here, all predictions seemed equally useless and definitely worse than the results obtained by the other approaches investigated thus far.

\xxx get this section in shape
Another finding relates to the training progress curve -- such smooth courses of the training loss usually occur together with gradient-style predictions
as observed in test chromosome 19. 
These gradients seem to be a local minimum with regard to mean squared error in which the optimizer may end up under certain circumstances.
However, these types of predictions generally cause Pearson correlations close to zero, and it remained unclear why this did not happen here -- the other
parts of chromosome 19 did not look much better than the \SI{10}{\mega\bp}-cutout shown in \cref{fig:results:25plus50_r2}.
\xxx

\begin{figure}[p]%trained at 25k and 50k simultaneously, pearson and progress for 25k
    \begin{subfigure}{0.45\textwidth}
        \scriptsize 
        \resizebox{\textwidth}{!}{
        \import{figures/DNN_25plus50/}{pearson_chr03.pdf_tex}}
        \caption{chr3}
    \end{subfigure} \hfill
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/DNN_25plus50/}{pearson_chr05.pdf_tex}}
        \caption{chr5}
    \end{subfigure}\\[5mm]
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/DNN_25plus50/}{pearson_chr10.pdf_tex}}
        \caption{chr10}
    \end{subfigure}\hfill
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/DNN_25plus50/}{pearson_chr19.pdf_tex}}
        \caption{chr19}
    \end{subfigure}\\[3mm]
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/DNN_25plus50/}{pearson_chr21.pdf_tex}}
        \caption{chr21}
    \end{subfigure}\hfill
    \begin{subfigure}{0.45\textwidth}
        \resizebox{\textwidth}{!}{
        \scriptsize
        \import{figures/DNN_25plus50/}{lossOverEpochs.pdf_tex}}
        \caption{learning progress} \label{fig:results:25plus50_lossEpochs}
    \end{subfigure}
    \caption{results\,/\,metrics, DNN trained at \SI{25}{\kilo\bp} and \SI{50}{\kilo\bp} simultaneously} \label{fig:results:DNN25plus50_pearson}
\end{figure}
\begin{figure}[p] %25plus50, matrices at 25k
    \begin{subfigure}{\textwidth}
        \centering
        \scriptsize
        \import{figures/DNN_25plus50/}{pred00500_chr21_030-040.pdf_tex}
        \caption{example  region 1} \label{fig:results:25plus50_r1}
    \end{subfigure}\\[6mm]
    \begin{subfigure}{\textwidth}
        \centering
        \scriptsize
        \import{figures/DNN_25plus50/}{pred00500_chr19_030-040.pdf_tex}
        \caption{example region 2} \label{fig:results:25plus50_r2}
    \end{subfigure}\\[6mm]
    \begin{subfigure}{\textwidth}
        \centering
        \scriptsize
        \import{figures/DNN_25plus50/}{pred00500_chr3_030-040.pdf_tex}
        \caption{example region 3} \label{fig:results:25plus50_r3}
    \end{subfigure}
    \caption{example predictions,  DNN trained at \SI{25}{\kilo\bp} and \SI{50}{\kilo\bp} simultaneously, \SI{25}{\kilo\bp}, 500 epochs}\label{fig:results:25plus50_matrices}
\end{figure}

\clearpage
\subsection{Hi-cGAN approaches} \label{sec:results:cgan}
\subsubsection{cGAN with DNN embedding} \label{sec:results:cgan_dnn}
Results after pretraining see figures \ref{fig:results:GAN64_pretrain-dnn_pearson} and \ref{fig:results:cGAN64_pretrain-dnn_matrices}.
Results without pretraining see figures \ref{fig:results:GAN64-dnn_pearson} and \ref{fig:results:cGAN64-dnn_matrices}.
\begin{figure}[p] %cGAN with DNN, no pretraining, windowsize 64, pearson and progress
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/GAN_64_dnn/}{pearson_chr03.pdf_tex}}
        \caption{chr3}
    \end{subfigure} \hfill
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/GAN_64_dnn/}{pearson_chr05.pdf_tex}}
        \caption{chr5}
    \end{subfigure}\\[5mm]
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/GAN_64_dnn/}{pearson_chr10.pdf_tex}}
        \caption{chr10}
    \end{subfigure}\hfill
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/GAN_64_dnn/}{pearson_chr19.pdf_tex}}
        \caption{chr19}
    \end{subfigure}\\[3mm]
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/GAN_64_dnn/}{pearson_chr21.pdf_tex}}
        \caption{chr21}
    \end{subfigure} \hfill
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/GAN_64_dnn/}{lossOverEpochs.pdf_tex}}
        \caption{learning progress} \label{fig:results:GAN64-dnn_lossEpochs}
    \end{subfigure}
    \caption{results\,/\,metrics cGAN, DNN embedding, no pre-training, $w=64$, test chromosomes}   \label{fig:results:GAN64-dnn_pearson}
\end{figure}
\begin{figure}[p] %cgan with DNN, no pre-training, winsize 64, matrices
    \begin{subfigure}{\textwidth}
        \centering
        \scriptsize
        \import{figures/GAN_64_dnn/}{pred00080_chr21_030-040.pdf_tex}
        \caption{example  region 1} \label{fig:results:cGAN64-dnn_r1}
    \end{subfigure}\\[6mm]
    \begin{subfigure}{\textwidth}
        \centering
        \scriptsize
        \import{figures/GAN_64_dnn/}{pred00080_chr19_030-040.pdf_tex}
        \caption{example region 2} \label{fig:results:cGAN64-dnn_r2}
    \end{subfigure}\\[6mm]
    \begin{subfigure}{\textwidth}
        \centering
        \scriptsize
        \import{figures/GAN_64_dnn/}{pred00080_chr3_030-040.pdf_tex}
        \caption{example region 3} \label{fig:results:cGAN64-dnn_r3}
    \end{subfigure}
    \caption{example predictions, cGAN, DNN embedding, no pre-training, $w=64$, 80 epochs} \label{fig:results:cGAN64-dnn_matrices}
\end{figure}
\begin{figure}[p] %cGAN with DNN, pretrained, windowsize 64, pearson and progress
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/GAN_64_pretrain-dnn/}{pearson_chr03.pdf_tex}}
        \caption{chr3}
    \end{subfigure} \hfill
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/GAN_64_pretrain-dnn/}{pearson_chr05.pdf_tex}}
        \caption{chr5}
    \end{subfigure}\\[5mm]
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/GAN_64_pretrain-dnn/}{pearson_chr10.pdf_tex}}
        \caption{chr10}
    \end{subfigure}\hfill
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/GAN_64_pretrain-dnn/}{pearson_chr19.pdf_tex}}
        \caption{chr19}
    \end{subfigure}\\[3mm]
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/GAN_64_pretrain-dnn/}{pearson_chr21.pdf_tex}}
        \caption{chr21}
    \end{subfigure} \hfill
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/GAN_64_pretrain-dnn/}{lossOverEpochs.pdf_tex}}
        \caption{learning progress} \label{fig:results:GAN64__pretrain-dnn_lossEpochs}
    \end{subfigure}
    \caption{results\,/\,metrics, cGAN, DNN embedding, pre-trained, $w=64$, test chromosomes}   \label{fig:results:GAN64_pretrain-dnn_pearson}
\end{figure}
\begin{figure}[p] %cgan with DNN, pretrained, winsize 64, matrices
    \begin{subfigure}{\textwidth}
        \centering
        \scriptsize
        \import{figures/GAN_64_pretrain-dnn/}{pred00100_chr21_030-040.pdf_tex}
        \caption{example  region 1} \label{fig:results:cGAN64_pretrain-dnn_r1}
    \end{subfigure}\\[6mm]
    \begin{subfigure}{\textwidth}
        \centering
        \scriptsize
        \import{figures/GAN_64_pretrain-dnn/}{pred00100_chr19_030-040.pdf_tex}
        \caption{example region 2} \label{fig:results:cGAN64_pretrain-dnn_r2}
    \end{subfigure}\\[6mm]
    \begin{subfigure}{\textwidth}
        \centering
        \scriptsize
        \import{figures/GAN_64_pretrain-dnn/}{pred00100_chr3_030-040.pdf_tex}
        \caption{example region 3} \label{fig:results:cGAN64_pretrain-dnn_r3}
    \end{subfigure}
    \caption{example predictions, cGAN, DNN embedding, pre-trained, $w=64$, 100 epochs} \label{fig:results:cGAN64_pretrain-dnn_matrices}
\end{figure}


\subsubsection{cGAN with CNN embedding} \label{sec:results:cgan_cnn}
The results from the cGAN were generally better than the best results from the DNN,
and from windowsize 128, they were also close to the baseline or better.
Interestingly, acceptable results were obtained already after only 25 epochs.
Fast convergence is well known from pix2pix \cite{Isola2017}, but it is still surprising that
this property was maintained despite the changes made to the original network.

For windowsizes 64 and 128 bins, the optimal number of epochs seemed to be around 80,
while for windowsize 256, a number greater 100 epochs might have further improved the results.
However, this would have come at a large computation time, 
since average training time was around \SI{108}{\min} per epoch on the given hardware.

\begin{figure}[p] %cGAN with CNN, windowsize 64, pearson and progress
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/GAN_64/}{pearson_chr03.pdf_tex}}
        \caption{chr3}
    \end{subfigure} \hfill
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/GAN_64/}{pearson_chr05.pdf_tex}}
        \caption{chr5}
    \end{subfigure}\\[5mm]
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/GAN_64/}{pearson_chr10.pdf_tex}}
        \caption{chr10}
    \end{subfigure}\hfill
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/GAN_64/}{pearson_chr19.pdf_tex}}
        \caption{chr19}
    \end{subfigure}\\[3mm]
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/GAN_64/}{pearson_chr21.pdf_tex}}
        \caption{chr21}
    \end{subfigure} \hfill
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/GAN_64/}{lossOverEpochs.pdf_tex}}
        \caption{learning progress} \label{fig:results:GAN64_lossEpochs}
    \end{subfigure}
    \caption{results\,/\,metrics cGAN, CNN embedding, $w=64$, test chromosomes}   \label{fig:results:GAN64_pearson}
\end{figure}
\begin{figure}[p] %cgan CNN, 64, matrices
    \begin{subfigure}{\textwidth}
        \centering
        \scriptsize
        \import{figures/GAN_64/}{pred00100_chr21_030-040.pdf_tex}
        \caption{example  region 1} \label{fig:results:cGAN64_r1}
    \end{subfigure}\\[6mm]
    \begin{subfigure}{\textwidth}
        \centering
        \scriptsize
        \import{figures/GAN_64/}{pred00100_chr19_030-040.pdf_tex}
        \caption{example region 2} \label{fig:results:cGAN64_r2}
    \end{subfigure}\\[6mm]
    \begin{subfigure}{\textwidth}
        \centering
        \scriptsize
        \import{figures/GAN_64/}{pred00100_chr3_030-040.pdf_tex}
        \caption{example region 3} \label{fig:results:cGAN64_r3}
    \end{subfigure}
    \caption{example predictions, cGAN, CNN embedding, $w=64$, 100 epochs}
\end{figure}

\begin{figure}[p] %cGAN cnn 128, pearson and progress
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/GAN_128/}{pearson_chr03.pdf_tex}}
        \caption{chr3}
    \end{subfigure} \hfill
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/GAN_128/}{pearson_chr05.pdf_tex}}
        \caption{chr5}
    \end{subfigure}\\[5mm]
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/GAN_128/}{pearson_chr10.pdf_tex}}
        \caption{chr10}
    \end{subfigure}\hfill
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/GAN_128/}{pearson_chr19.pdf_tex}}
        \caption{chr19}
    \end{subfigure}\\[3mm]
    \centering
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/GAN_128/}{pearson_chr21.pdf_tex}}
        \caption{chr21}
    \end{subfigure} \hfill
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/GAN_128/}{lossOverEpochs.pdf_tex}}
        \caption{learning progress} \label{fig:results:GAN128_lossEpochs}
    \end{subfigure}
    \caption{results\,/\,metrics cGAN, CNN embedding. $w=128$, test chromosomes}   \label{fig:results:GAN128_pearson}
\end{figure}
\begin{figure}[p] %cgan CNN, 128, matrices
    \begin{subfigure}{\textwidth}
        \centering
        \resizebox{0.9\textwidth}{!}{
        \scriptsize
        \import{figures/GAN_128/}{pred00100_chr21_030-040.pdf_tex}}
        \caption{example  region 1} \label{fig:results:cGAN128_r1}
    \end{subfigure}\\[3mm]
    \begin{subfigure}{\textwidth}
        \centering
        \resizebox{0.9\textwidth}{!}{
        \scriptsize
        \import{figures/GAN_128/}{pred00100_chr19_030-040.pdf_tex}}
        \caption{example region 2} \label{fig:results:cGAN128_r2}
    \end{subfigure}\\[3mm]
    \begin{subfigure}{\textwidth}
        \centering
        \resizebox{0.9\textwidth}{!}{
        \scriptsize
        \import{figures/GAN_128/}{pred00100_chr3_030-040.pdf_tex}}
        \caption{example region 3} \label{fig:results:cGAN128_r3}
    \end{subfigure}
    \caption{example predictions, cGAN, CNN embedding, $w=128$, 100 epochs} \label{fig:results:cGAN128_matrices}
\end{figure}

\begin{figure}[p] %cGAN CNN 256, pearson and progress
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/GAN_256/}{pearson_chr03.pdf_tex}}
        \caption{chr3}
    \end{subfigure} \hfill
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/GAN_256/}{pearson_chr05.pdf_tex}}
        \caption{chr5}
    \end{subfigure}\\[5mm]
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/GAN_256/}{pearson_chr10.pdf_tex}}
        \caption{chr10}
    \end{subfigure}\hfill
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/GAN_256/}{pearson_chr19.pdf_tex}}
        \caption{chr19}
    \end{subfigure}\\[3mm]
    \centering
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/GAN_256/}{pearson_chr21.pdf_tex}}
        \caption{chr21}
    \end{subfigure} \hfill
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/GAN_256/}{lossOverEpochs.pdf_tex}}
        \caption{learning progress} \label{fig:results:GAN256_lossEpochs}
    \end{subfigure}
    \caption{results\,/\,metrics cGAN, CNN embedding, $w=256$, test chromosomes}   \label{fig:results:GAN256_pearson}
\end{figure}
\begin{figure}[p] %cgan CNN, 256k, matrices
    \begin{subfigure}{\textwidth}
        \centering
        \scriptsize
        \import{figures/GAN_256/}{pred00100_chr21_030-040.pdf_tex}
        \caption{example  region 1} \label{fig:results:cGAN256_r1}
    \end{subfigure}\\[6mm]
    \begin{subfigure}{\textwidth}
        \centering
        \scriptsize
        \import{figures/GAN_256/}{pred00100_chr19_030-040.pdf_tex}
        \caption{example region 2} \label{fig:results:cGAN256_r2}
    \end{subfigure}
    \caption{example predictions, cGAN, CNN embedding, $w=256$, 100 epochs}
\end{figure}
\begin{figure}\ContinuedFloat
    \begin{subfigure}{\textwidth}
        \centering
        \scriptsize
        \import{figures/GAN_256/}{pred00100_chr3_030-040.pdf_tex}
        \caption{example region 3} \label{fig:results:cGAN256_r3}
    \end{subfigure}
    \caption{example predictions, cGAN, CNN embedding, $w=256$, 100 epochs} \label{fig:results:cGAN256_matrices}
\end{figure}

\subsubsection{cGAN with mixed DNN / CNN embedding}
The results from the cGAN with mixed embedding, i.\,e. DNN-embedding for the generator
and CNN-embedding for the discriminator are shown in \cref{fig:results:GAN64_pretrained_mixed_pearson} and \ref{fig:results:GAN64_pretrained_mixed_matrices} 
(with pre-trained DNN-embedding network) and \xxx, \xxx (with standard weight initialization)

\begin{figure}[p] %cGAN mixed 64, not pretrained, pearson and progress
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/GAN_64_mixed/}{pearson_chr03.pdf_tex}}
        \caption{chr3}
    \end{subfigure} \hfill
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/GAN_64_mixed/}{pearson_chr05.pdf_tex}}
        \caption{chr5}
    \end{subfigure}\\[5mm]
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/GAN_64_mixed/}{pearson_chr10.pdf_tex}}
        \caption{chr10}
    \end{subfigure}\hfill
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/GAN_64_mixed/}{pearson_chr19.pdf_tex}}
        \caption{chr19}
    \end{subfigure}\\[3mm]
    \centering
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/GAN_64_mixed/}{pearson_chr21.pdf_tex}}
        \caption{chr21}
    \end{subfigure} \hfill
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/GAN_64_mixed/}{lossOverEpochs.pdf_tex}}
        \caption{learning progress} \label{fig:results:GAN64_mixed_lossEpochs}
    \end{subfigure}
    \caption{results\,/\,metrics cGAN, mixed embedding, no pre-training, $w=64$, test chromosomes}   \label{fig:results:GAN64_mixed_pearson}
\end{figure}
\begin{figure}[p] %cgan mixed 64, pretrained, matrices
    \begin{subfigure}{\textwidth}
        \centering
        \resizebox{0.9\textwidth}{!}{
        \scriptsize
        \import{figures/GAN_64_mixed/}{pred00080_chr21_030-040.pdf_tex}}
        \caption{example  region 1} \label{fig:results:cGAN64_mixed_r1}
    \end{subfigure}\\[3mm]
    \begin{subfigure}{\textwidth}
        \centering
        \resizebox{0.9\textwidth}{!}{
        \scriptsize
        \import{figures/GAN_64_mixed/}{pred00080_chr19_030-040.pdf_tex}}
        \caption{example region 2} \label{fig:results:cGAN64_mixed_r2}
    \end{subfigure}\\[3mm]
    \begin{subfigure}{\textwidth}
        \centering
        \resizebox{0.9\textwidth}{!}{
        \scriptsize
        \import{figures/GAN_64_mixed/}{pred00080_chr3_030-040.pdf_tex}}
        \caption{example region 3} \label{fig:results:cGAN64_mixed_r3}
    \end{subfigure}
    \caption{example predictions cGAN, mixed embedding, no pre-training, $w=64$, 80 epochs} 
     \label{fig:results:GAN64_mixed_matrices}
\end{figure}

\begin{figure}[p] %cGAN mixed 64, PRE-TRAINED, pearson and progress
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/GAN_64_pretrain-mixed/}{pearson_chr03.pdf_tex}}
        \caption{chr3}
    \end{subfigure} \hfill
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/GAN_64_pretrain-mixed/}{pearson_chr05.pdf_tex}}
        \caption{chr5}
    \end{subfigure}\\[5mm]
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/GAN_64_pretrain-mixed/}{pearson_chr10.pdf_tex}}
        \caption{chr10}
    \end{subfigure}\hfill
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/GAN_64_pretrain-mixed/}{pearson_chr19.pdf_tex}}
        \caption{chr19}
    \end{subfigure}\\[3mm]
    \centering
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/GAN_64_pretrain-mixed/}{pearson_chr21.pdf_tex}}
        \caption{chr21}
    \end{subfigure} \hfill
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/GAN_64_pretrain-mixed/}{lossOverEpochs.pdf_tex}}
        \caption{learning progress} \label{fig:results:GAN64_pretrained_mixed_lossEpochs}
    \end{subfigure}
    \caption{results\,/\,metrics cGAN, mixed embedding, DNN pre-trained, $w=64$, test chromosomes}   \label{fig:results:GAN64_pretrained_mixed_pearson}
\end{figure}
\begin{figure}[p] %cgan mixed 64, pretrained, matrices
    \begin{subfigure}{\textwidth}
        \centering
        \resizebox{0.9\textwidth}{!}{
        \scriptsize
        \import{figures/GAN_64_pretrain-mixed/}{pred00100_chr21_030-040.pdf_tex}}
        \caption{example  region 1} \label{fig:results:cGAN64_pretrained_mixed_r1}
    \end{subfigure}\\[3mm]
    \begin{subfigure}{\textwidth}
        \centering
        \resizebox{0.9\textwidth}{!}{
        \scriptsize
        \import{figures/GAN_64_pretrain-mixed/}{pred00100_chr19_030-040.pdf_tex}}
        \caption{example region 2} \label{fig:results:cGAN64_pretrained_mixed_r2}
    \end{subfigure}\\[3mm]
    \begin{subfigure}{\textwidth}
        \centering
        \resizebox{0.9\textwidth}{!}{
        \scriptsize
        \import{figures/GAN_64_pretrain-mixed/}{pred00100_chr3_030-040.pdf_tex}}
        \caption{example region 3} \label{fig:results:cGAN64_pretrained_mixed_r3}
    \end{subfigure}
    \caption{example predictions cGAN, mixed embedding, DNN pre-trained, $w=64$, 100 epochs} 
     \label{fig:results:GAN64_pretrained_mixed_matrices}
\end{figure}

\clearpage
\subsection{Comparison with other approaches}\label{sec:results:comparison}
When comparing the random-forest-based method by Zhang et al., HiC-Reg \cite{Zhang2019}, to the cGAN model trained on the typical training chromosome set, cf. \ref{sec:methods:sample_gen},
the cGAN approach seemed superior to all others for distances up to about \SI{200}{\kilo\bp}, 
while both the multicell- and window- approach by Zhang et al. outperformed DNN and Hi-cGAN for distances between 
\SI{200}{\kilo\bp} and \SI{1}{\mega\bp}, \cref{fig:results:zhang-vs-ours_pearson}.
This was also reflected in the matrix plots, \cref{fig:results:zhang-vs-ours_matrices}. 
While the cGAN often predicted smaller structures up to about \SI{400}{\kilo\bp} very well and offered distinct boundaries
even among nested structures, the approach by Zhang et al. showed better performance for interactions in the upper half of the windowsize,
see e.\,g. chromosome 17, 30...\SI{34.5}{\mega\bp}.
Note that the HiC-Reg WINDOW data stems from a random forest trained only on chromatin features and the Hi-C matrix from GM12878,
while the HiC-Reg MULTICELL data has been obtained by training on chromatin feature data from GM12878, K562, HMEC, HUVEC and NHEK and a GM12878 matrix.
Both HiC-Reg methods have used only data from chromosome 14 or 17 at binsize \SI{5}{\kilo\bp},
while in this setting, Hi-cGAN has been trained on feature- and matrix data from GM12878, chromosomes 1, 2, 4, 7, 9, 11, 13, 14, 16, 17, 18, 20, and 22 at binsizes of \SI{25}{\kilo\bp}.

To get a better comparison with the HiC-Reg WINDOW approach, in a second setting, Hi-cGAN was trained on data from chromosome 14 or 17 only.
Surprisingly, despite the low amount of training samples, the training process converged with good Pearson correlations for the (training\mbox{-})chromosomes and 
visually good matrices, \cref{fig:results:zhang-vs-ours_pearson_14singlechrom}/\ref{fig:results:zhang-vs-ours_pearson_17singlechrom} 
and \ref{fig:results:zhang-vs-ours_matrices_ours-only-14}/\ref{fig:results:zhang-vs-ours_matrices_ours-only-17}.
In general, Hi-cGAN was still better than HiC-Reg MULTICELL and WINDOW at smaller distances and worse at larger ones, 
but structures sized approximately 500 to \SI{1000}{\kilo\bp} now appeared more clearly and the ``intersecting point'' on the Pearson correlation graphs 
moved further to the right.
Interestingly, interacting pairs at distances beyond approximately \SI{1.3}{\kilo\bp} were all predicted zero, likely due to the small number of samples in this setting,
maybe in combination with insufficient training. 
The effect was more pronounced for chromosome 17, which indeed has less training samples.

Additionally, we tried training our own implementation of HiC-Reg from a previous study project \cite{Krauth2020} on the same training data, binsizes and windowsizes as the cGAN model above
to allow for a direct comparison. 
Unfortunately, we could not confirm the good results of HiC-reg, \cref{fig:results:randomforest_masterproject_pearson} and \ref{fig:results:randomforest_masterproject_matrices}, 
and it is currently unknown whether this was just due to our implementation or due to a general problem with the HiC-Reg approach in this setting.
However, we have generally not been able to reproduce the results of HiC-Reg with our implementation so far.

\begin{figure}[htbp]
 \begin{subfigure}{0.45\textwidth}
  \resizebox{\textwidth}{!}{
  \scriptsize
  \import{figures/randomforest/}{pearson_chr14_overview_25k.pdf_tex}}
  \caption{Overview chromosome 14}\label{fig:results:zhang-vs-ours_pearson_14overview}
 \end{subfigure}\hfill
\begin{subfigure}{0.45\textwidth}
  \resizebox{\textwidth}{!}{
  \scriptsize
  \import{figures/randomforest/}{pearson_chr17_overview_25k.pdf_tex}}
  \caption{Overview chromosome 17}\label{fig:results:zhang-vs-ours_pearson_17overview}
 \end{subfigure}\\[4mm]
 \begin{subfigure}{0.45\textwidth}
  \resizebox{\textwidth}{!}{
  \scriptsize
  \import{figures/randomforest/}{pearson_chr14_detail_25k.pdf_tex} }
  \caption{chr14, detail 0...\SI{1}{\mega\bp}}\label{fig:results:zhang-vs-ours_pearson_14detail}
 \end{subfigure}\hfill
 \begin{subfigure}{0.45\textwidth}
  \resizebox{\textwidth}{!}{
  \scriptsize
  \import{figures/randomforest/}{pearson_chr17_detail_25k.pdf_tex}}
  \caption{chr17, detail 0...\SI{1}{\mega\bp}}\label{fig:results:zhang-vs-ours_pearson_17detail}
 \end{subfigure}\\[4mm]
 \begin{subfigure}{0.45\textwidth}
    \resizebox{\textwidth}{!}{
    \scriptsize
    \import{figures/randomforest/}{pearson_chr14_detail_25k_singleChrom.pdf_tex}}
    \caption{chr14 cGAN single chrom}\label{fig:results:zhang-vs-ours_pearson_14singlechrom}
 \end{subfigure}\hfill
 \begin{subfigure}{0.45\textwidth}
    \resizebox{\textwidth}{!}{
    \scriptsize
    \import{figures/randomforest/}{pearson_chr17_detail_25k_singleChrom.pdf_tex}}
    \caption{chr17 cGAN single chrom}\label{fig:results:zhang-vs-ours_pearson_17singlechrom}
 \end{subfigure}
  \caption{Pearson correlation comparison Hi-cGAN / DNN and HiC-Reg \cite{Zhang2019}} \label{fig:results:zhang-vs-ours_pearson}
\end{figure}

\begin{figure}[htbp]
\begin{subfigure}{\textwidth}
 \centering
 \scriptsize
 \import{figures/randomforest/}{pred0014_multicell_chr14_030-040.pdf_tex}
 \caption{Hic-Reg (random forest), MULTICELL, reconstructed from \cite{Zhang2019}} \label{fig:results:zhang-vs-ours_matrices_multicell-14}
\end{subfigure}\\[5mm]
\begin{subfigure}{\textwidth}
 \centering
 \scriptsize
 \import{figures/randomforest/}{pred00100_restriced_chr14_030-040.pdf_tex}
 \caption{Hi-cGAN with windowsize $w=256$; plot restricted to first \SI{1}{\mega\bp}} \label{fig:results:zhang-vs-ours_matrices_ours-full-14}
\end{subfigure}\\[5mm]
\begin{subfigure}{\textwidth}
 \centering
 \scriptsize
 \import{figures/randomforest/}{pred00100_chr14only_030-040.pdf_tex}
 \caption{Hi-cGAN with windowsize $w=64$, trained on chr14 only} \label{fig:results:zhang-vs-ours_matrices_ours-only-14}
\end{subfigure}\\[8mm]
\begin{subfigure}{\textwidth}
 \centering
 \scriptsize
 \import{figures/randomforest/}{pred0017_multicell_chr17_030-040.pdf_tex}
 \caption{Hic-Reg (random forest), MULTICELL, chr17, reconstructed from \cite{Zhang2019}} \label{fig:results:zhang-vs-ours_matrices_multicell-17}
\end{subfigure}\\[5mm]
\begin{subfigure}{\textwidth}
 \centering
 \scriptsize
 \import{figures/randomforest/}{pred00100_restriced_chr17_030-040.pdf_tex}
 \caption{Hi-cGAN with windowsize $w=256$, chr17, plot restricted to first \SI{1}{\mega\bp}}\label{fig:results:zhang-vs-ours_matrices_ours-full-17}
\end{subfigure}\\[5mm]
\begin{subfigure}{\textwidth}
 \centering
 \scriptsize
 \import{figures/randomforest/}{pred00100_chr17only_030-040.pdf_tex}
 \caption{Hi-cGAN with windowsize $w=64$, trained on chr17 only} \label{fig:results:zhang-vs-ours_matrices_ours-only-17}
\end{subfigure}
\caption{Comparison HiC-Reg \cite{Zhang2019} and Hi-cGAN (CNN embedding)} \label{fig:results:zhang-vs-ours_matrices}
\end{figure}

\begin{figure}[p] %our random forest, w=80, same train chroms as everywhere
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/randomforest_masterproject/}{pearson_chr03.pdf_tex}}
        \caption{chr3}
    \end{subfigure} \hfill
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/randomforest_masterproject/}{pearson_chr05.pdf_tex}}
        \caption{chr5}
    \end{subfigure}\\[5mm]
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/randomforest_masterproject/}{pearson_chr10.pdf_tex}}
        \caption{chr10}
    \end{subfigure}\hfill
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/randomforest_masterproject/}{pearson_chr19.pdf_tex}}
        \caption{chr19}
    \end{subfigure}\\[3mm]
    \centering
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/randomforest_masterproject/}{pearson_chr21.pdf_tex}}
        \caption{chr21}
    \end{subfigure} \hfill
    \caption{results\,/\,metrics, random forest from study project \cite{Krauth2020}, windowsize 80, \SI{25}{\kilo\bp}, test chromosomes}  \label{fig:results:randomforest_masterproject_pearson}
\end{figure}
\begin{figure}[p] %our random forest, windowsize 80, matrices
    \begin{subfigure}{\textwidth}
        \centering
        \resizebox{0.9\textwidth}{!}{
        \scriptsize
        \import{figures/randomforest_masterproject/}{pred00021_chr21_030-040.pdf_tex}}
        \caption{example  region 1} \label{fig:results:randomforest_masterproject_r1}
    \end{subfigure}\\[3mm]
    \begin{subfigure}{\textwidth}
        \centering
        \resizebox{0.9\textwidth}{!}{
        \scriptsize
        \import{figures/randomforest_masterproject/}{pred00019_chr19_030-040.pdf_tex}}
        \caption{example region 2} \label{fig:results:randomforest_masterproject_r2}
    \end{subfigure}\\[3mm]
    \begin{subfigure}{\textwidth}
        \centering
        \resizebox{0.9\textwidth}{!}{
        \scriptsize
        \import{figures/randomforest_masterproject/}{pred00003_chr3_030-040.pdf_tex}}
        \caption{example region 3} \label{fig:results:randomforest_masterproject_r3}
    \end{subfigure}
    \caption{example predictions random forest \cite{Krauth2020}, \SI{25}{\kilo\bp}, $w=80$} 
     \label{fig:results:randomforest_masterproject_matrices}
\end{figure}
