\section{Results}
In the following two sections, 
the results of the \acrshort{dnn}- and \acrshort{cgan}-based approaches for predicting Hi-C contact matrices will be presented.
While the various modifications of the dense neural network described in \cref{sec:improve:DNNapproach} 
did  not really help improve the status quo, the novel conditional generative adversarial network laid out in \cref{sec:improve:Hi-cGAN}
showed interesting properties.

\subsection{Dense Neural Network approaches} \label{sec:results:DNN}
In the following subsections, the resulting predictions for modifications of the dense neural network originally conceived 
by Farr\'e, Heurteau, Cuvier and Emberly \cite{Farre2018a} will be shown. 
This includes variations of the convolutional layer(s), custom loss functions with the intent of reducing blurriness in the predictions
and tuning window- and bin size, cf. \cref{sec:improve:DNNapproach}.
As a start, however, the results of the initial network without any modifications will be shown for comparison.

\subsubsection{Initial results for comparison} \label{sec:initialDNNresults}
The basic network was setup and trained as explained in \cref{sec:methods:basicSetup}.
Here, the validation error (MSE) reached its minimum of about \SI{150000}{} 
after approximately 500 epochs for \SI{25}{\kilo\bp} bin size and around 400 epochs for \SI{10}{\kilo\bp} bin size, \cref{fig:results:basicDNN_lossEpochs_25} and \ref{fig:results:basicDNN_lossEpochs_10}.
Beyond that, the learning curve indicated overfitting, but the resulting test matrices often still looked fairly similar, 
compare e.\,g. the matrix plots after 500 and 1000 epochs in figures \ref{fig:results:basic500} and \ref{fig:results:basic1000}.

\Cref{fig:results:basicDNN_pearson} and \cref{fig:results:basicDNN_10k_pearson} 
show the distance stratified Pearson correlations (cf. \cref{sec:methods:metrics}) 
alongside \acrfull{auc} for the five test chromosomes at bin sizes 25 and \SI{10}{\kilo\bp}, respectively.
The red curves in each correlation plot show the correlation between the corresponding training chromosome from GM12878 and the target chromosome from K562.
It is obvious that all predicted test matrices have a strictly positive Pearson correlation with respect to the experimentally derived matrices by Rao et al. \cite{Rao2014},
but are not better than simply taking data from the training cell line as prediction for the target cell line.

The predicted matrices themselves looked modest when plotted with pygenometracks. 
While the \acrshort{dnn} generally produced high interaction counts in regions with many true interactions
and low interaction counts in regions with few true interactions, the (TAD-)boundaries between different interacting domains 
were mostly not discernible, \cref{fig:results:basic500} and \ref{fig:results:basic1000}.
This finding is in line with the clearly positive, but medium-valued Pearson correlations.
Exceptions with more distinct boundaries existed in all of the five test chromosomes,
for example chr19, 34 to \SI{35}{\kilo\bp} (\cref{fig:results:basic_r2}), but were rare. 
Interestingly, medium-sized interacting structures, for example chr21, 31 to \SI{32.5}{\kilo\bp} 
or between chr19, 31.2 to \SI{32.7}{\kilo\bp} often seemed to be missing altogether -- 
while structures larger than the window size, for example chr3, 34 and \SI{36.7}{\kilo\bp} and 36.7 and \SI{39.5}{\kilo\bp}
sometimes were at least indicated.

Reducing the bin size to $b_\mathit{feat}=b_\mathit{mat}=\SI{10}{\kilo\bp}$ as in the paper by Farr\'e et al. \cite{Farre2018a} 
led to somewhat different results.
The area under the correlation curves was approximately the same for test chromosomes 3 and 5, slightly better for chromosome 10, but worse for chromosome 19 and 21,
cf.~\cref{fig:results:basicDNN_pearson} and \ref{fig:results:basicDNN_10k_pearson}.
However, the ability to predict larger structures was lost, and the matrix plots thus did not look better than the ones for bin size \SI{25}{\kilo\bp}.
The comparatively bad result for test chromosome 21 might result from the low chromatin feature coverage of this particular chromosome.

No obvious correlation between comparatively ``good'' and ``bad'' predictions with open and closed states of the chromatin was observed.
However, formally computing such a correlation is challenging, because no adequate objective measure for ``good'' and ``bad'' is known, 
especially considering the rather blurry results obtained so far.
Furthermore, even if suchlike correlations existed, exploiting them for improving predictions would still be, at best, not straightforward.

\begin{figure}[p]
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/DNN_basic/}{pearson_chr03.pdf_tex}}
        \caption{chr3}
    \end{subfigure} \hfill
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/DNN_basic/}{pearson_chr05.pdf_tex}}
        \caption{chr5}
    \end{subfigure}\\[5mm]
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/DNN_basic/}{pearson_chr10.pdf_tex}}
        \caption{chr10}
    \end{subfigure}\hfill
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/DNN_basic/}{pearson_chr19.pdf_tex}}
        \caption{chr19}
    \end{subfigure}\\[3mm]
    \centering
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/DNN_basic/}{pearson_chr21.pdf_tex}}
        \caption{chr21}
    \end{subfigure}\hfill
    \begin{subfigure}{0.45\textwidth}
        \resizebox{\textwidth}{!}{
        \scriptsize
        \import{figures/DNN_basic/}{lossOverEpochs.pdf_tex}}
        \caption{Learning progress}\label{fig:results:basicDNN_lossEpochs_25}
    \end{subfigure}
    \caption{Results\,/\,metrics, basic \acrshort{dnn}, \SI{25}{\kilo\bp}, test chromosomes}
    \label{fig:results:basicDNN_pearson}
\end{figure}

%10k Pearson and progress
\begin{figure}[p]
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/DNN_basic10k/}{pearson_chr03.pdf_tex}}
        \caption{chr3}
    \end{subfigure} \hfill
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/DNN_basic10k/}{pearson_chr05.pdf_tex}}
        \caption{chr5}
    \end{subfigure}\\[5mm]
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/DNN_basic10k/}{pearson_chr10.pdf_tex}}
        \caption{chr10}
    \end{subfigure}\hfill
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/DNN_basic10k/}{pearson_chr19.pdf_tex}}
        \caption{chr19}
    \end{subfigure}\\[3mm]
    \centering
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/DNN_basic10k/}{pearson_chr21.pdf_tex}}
        \caption{chr21}
    \end{subfigure}\hfill
     \begin{subfigure}{0.45\textwidth}
        \resizebox{\textwidth}{!}{
        \scriptsize
        \import{figures/DNN_basic10k/}{lossOverEpochs.pdf_tex}}
        \caption{Learning progress}\label{fig:results:basicDNN_lossEpochs_10}
    \end{subfigure}
    \caption{Results\,/\,metrics, basic \acrshort{dnn}, \SI{10}{\kilo\bp}, test chromosomes}
    \label{fig:results:basicDNN_10k_pearson}
\end{figure}
%25k matrices, after 500 epochs
\begin{figure}[p]
    \begin{subfigure}{\textwidth}
        \centering
        \scriptsize
        \import{figures/DNN_basic/}{pred00500_chr21_030-040.pdf_tex}
        \caption{example  region 1, 500 epochs} \label{fig:results:basic_r1}
    \end{subfigure}\\[2mm]
    \centering{\scriptsize{see \cref{sec:methods:matrix_plots} for track explanations}}\\[3mm]
    \begin{subfigure}{\textwidth}
        \centering
        \scriptsize
        \import{figures/DNN_basic/}{pred00500_chr19_030-040.pdf_tex}
        \caption{example region 2, 500 epochs} \label{fig:results:basic_r2}
    \end{subfigure}\\[2mm]
    \centering{\scriptsize{see \cref{sec:methods:matrix_plots} for track explanations}}\\[3mm]
    \begin{subfigure}{\textwidth}
        \centering
        \scriptsize
        \import{figures/DNN_basic/}{pred00500_chr3_030-040.pdf_tex}
        \caption{example region 2, 500 epochs} \label{fig:results:basic_r3}
    \end{subfigure}
    \caption{Example predictions GM12878 $\rightarrow$ K562, basic \acrshort{dnn}, \SI{25}{\kilo\bp}, 500~epochs} \label{fig:results:basic500}
\end{figure}
%25k matrices, after 1000 epochs
\begin{figure}[p]
    \begin{subfigure}{\textwidth}
        \centering
        \scriptsize
        \import{figures/DNN_basic/}{pred01000_chr21_030-040.pdf_tex}
        \caption{Example region 1, 1000 epochs} \label{fig:results:basic_r1_1000}
    \end{subfigure}\\[2mm]
    \centering{\scriptsize{see \cref{sec:methods:matrix_plots} for track explanations}}\\[3mm]
    \begin{subfigure}{\textwidth}
        \centering
        \scriptsize
        \import{figures/DNN_basic/}{pred01000_chr19_030-040.pdf_tex}
        \caption{Example region 2, 1000 epochs} \label{fig:results:basic_r2_1000}
    \end{subfigure}\\[2mm]
    \centering{\scriptsize{see \cref{sec:methods:matrix_plots} for track explanations}}\\[3mm]
    \begin{subfigure}{\textwidth}
        \centering
        \scriptsize
        \import{figures/DNN_basic/}{pred01000_chr3_030-040.pdf_tex}
        \caption{Example region 2, 1000 epochs} \label{fig:results:basic_r3_1000}
    \end{subfigure}
    \caption{Example predictions GM12878 $\rightarrow$ K562, basic \acrshort{dnn}, \SI{25}{\kilo\bp}, 1000~epochs} \label{fig:results:basic1000}
\end{figure}
%10k matrices
\begin{figure}[p]
    \begin{subfigure}{\textwidth}
        \centering
        \scriptsize
        \import{figures/DNN_basic10k/}{pred01000_chr21_030-040.pdf_tex}
        \caption{Example  region 1} \label{fig:results:basic10k_r1}
    \end{subfigure}\\[2mm]
    \centering{\scriptsize{see \cref{sec:methods:matrix_plots} for track explanations}}\\[3mm]
    \begin{subfigure}{\textwidth}
        \centering
        \scriptsize
        \import{figures/DNN_basic10k/}{pred01000_chr19_030-040.pdf_tex}
        \caption{Example region 2} \label{fig:results:basic10k_r2}
    \end{subfigure}\\[2mm]
    \centering{\scriptsize{see \cref{sec:methods:matrix_plots} for track explanations}}\\[3mm]
    \begin{subfigure}{\textwidth}
        \centering
        \scriptsize
        \import{figures/DNN_basic10k/}{pred01000_chr3_030-040.pdf_tex}
        \caption{Example region 3} \label{fig:results:basic10k_r3}
    \end{subfigure}
    \caption{Example predictions GM12878 $\rightarrow$ K562, basic \acrshort{dnn} \SI{10}{\kilo\bp}, 1000~epochs}
\end{figure}

\subsubsection{Results for variations of the convolutional part} \label{sec:results:wider-longer-etc}
The results for the ``wider'' network, which featured a wider convolutional filter in the first network layer, 
cf. \cref{sec:improve:convolution_extensions} and \ref{sec:methods:variants}, 
were generally similar to the initial results, both in terms of Pearson correlations and in terms of matrix plots, 
\cref{fig:results:widerDNN_pearson} and \ref{fig:results:wider_matrices}.
Given the small increase in the number of trainable parameters and overall similar network topology, this is not surprising.
Overfitting was less obvious than with the initial setup and the training process looked more smooth, 
but the remaining validation error was slightly higher than for the initial approach, \cref{fig:results:widerDNN_lossEpochs}.
\begin{figure}[p] %wider variant Pearson
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/DNN_wider/}{pearson_chr03.pdf_tex}}
        \caption{chr3}
    \end{subfigure} \hfill
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/DNN_wider/}{pearson_chr05.pdf_tex}}
        \caption{chr5}
    \end{subfigure}\\[5mm]
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/DNN_wider/}{pearson_chr10.pdf_tex}}
        \caption{chr10}
    \end{subfigure}\hfill
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/DNN_wider/}{pearson_chr19.pdf_tex}}
        \caption{chr19}
    \end{subfigure}\\[3mm]
    \centering
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/DNN_wider/}{pearson_chr21.pdf_tex}}
        \caption{chr21}
    \end{subfigure}\hfill
    \begin{subfigure}{0.45\textwidth}
        \resizebox{\textwidth}{!}{
        \scriptsize
        \import{figures/DNN_wider/}{lossOverEpochs.pdf_tex}}
        \caption{Learning progress for wider \acrshort{dnn}} \label{fig:results:widerDNN_lossEpochs}
    \end{subfigure}
    \caption{Pearson correlations, ``wider'' variant of \acrshort{dnn},  test chromosomes}
    \label{fig:results:widerDNN_pearson}
\end{figure}
%wider variant matrices
\begin{figure}[p]
    \begin{subfigure}{\textwidth}
        \centering
        \scriptsize
        \import{figures/DNN_wider/}{pred01000_chr21_030-040.pdf_tex}
        \caption{Example  region 1} \label{fig:results:wider_r1}
    \end{subfigure}\\[2mm]
    \centering{\scriptsize{see \cref{sec:methods:matrix_plots} for track explanations}}\\[3mm]
    \begin{subfigure}{\textwidth}
        \centering
        \scriptsize
        \import{figures/DNN_wider/}{pred01000_chr19_030-040.pdf_tex}
        \caption{Example region 2} \label{fig:results:wider_r2}
    \end{subfigure}\\[2mm]
    \centering{\scriptsize{see \cref{sec:methods:matrix_plots} for track explanations}}\\[3mm]
    \begin{subfigure}{\textwidth}
        \centering
        \scriptsize
        \import{figures/DNN_wider/}{pred01000_chr3_030-040.pdf_tex}
        \caption{Example region 3} \label{fig:results:wider_r3}
    \end{subfigure}
    \caption{Example predictions GM12878 $\rightarrow$ K562, ``wider'' variant of \acrshort{dnn},\\binsize \SI{25}{\kilo\bp}, 1000~epochs}\label{fig:results:wider_matrices}
\end{figure}

The predictions from the ``longer'' variant with three convolutional filter layers instead of a single one were better than the initial predictions in terms of
Pearson correlations for test chromosomes 10, 19 and 21, but worse for test chromosomes 3 and 5, \cref{fig:results:longerDNN_pearson}.
Interestingly, correlations for some of the larger distances could not be computed after 250 and 500 epochs, 
which generally means that the same values were predicted for these distances, cf. \cref{sec:methods:metrics}.
The reason for this behavior is not fully understood yet, but comparatively few neurons in the outermost layer are responsible for predictions at 
longer distances due to the chosen network setup, cf. \cref{sec:methods:sample_gen}, \cref{fig:methods:prediction}.
Since the longer network variant has a considerably larger number of trainable parameters,
it is assumed that 500 epochs might not have been enough to activate some of the outer neurons.
Slow training can occur when ReLU activations are used (as in the given case) and the gradients are close to zero \cite{Maas2013}.
Apart from that, the learning process for the ``longer'' variant in general looked more smooth and reached a lower validation error than before, \cref{fig:results:longerDNN_lossEpochs},
but the matrix plots did not show any obvious improvement over the initial ones, \cref{fig:results:longer_matrices}.
\begin{figure}[p] %longer variant pearson
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/DNN_longer/}{pearson_chr03.pdf_tex}}
        \caption{chr3}
    \end{subfigure} \hfill
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/DNN_longer/}{pearson_chr05.pdf_tex}}
        \caption{chr5}
    \end{subfigure}\\[5mm]
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/DNN_longer/}{pearson_chr10.pdf_tex}}
        \caption{chr10}
    \end{subfigure}\hfill
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/DNN_longer/}{pearson_chr19.pdf_tex}}
        \caption{chr19}
    \end{subfigure}\\[3mm]
    \centering
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/DNN_longer/}{pearson_chr21.pdf_tex}}
        \caption{chr21}
    \end{subfigure}\hfill
    \begin{subfigure}{0.45\textwidth}
        \resizebox{\textwidth}{!}{
        \scriptsize
        \import{figures/DNN_longer/}{lossOverEpochs.pdf_tex}}
        \caption{Learning progress for longer \acrshort{dnn}} \label{fig:results:longerDNN_lossEpochs}
    \end{subfigure}
    \caption{Results\,/\,metrics, ``longer'' variant of \acrshort{dnn},  test chromosomes}
    \label{fig:results:longerDNN_pearson}
\end{figure}
%longer variant matrices
\begin{figure}[p]
    \begin{subfigure}{\textwidth}
        \centering
        \scriptsize
        \import{figures/DNN_longer/}{pred01000_chr21_030-040.pdf_tex}
        \caption{Example  region 1} \label{fig:results:longer_r1}
    \end{subfigure}\\[2mm]
    \centering{\scriptsize{see \cref{sec:methods:matrix_plots} for track explanations}}\\[3mm]
    \begin{subfigure}{\textwidth}
        \centering
        \scriptsize
        \import{figures/DNN_longer/}{pred01000_chr19_030-040.pdf_tex}
        \caption{Example region 2} \label{fig:results:longer_r2}
    \end{subfigure}\\[2mm]
    \centering{\scriptsize{see \cref{sec:methods:matrix_plots} for track explanations}}\\[3mm]
    \begin{subfigure}{\textwidth}
        \centering
        \scriptsize
        \import{figures/DNN_longer/}{pred01000_chr3_030-040.pdf_tex}
        \caption{Example region 3} \label{fig:results:longer_r3}
    \end{subfigure}
    \caption{Example predictions GM12878 $\rightarrow$ K562, ``longer'' variant of \acrshort{dnn},\\binsize \SI{25}{\kilo\bp}, 1000~epochs} \label{fig:results:longer_matrices}
\end{figure}

Combining the ``longer'' and ``wider'' variants in the ``wider-longer'' setup with more convolutional layers and wider 
filter kernels also did not perform as expected.
While improvements in the Pearson correlations could again be seen for 3 of 5 test chromosomes compared to the initial network, \cref{fig:results:wider-longerDNN_pearson},
the observed correlations were worse than the ones from the highly similar ``longer''-variant alone. 
Like with the similar ``longer''-approach predictions at longer distances were partially missing. 
Compared to the other variants, the validation error was generally higher and stopped decreasing after very few epochs, \cref{fig:results:wider-longerDNN_lossEpochs}.
In terms of matrix plots, the predictions surprisingly were still quite similar to the initial ones, but seemed a bit more blurry, \ref{fig:results:wider-longer_matrices}.
\begin{figure}[p]%wider-longer Pearson
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/DNN_wider-longer/}{pearson_chr03.pdf_tex}}
        \caption{chr3}
    \end{subfigure} \hfill
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/DNN_wider-longer/}{pearson_chr05.pdf_tex}}
        \caption{chr5}
    \end{subfigure}\\[5mm]
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/DNN_wider-longer/}{pearson_chr10.pdf_tex}}
        \caption{chr10}
    \end{subfigure}\hfill
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/DNN_wider-longer/}{pearson_chr19.pdf_tex}}
        \caption{chr19}
    \end{subfigure}\\[3mm]
    \centering
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/DNN_wider-longer/}{pearson_chr21.pdf_tex}}
        \caption{chr21}
    \end{subfigure}\hfill
    \begin{subfigure}{0.45\textwidth}
        \resizebox{\textwidth}{!}{
        \scriptsize
        \import{figures/DNN_wider-longer/}{lossOverEpochs.pdf_tex}}
        \caption{Learning progress} \label{fig:results:wider-longerDNN_lossEpochs}
    \end{subfigure}
    \caption{Results\,/\,metrics, ``wider-longer'' variant of \acrshort{dnn},  test chromosomes}
    \label{fig:results:wider-longerDNN_pearson}
\end{figure}
%wider-longer variant matrices
\begin{figure}[p]
    \begin{subfigure}{\textwidth}
        \centering
        \scriptsize
        \import{figures/DNN_wider-longer/}{pred01000_chr21_030-040.pdf_tex}
        \caption{Example  region 1} \label{fig:results:wider-longer_r1}
    \end{subfigure}\\[2mm]
    \centering{\scriptsize{see \cref{sec:methods:matrix_plots} for track explanations}}\\[3mm]
    \begin{subfigure}{\textwidth}
        \centering
        \scriptsize
        \import{figures/DNN_wider-longer/}{pred01000_chr19_030-040.pdf_tex}
        \caption{Example region 2} \label{fig:results:wider-longer_r2}
    \end{subfigure}\\[2mm]
    \centering{\scriptsize{see \cref{sec:methods:matrix_plots} for track explanations}}\\[3mm]
    \begin{subfigure}{\textwidth}
        \centering
        \scriptsize
        \import{figures/DNN_wider-longer/}{pred01000_chr3_030-040.pdf_tex}
        \caption{Example region 3} \label{fig:results:wider-longer_r3}
    \end{subfigure}
    \caption{Example predictions GM12878 $\rightarrow$ K562, ``wider-longer'' variant of \acrshort{dnn}, \SI{25}{\kilo\bp}, 1000~epochs} \label{fig:results:wider-longer_matrices}
\end{figure}

Predictions and metrics from the generalized \acrshort{dnn} approach with feature bin size \SI{5}{\kilo\bp} and matrix bin size \SI{25}{\kilo\bp}
are shown in \cref{fig:results:25k5DNN_pearson} and \ref{fig:results:25k5_matrices}.
Unfortunately, the results did again not improve compared to the initial predictions.
While the learning curve was smooth and showed signs of slight overfitting beyond 300 epochs, \cref{fig:results:25k5DNN_lossEpochs},
the matrix plots seemed worse than the initial ones. 
For example, the large structure at chromosome 3, 34 to \SI{36.7}{\kilo\bp}, which had been detected by the previous approaches, was completely missing.
\begin{figure}[p]%25k5 Pearson
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/DNN_25k5/}{pearson_chr03.pdf_tex}}
        \caption{chr3}
    \end{subfigure} \hfill
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/DNN_25k5/}{pearson_chr05.pdf_tex}}
        \caption{chr5}
    \end{subfigure}\\[5mm]
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/DNN_25k5/}{pearson_chr10.pdf_tex}}
        \caption{chr10}
    \end{subfigure}\hfill
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/DNN_25k5/}{pearson_chr19.pdf_tex}}
        \caption{chr19}
    \end{subfigure}\\[3mm]
    \centering
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/DNN_25k5/}{pearson_chr21.pdf_tex}}
        \caption{chr21}
    \end{subfigure}\hfill
    \begin{subfigure}{0.45\textwidth}
        \resizebox{\textwidth}{!}{
        \scriptsize
        \import{figures/DNN_25k5/}{lossOverEpochs.pdf_tex}}
        \caption{Learning progress} \label{fig:results:25k5DNN_lossEpochs}
    \end{subfigure}
    \caption{Results\,/\,metrics, ``5k -- 25k'' variant of \acrshort{dnn} with $b_\mathit{feat}=\SI{5}{\kilo\bp}$ and $b_\mathit{mat}=\SI{25}{\kilo\bp}$,  test chromosomes}
    \label{fig:results:25k5DNN_pearson}
\end{figure}
%25k5 matrices
\begin{figure}[p]
    \begin{subfigure}{\textwidth}
        \centering
        \scriptsize
        \import{figures/DNN_25k5/}{pred01000_chr21_030-040.pdf_tex}
        \caption{Example  region 1} \label{fig:results:25k5_r1}
    \end{subfigure}\\[2mm]
    \centering{\scriptsize{see \cref{sec:methods:matrix_plots} for track explanations}}\\[3mm]
    \begin{subfigure}{\textwidth}
        \centering
        \scriptsize
        \import{figures/DNN_25k5/}{pred01000_chr19_030-040.pdf_tex}
        \caption{Example region 2} \label{fig:results:25k5_r2}
    \end{subfigure}\\[2mm]
    \centering{\scriptsize{see \cref{sec:methods:matrix_plots} for track explanations}}\\[3mm]
    \begin{subfigure}{\textwidth}
        \centering
        \scriptsize
        \import{figures/DNN_25k5/}{pred01000_chr3_030-040.pdf_tex}
        \caption{Example region 3} \label{fig:results:25k5_r3}
    \end{subfigure}
    \caption{Example predictions GM12878 $\rightarrow$ K562, ``5k -- 25k'' variant of \acrshort{dnn}, 1000~epochs} \label{fig:results:25k5_matrices}
\end{figure}

\subsubsection{Results for combined loss function} \label{sec:results:loss_functions}
Exchanging the mean squared error for a combined loss function consisting of MSE, TV loss and perceptual loss did not improve the results in the given setting.
The results for optimizing a loss function according to \cref{eq:methods:combined_loss} with weighting parameters 
are shown in \cref{fig:results:combilossDNN_pearson} and \ref{fig:results:combiloss_matrices}.

For all test chromosomes, the correlations were highly similar to the initial network's, \cref{fig:results:combilossDNN_pearson},
and the matrix plots also looked similar, chromosome 21 probably being the most different, \cref{fig:results:combiloss_matrices}.
The results plotted are the best ones obtained by manual tuning of the multiplicative parameters $\lambda$. 
Guided parameter tuning was unfortunately infeasible within the thesis at hand due to the training times required for optimizing the combined loss function.
Other options which where not explored for the same reason include truncating the \emph{VGG-16} network at a different layer, using a loss function based on
more than one of the intermediate \emph{VGG-16} layers \cite{Johnson2016} or taking another loss network.
However, the results obtained thus far were also not encouraging towards such investigations.
\begin{figure}[p] %combiloss pearson and progress
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/DNN_combiloss/}{pearson_chr03.pdf_tex}}
        \caption{chr3}
    \end{subfigure} \hfill
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/DNN_combiloss/}{pearson_chr05.pdf_tex}}
        \caption{chr5}
    \end{subfigure}\\[5mm]
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/DNN_combiloss/}{pearson_chr10.pdf_tex}}
        \caption{chr10}
    \end{subfigure}\hfill
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/DNN_combiloss/}{pearson_chr19.pdf_tex}}
        \caption{chr19}
    \end{subfigure}\\[3mm]
    \centering
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/DNN_combiloss/}{pearson_chr21.pdf_tex}}
        \caption{chr21}
    \end{subfigure}\hfill
    \begin{subfigure}{0.45\textwidth}
        \resizebox{\textwidth}{!}{
        \scriptsize
        \import{figures/DNN_combiloss/}{lossOverEpochs.pdf_tex}}
        \caption{Learning progress} \label{fig:results:combilossDNN_lossEpochs}
    \end{subfigure}
    \caption{Results\,/\,metrics, \acrshort{dnn} with combined loss function (MSE, TV, VGG-16),  test chromosomes}
    \label{fig:results:combilossDNN_pearson}
\end{figure}
%combiloss matrices
\begin{figure}[p]
    \begin{subfigure}{\textwidth}
        \centering
        \scriptsize
        \import{figures/DNN_combiloss/}{pred00500_chr21_030-040.pdf_tex}
        \caption{Example  region 1} \label{fig:results:combiloss_r1}
    \end{subfigure}\\[2mm]
    \centering{\scriptsize{see \cref{sec:methods:matrix_plots} for track explanations}}\\[3mm]
    \begin{subfigure}{\textwidth}
        \centering
        \scriptsize
        \import{figures/DNN_combiloss/}{pred00500_chr19_030-040.pdf_tex}
        \caption{Example region 2} \label{fig:results:combiloss_r2}
    \end{subfigure}\\[2mm]
    \centering{\scriptsize{see \cref{sec:methods:matrix_plots} for track explanations}}\\[3mm]
    \begin{subfigure}{\textwidth}
        \centering
        \scriptsize
        \import{figures/DNN_combiloss/}{pred00500_chr3_030-040.pdf_tex}
        \caption{Example region 3} \label{fig:results:combiloss_r3}
    \end{subfigure}
    \caption{Example predictions GM12878 $\rightarrow$ K562, \acrshort{dnn} with combined loss function (MSE, TV, VGG-16), 500~epochs} \label{fig:results:combiloss_matrices}
\end{figure}
While manually searching better parameters $\lambda$ was not successful,
it was found that the TV loss weight $\lambda_\mathit{TV}$ needed to be much smaller than the two other weights.
Otherwise, many true interactions off the matrix diagonals were considered as noise and optimized away early in the training process, cf.
\cref{fig:appendix:failed_tv_loss} (p. \pageref{fig:appendix:failed_tv_loss}).

\subsubsection{Results for score-based loss function} \label{sec:results:scorebased}
Exchanging the MSE- loss function by a combination between score-based and MSE loss
allowed for a smooth learning process and a slightly lower validation error compared to the initial approach.
However, at around \SI{7}{\min} per epoch on a GPU, the training process also was about seven times slower than the initial approach on CPU.
Unfortunately, the higher effort did not lead to obvious improvements.

The Pearson correlations for a score-based loss function with parameters $\lambda_\mathit{MSE}=1.0,\; \lambda_\mathit{score}=100,\; ds=12$ 
are shown in figure \ref{fig:results:scoreLossDNN_pearson}.
While a slight improvement was achieved for test chromosome 21, the correlations of the others remained widely unchanged.
The matrix plots also looked fairly similar to the initial ones, \cref{fig:results:scoreloss_matrices}, chromosome 21 again being the 
most different compared to the initial predictions.

In the matrix plots, the true- and predicted scores have been added as a second track, replacing the PCA track. 
Indeed, the score curve computed from the true matrices showed local minima at putative TAD boundaries, as set forth in \ref{sec:improve:TAD_loss},
so score computation with the chosen diamond size seemed sound.
However, despite the optimization term in the loss function, the score curve of the predicted matrices compared to the true curve 
somewhat like the predicted matrices compared to the true ones:
The predicted score was generally high, when the true score was high, and low when the true score was also low,
but high peaks (local maxima) and steep valleys (local minima) in the plots were usually averaged out.

Long training times forbade a targeted parameter tuning by grid- or tree-search,
so the results presented in this section should not be interpreted as the optimal ones achievable by a score-based loss function.
\begin{figure}[p]%score loss pearson and progress
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/DNN_scoreLoss/}{pearson_chr03.pdf_tex}}
        \caption{chr3}
    \end{subfigure} \hfill
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/DNN_scoreLoss/}{pearson_chr05.pdf_tex}}
        \caption{chr5}
    \end{subfigure}\\[5mm]
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/DNN_scoreLoss/}{pearson_chr10.pdf_tex}}
        \caption{chr10}
    \end{subfigure}\hfill
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/DNN_scoreLoss/}{pearson_chr19.pdf_tex}}
        \caption{chr19}
    \end{subfigure}\\[3mm]
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/DNN_scoreLoss/}{pearson_chr21.pdf_tex}}
        \caption{chr21}
    \end{subfigure}\hfill
    \begin{subfigure}{0.45\textwidth}
        \resizebox{\textwidth}{!}{
        \scriptsize
        \import{figures/DNN_scoreLoss/}{lossOverEpochs.pdf_tex}}
        \caption{Learning progress} \label{fig:results:scoreLossDNN_lossEpochs}
    \end{subfigure}
    \caption{Results\,/\,metrics, \acrshort{dnn} with score-based loss function, test chromosomes\\ ($\lambda_\mathit{MSE}=1.0,\; \lambda_\mathit{score}=100,\; ds=12$)} \label{fig:results:scoreLossDNN_pearson}
\end{figure}
\begin{figure}[p] %score loss matrices
    \begin{subfigure}{\textwidth}
        \centering
        \scriptsize
        \import{figures/DNN_scoreLoss/}{pred00500_chr21_030-040.pdf_tex}
        \caption{Example  region 1} \label{fig:results:scoreloss_r1}
    \end{subfigure}\\[2mm]
    \centering{\scriptsize{see \cref{sec:methods:matrix_plots} for track explanations}}\\[3mm]
    \begin{subfigure}{\textwidth}
        \centering
        \scriptsize
        \import{figures/DNN_scoreLoss/}{pred00500_chr19_030-040.pdf_tex}
        \caption{Example region 2} \label{fig:results:scoreloss_r2}
    \end{subfigure}\\[2mm]
    \centering{\scriptsize{see \cref{sec:methods:matrix_plots} for track explanations}}\\[3mm]
    \begin{subfigure}{\textwidth}
        \centering
        \scriptsize
        \import{figures/DNN_scoreLoss/}{pred00500_chr3_030-040.pdf_tex}
        \caption{Example region 3} \label{fig:results:scoreloss_r3}
    \end{subfigure}
    \caption{Example predictions GM12878 $\rightarrow$ K562,  \acrshort{dnn} with score-based loss function, 500~epochs} \label{fig:results:scoreloss_matrices}
\end{figure}

\subsubsection{Results for different binsizes and windowsizes} \label{sec:results:binsize_winsize}
To assess predictions at larger bin sizes, five different approaches were compared, cf. \cref{sec:improve:binsize_winsize}:
\begin{enumerate}
 \item ``50k direct'': \\directly training a network at bin size \SI{50}{\kilo\bp} and predicting at that same bin size
 \item ``initial 25k coarsened'': \\coarsening the results of the initial network discussed above (\cref{sec:initialDNNresults})
                    by summarizing bins via \texttt{cooler coarsen}, cf. \cref{sec:methods:input_data}
 \item ``initial 25k$\rightarrow$50k'': \\using the initial network trained at \SI{25}{\kilo\bp} (cf. \cref{sec:initialDNNresults}) to predict at \SI{50}{\kilo\bp}
 \item ``25k+50k$\rightarrow$50k'': \\predicting at \SI{50}{\kilo\bp} from a network \emph{simultaneously} trained with bin sizes 25 and \SI{50}{\kilo\bp}
 \item ``25k+50k$\rightarrow$25k'': \\predicting at \SI{25}{\kilo\bp} from a network \emph{simultaneously} trained with bin sizes 25 and \SI{50}{\kilo\bp}
\end{enumerate}

The best Pearson correlations at bin size \SI{50}{\kilo\bp} were generally obtained either by coarsening the initial results to \SI{50}{\kilo\bp} 
(method b) or by taking the network trained at \SI{25}{\kilo\bp} for predicting at \SI{50}{\kilo\bp} (method c), \cref{fig:results:DNN50k_pearson}.
Compared to coarsening, the latter approach had the advantage of doubling the window size (in base pairs) and it worked better for test chromosome 21.

Looking at the corresponding matrix plots, the desired effect of making larger structures more prominent by increasing the bin size was only partially achieved, \cref{fig:results:50k_from25k_matrices}.
While all larger structures in the example cutout of test chromosome 3 indeed looked more prominent, \cref{fig:results:50k_from25k_r3},
no obvious improvement was observed for the medium-sized structures in the example regions of chromosome 19 and 21, 
figures \ref{fig:results:50k_from25k_r1} and \ref{fig:results:50k_from25k_r2}.

Direct predictions at bin size \SI{50}{\kilo\bp} (method a) were definitely not better than indirect methods derived from networks
trained at \SI{25}{\kilo\bp}. 
Both the Pearson correlations and the matrix plots were clearly better for method b) and c),
figures \ref{fig:results:DNN50k_pearson}, \ref{fig:results:50k_matrices} and \ref{fig:results:50k_from25k_matrices}.
It is not known why the direct predictions turned out worse. 
Potential reasons include the reduced number of samples (cf. \cref{tab:methods:samples}, p. \pageref{tab:methods:samples})
and the binning process, or a combination of both.
However, first investigations showed that binning the proteins using the maximum instead of the mean across the \SI{50}{\kilo\bp}-bins, cf. \cref{sec:methods:sample_gen},
did not improve the results.

Notably, the training process for the direct prediction at bin size \SI{50}{\kilo\bp} (method a) diverged after about 420 epochs.
One possible reason for this could be too high a learning rate, which could have been avoided by decreasing the learning rate over time.
However, no further investigations were made into the case, because the divergence occurred only after overfitting, \cref{fig:results:50k_lossEpochs}, and was thus not seen as too problematic here.
The minimum validation error was reached after about 150 epochs and thus about 100 epochs earlier than in the initial setup at \SI{25}{\kilo\bp}. 
This is not surprising, since there are only about half as many training samples at \SI{50}{\kilo\bp} compared to \SI{25}{\kilo\bp}, 
cf. \cref{tab:methods:samples} (p. \pageref{tab:methods:samples}).

\begin{figure}[p]%50k direct AND from 25k, pearson and progress
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/DNN_50k/}{pearson_chr03.pdf_tex}}
        \caption{chr3}
    \end{subfigure} \hfill
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/DNN_50k/}{pearson_chr05.pdf_tex}}
        \caption{chr5}
    \end{subfigure}\\[5mm]
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/DNN_50k/}{pearson_chr10.pdf_tex}}
        \caption{chr10}
    \end{subfigure}\hfill
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/DNN_50k/}{pearson_chr19.pdf_tex}}
        \caption{chr19}
    \end{subfigure}\\[3mm]
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/DNN_50k/}{pearson_chr21.pdf_tex}}
        \caption{chr21}
    \end{subfigure}\hfill
    \begin{subfigure}{0.45\textwidth}
        \resizebox{\textwidth}{!}{
        \scriptsize
        \import{figures/DNN_50k/}{lossOverEpochs.pdf_tex}}
        \caption{Learning progress 50k direct} \label{fig:results:50k_lossEpochs}
    \end{subfigure}
    \caption{Results\,/\,metrics, various \acrshort{dnn}s at \SI{50}{\kilo\bp}} \label{fig:results:DNN50k_pearson}
\end{figure}
\begin{figure}[p] %50k direct, matrices
    \begin{subfigure}{\textwidth}
        \centering
        \resizebox{0.71\textwidth}{!}{
        \scriptsize
        \import{figures/DNN_50k/}{pred00250_chr21_030-040.pdf_tex}}
        \caption{Example  region 1} \label{fig:results:50k_r1}
    \end{subfigure}\\[2mm]
    \centering{\scriptsize{see \cref{sec:methods:matrix_plots} for track explanations}}\\[3mm]
    \begin{subfigure}{\textwidth}
        \centering
        \resizebox{0.71\textwidth}{!}{
        \scriptsize
        \import{figures/DNN_50k/}{pred00250_chr19_030-040.pdf_tex}}
        \caption{Example region 2} \label{fig:results:50k_r2}
    \end{subfigure}\\[2mm]
    \centering{\scriptsize{see \cref{sec:methods:matrix_plots} for track explanations}}\\[3mm]
    \begin{subfigure}{\textwidth}
        \centering
        \resizebox{0.71\textwidth}{!}{
        \scriptsize
        \import{figures/DNN_50k/}{pred00250_chr3_030-040.pdf_tex}}
        \caption{Example region 3} \label{fig:results:50k_r3}
    \end{subfigure}
    \caption{Example predictions GM12878 $\rightarrow$ K562,  \acrshort{dnn} at \SI{50}{\kilo\bp} direct, 250 epochs} \label{fig:results:50k_matrices}
\end{figure}
\begin{figure}[p] %50k from 25k, matrices
    \begin{subfigure}{\textwidth}
        \centering
        \resizebox{0.71\textwidth}{!}{
        \scriptsize
        \import{figures/DNN_50k/}{pred00500_50k_chr21_030-040.pdf_tex}}
        \caption{Example  region 1} \label{fig:results:50k_from25k_r1}
    \end{subfigure}\\[2mm]
    \centering{\scriptsize{see \cref{sec:methods:matrix_plots} for track explanations}}\\[3mm]
    \begin{subfigure}{\textwidth}
        \centering
        \resizebox{0.71\textwidth}{!}{
        \scriptsize
        \import{figures/DNN_50k/}{pred00500_50k_chr19_030-040.pdf_tex}}
        \caption{Example region 2} \label{fig:results:50k_from25k_r2}
    \end{subfigure}\\[2mm]
    \centering{\scriptsize{see \cref{sec:methods:matrix_plots} for track explanations}}\\[3mm]
    \begin{subfigure}{\textwidth}
        \centering
        \resizebox{0.71\textwidth}{!}{
        \scriptsize
        \import{figures/DNN_50k/}{pred00500_50k_chr3_030-040.pdf_tex}}
        \caption{Example region 3} \label{fig:results:50k_from25k_r3}
    \end{subfigure}
    \caption{Example predictions GM12878 $\rightarrow$ K562,  \acrshort{dnn} trained at \SI{25}{\kilo\bp} predicting at \SI{50}{\kilo\bp}, 500~epochs} \label{fig:results:50k_from25k_matrices}
\end{figure}

Simultaneously training a network with matrix- and feature bin sizes of \SI{25}{\kilo\bp} and \SI{50}{\kilo\bp} (methods d, e)
turned out unproblematic with regard to convergence, \cref{fig:results:25plus50_lossEpochs}, 
but the Pearson correlations when predicting at both \SI{25}{\kilo\bp} and \SI{50}{\kilo\bp} were -- often significantly -- worse
than the initial predictions at the respective bin size, \cref{fig:results:DNN50k_pearson} (``25k+50k$\rightarrow$50k'') and \cref{fig:results:DNN25plus50_pearson} (``25k+50k$\rightarrow$25k'').
Looking into the matrix plots shown in \cref{fig:results:25plus50_matrices}, 
all predictions seemed equally useless and definitely worse than the results obtained by the other approaches investigated thus far.
It could not be clarified what caused the improvement in Pearson correlations for chromosome 21 compared to the initial predictions at \SI{25}{\kilo\bp}, 
\cref{fig:results:DNN_25_pearson_21}, but it is interesting that even predictions with such a high degree of blurriness as in \cref{fig:results:25plus50_r1}
can reach an \acrshort{auc} of around 0.65.

\begin{figure}[p]%trained at 25k and 50k simultaneously, pearson and progress for 25k
    \begin{subfigure}{0.45\textwidth}
        \scriptsize 
        \resizebox{\textwidth}{!}{
        \import{figures/DNN_25plus50/}{pearson_chr03.pdf_tex}}
        \caption{chr3}
    \end{subfigure} \hfill
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/DNN_25plus50/}{pearson_chr05.pdf_tex}}
        \caption{chr5}
    \end{subfigure}\\[5mm]
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/DNN_25plus50/}{pearson_chr10.pdf_tex}}
        \caption{chr10}
    \end{subfigure}\hfill
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/DNN_25plus50/}{pearson_chr19.pdf_tex}}
        \caption{chr19}
    \end{subfigure}\\[3mm]
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/DNN_25plus50/}{pearson_chr21.pdf_tex}}
        \caption{chr21} \label{fig:results:DNN_25_pearson_21}
    \end{subfigure}\hfill
    \begin{subfigure}{0.45\textwidth}
        \resizebox{\textwidth}{!}{
        \scriptsize
        \import{figures/DNN_25plus50/}{lossOverEpochs.pdf_tex}}
        \caption{Learning progress} \label{fig:results:25plus50_lossEpochs}
    \end{subfigure}
    \caption{Results\,/\,metrics, \acrshort{dnn} trained at \SI{25}{\kilo\bp} and \SI{50}{\kilo\bp} simultaneously} \label{fig:results:DNN25plus50_pearson}
\end{figure}
\begin{figure}[p] %25plus50, matrices at 25k
    \begin{subfigure}{\textwidth}
        \centering
        \scriptsize
        \import{figures/DNN_25plus50/}{pred00500_chr21_030-040.pdf_tex}
        \caption{Example  region 1} \label{fig:results:25plus50_r1}
    \end{subfigure}\\[2mm]
    \centering{\scriptsize{see \cref{sec:methods:matrix_plots} for track explanations}}\\[3mm]
    \begin{subfigure}{\textwidth}
        \centering
        \scriptsize
        \import{figures/DNN_25plus50/}{pred00500_chr19_030-040.pdf_tex}
        \caption{Example region 2} \label{fig:results:25plus50_r2}
    \end{subfigure}\\[2mm]
    \centering{\scriptsize{see \cref{sec:methods:matrix_plots} for track explanations}}\\[3mm]
    \begin{subfigure}{\textwidth}
        \centering
        \scriptsize
        \import{figures/DNN_25plus50/}{pred00500_chr3_030-040.pdf_tex}
        \caption{Example region 3} \label{fig:results:25plus50_r3}
    \end{subfigure}
    \caption{Example predictions GM12878 $\rightarrow$ K562, \acrshort{dnn} trained at \SI{25}{\kilo\bp} and \SI{50}{\kilo\bp} simultaneously, \SI{25}{\kilo\bp}, 500 epochs}\label{fig:results:25plus50_matrices}
\end{figure}

\clearpage
\subsection{Hi-cGAN approaches} \label{sec:results:cgan}
In the following three subsections, the results from the conditional generative adversarial networks will be presented.
Although there is still room for improvements, at least one of the three \acrshort{cgan}-variants under investigation showed good performance. 

\subsubsection{cGAN with DNN embedding} \label{sec:results:cgan_dnn}
The \acrshort{cgan} with \acrshort{dnn}-embedding showed interesting results.
All in all, the training process was smooth and converged after around 60 epochs, mostly using the parameters suggested
by Isola et al. \cite{Isola2017} (cf. \ref{sec:methods:cGAN_initial}). 
Although \emph{pix2pix} has shown fast convergence in other applications, 
it was surprising that this still held after the changes made to the original network, especially adding the embedding network. 
While the Pearson correlations were mostly worse than the ones of the \acrshort{dnn} presented above, compare figures \ref{fig:results:DNN64_pearson} and \ref{fig:results:GAN64-dnn_pearson},
the matrices mostly looked visually better, showing more distinct boundaries between interacting and non-interacting regions, compare figures \ref{fig:results:DNN_matrices} and \ref{fig:results:cGAN64-dnn_matrices}.

Using weight transfer from a pre-trained \acrshort{dnn} for the embedding networks further stabilized the training process 
and made the discriminator reach a stable value of around 0.693 ($\approx -\log0.5$) on epoch average after only 2 epochs, 
while the generator validation loss reached a stable minimum value after about 15 epochs, \cref{fig:results:GAN64_pretrain-dnn_lossEpochs}.
However, the resulting predictions did not really improve.
While the Pearson correlations showed showed slightly better values for the pre-trained \acrshort{dnn}-embedding compared to the non-pre-trained, \cref{fig:results:GAN64_pretrain-dnn_pearson},
the matrices were visually clearly worse than without pre-training, and also worse than the results from the \acrshort{dnn} alone, \cref{fig:results:cGAN64_pretrain-dnn_matrices}.

\begin{figure}[p] %cGAN with DNN, no pretraining, windowsize 64, pearson and progress
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/GAN_64_dnn/}{pearson_chr03.pdf_tex}}
        \caption{chr3}
    \end{subfigure} \hfill
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/GAN_64_dnn/}{pearson_chr05.pdf_tex}}
        \caption{chr5}
    \end{subfigure}\\[5mm]
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/GAN_64_dnn/}{pearson_chr10.pdf_tex}}
        \caption{chr10}
    \end{subfigure}\hfill
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/GAN_64_dnn/}{pearson_chr19.pdf_tex}}
        \caption{chr19}
    \end{subfigure}\\[3mm]
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/GAN_64_dnn/}{pearson_chr21.pdf_tex}}
        \caption{chr21}
    \end{subfigure} \hfill
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/GAN_64_dnn/}{lossOverEpochs.pdf_tex}}
        \caption{Learning progress} \label{fig:results:GAN64-dnn_lossEpochs}
    \end{subfigure}
    \caption{Results\,/\,metrics \acrshort{cgan}, \acrshort{dnn} embedding, no pre-training, $w=64$, test chromosomes}   \label{fig:results:GAN64-dnn_pearson}
\end{figure}
\begin{figure}[p] %cgan with DNN, no pre-training, winsize 64, matrices
    \begin{subfigure}{\textwidth}
        \centering
        \scriptsize
        \import{figures/GAN_64_dnn/}{pred00060_chr21_030-040.pdf_tex}
        \caption{Example  region 1} \label{fig:results:cGAN64-dnn_r1}
    \end{subfigure}\\[2mm]
    \centering{\scriptsize{see \cref{sec:methods:matrix_plots} for track explanations}}\\[3mm]
    \begin{subfigure}{\textwidth}
        \centering
        \scriptsize
        \import{figures/GAN_64_dnn/}{pred00060_chr19_030-040.pdf_tex}
        \caption{Example region 2} \label{fig:results:cGAN64-dnn_r2}
    \end{subfigure}\\[2mm]
    \centering{\scriptsize{see \cref{sec:methods:matrix_plots} for track explanations}}\\[3mm]
    \begin{subfigure}{\textwidth}
        \centering
        \scriptsize
        \import{figures/GAN_64_dnn/}{pred00060_chr3_030-040.pdf_tex}
        \caption{Example region 3} \label{fig:results:cGAN64-dnn_r3}
    \end{subfigure}
    \caption{Example predictions GM12878 $\rightarrow$ K562, \acrshort{cgan}, \acrshort{dnn} embedding, no pre-training, $w=64$, 60 epochs} \label{fig:results:cGAN64-dnn_matrices}
\end{figure}
\begin{figure}[p] %cGAN with DNN, pretrained, windowsize 64, pearson and progress
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/GAN_64_pretrain-dnn/}{pearson_chr03.pdf_tex}}
        \caption{chr3}
    \end{subfigure} \hfill
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/GAN_64_pretrain-dnn/}{pearson_chr05.pdf_tex}}
        \caption{chr5}
    \end{subfigure}\\[5mm]
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/GAN_64_pretrain-dnn/}{pearson_chr10.pdf_tex}}
        \caption{chr10}
    \end{subfigure}\hfill
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/GAN_64_pretrain-dnn/}{pearson_chr19.pdf_tex}}
        \caption{chr19}
    \end{subfigure}\\[3mm]
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/GAN_64_pretrain-dnn/}{pearson_chr21.pdf_tex}}
        \caption{chr21}
    \end{subfigure} \hfill
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/GAN_64_pretrain-dnn/}{lossOverEpochs.pdf_tex}}
        \caption{Learning progress} \label{fig:results:GAN64_pretrain-dnn_lossEpochs}
    \end{subfigure}
    \caption{Results\,/\,metrics, \acrshort{cgan}, \acrshort{dnn} embedding, pre-trained, $w=64$, test chromosomes}   \label{fig:results:GAN64_pretrain-dnn_pearson}
\end{figure}
\begin{figure}[p] %cgan with DNN, pretrained, winsize 64, matrices
    \begin{subfigure}{\textwidth}
        \centering
        \scriptsize
        \import{figures/GAN_64_pretrain-dnn/}{pred00100_chr21_030-040.pdf_tex}
        \caption{Example  region 1} \label{fig:results:cGAN64_pretrain-dnn_r1}
    \end{subfigure}\\[2mm]
    \centering{\scriptsize{see \cref{sec:methods:matrix_plots} for track explanations}}\\[3mm]
    \begin{subfigure}{\textwidth}
        \centering
        \scriptsize
        \import{figures/GAN_64_pretrain-dnn/}{pred00100_chr19_030-040.pdf_tex}
        \caption{Example region 2} \label{fig:results:cGAN64_pretrain-dnn_r2}
    \end{subfigure}\\[2mm]
    \centering{\scriptsize{see \cref{sec:methods:matrix_plots} for track explanations}}\\[3mm]
    \begin{subfigure}{\textwidth}
        \centering
        \scriptsize
        \import{figures/GAN_64_pretrain-dnn/}{pred00100_chr3_030-040.pdf_tex}
        \caption{Example region 3} \label{fig:results:cGAN64_pretrain-dnn_r3}
    \end{subfigure}
    \caption{Example predictions GM12878 $\rightarrow$ K562, \acrshort{cgan}, \acrshort{dnn} embedding, pre-trained, $w=64$, 100~epochs} \label{fig:results:cGAN64_pretrain-dnn_matrices}
\end{figure}


\subsubsection{cGAN with CNN embedding} \label{sec:results:cgan_cnn}
The \acrshort{cnn} embedding also allowed for a stable training of the \acrshort{cgan},
whereby the discriminator loss reached stable epoch-average values after 20 to 30 epochs, depending on windowsize, \cref{fig:results:cGAN64_lossEpochs}.
For windowsize $w=64$, the Pearson correlations showed worse values than the \acrshort{dnn}-approach or the \acrshort{cgan} with \acrshort{dnn} embedding, \cref{fig:results:cGAN64_pearson}.
Nevertheless, the predicted matrices looked visually better, at least for the given test cutouts shown in \cref{fig:results:cGAN64_matrices}. 
For windowsizes $w\in\{128,256\}$, the Pearson correlations often reached and partially exceeded the baseline (``transfer GM12878 to K562''),
and the matrix plots were visually in good accordance with the targets, figures \ref{fig:results:cGAN128_pearson} to \ref{fig:results:cGAN256_matrices}.

Predictions with swapped training- and test cell line, i.\,e. with data from K562 for training and data from GM12878 for prediction 
were investigated only for windowsize $w=256$, which seemed most promising. 
Even though the training process was not as stable as for other investigations at smaller windowsizes, 
the predictions indeed showed Pearson correlations largely above the baseline (``transfer K562 to GM12878'') 
and matrix plots were in good accordance with their respective targets, cf. \cref{fig:appendix:GAN256_K-on-G_pearson} and \ref{fig:appendix:GAN256_K-on-G_matrices}.

Average training times per epoch were around \SI{110}{\min} for windowsize $w=256$
and \SI{28}{\min} and \SI{14}{\min} for windowsizes $w=128$ and $w=64$, respectively.
Here, training was performed on machine 1 for windowsizes $w=\{64,128\}$ and on machine 2 for windowsize $w=256$,
due to memory limitations on machine 1, cf. \cref{sec:appendix:hardware}.

\begin{figure}[p] %cGAN with CNN, windowsize 64, pearson and progress
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/GAN_64_cnn/}{pearson_chr03.pdf_tex}}
        \caption{chr3}
    \end{subfigure} \hfill
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/GAN_64_cnn/}{pearson_chr05.pdf_tex}}
        \caption{chr5}
    \end{subfigure}\\[5mm]
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/GAN_64_cnn/}{pearson_chr10.pdf_tex}}
        \caption{chr10}
    \end{subfigure}\hfill
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/GAN_64_cnn/}{pearson_chr19.pdf_tex}}
        \caption{chr19}
    \end{subfigure}\\[3mm]
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/GAN_64_cnn/}{pearson_chr21.pdf_tex}}
        \caption{chr21}
    \end{subfigure} \hfill
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/GAN_64_cnn/}{lossOverEpochs.pdf_tex}}
        \caption{Learning progress} \label{fig:results:cGAN64_lossEpochs}
    \end{subfigure}
    \caption{Results\,/\,metrics \acrshort{cgan}, \acrshort{cnn} embedding, $w=64$, test chromosomes}   \label{fig:results:cGAN64_pearson}
\end{figure}
\begin{figure}[p] %cgan CNN, 64, matrices
    \begin{subfigure}{\textwidth}
        \centering
        \scriptsize
        \import{figures/GAN_64_cnn/}{pred00080_chr21_030-040.pdf_tex}
        \caption{Example  region 1} \label{fig:results:cGAN64_r1}
    \end{subfigure}\\[2mm]
    \centering{\scriptsize{see \cref{sec:methods:matrix_plots} for track explanations}}\\[3mm]
    \begin{subfigure}{\textwidth}
        \centering
        \scriptsize
        \import{figures/GAN_64_cnn/}{pred00080_chr19_030-040.pdf_tex}
        \caption{Example region 2} \label{fig:results:cGAN64_r2}
    \end{subfigure}\\[2mm]
    \centering{\scriptsize{see \cref{sec:methods:matrix_plots} for track explanations}}\\[3mm]
    \begin{subfigure}{\textwidth}
        \centering
        \scriptsize
        \import{figures/GAN_64_cnn/}{pred00080_chr3_030-040.pdf_tex}
        \caption{Example region 3} \label{fig:results:cGAN64_r3}
    \end{subfigure}
    \caption{Example predictions GM12878 $\rightarrow$ K562, \acrshort{cgan}, \acrshort{cnn} embedding, $w=64$, 80~epochs}
\end{figure} \label{fig:results:cGAN64_matrices}

\begin{figure}[p] %cGAN cnn 128, pearson and progress
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/GAN_128/}{pearson_chr03.pdf_tex}}
        \caption{chr3}
    \end{subfigure} \hfill
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/GAN_128/}{pearson_chr05.pdf_tex}}
        \caption{chr5}
    \end{subfigure}\\[5mm]
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/GAN_128/}{pearson_chr10.pdf_tex}}
        \caption{chr10}
    \end{subfigure}\hfill
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/GAN_128/}{pearson_chr19.pdf_tex}}
        \caption{chr19}
    \end{subfigure}\\[3mm]
    \centering
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/GAN_128/}{pearson_chr21.pdf_tex}}
        \caption{chr21}
    \end{subfigure} \hfill
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/GAN_128/}{lossOverEpochs.pdf_tex}}
        \caption{Learning progress} \label{fig:results:cGAN128_lossEpochs}
    \end{subfigure}
    \caption{Results\,/\,metrics \acrshort{cgan}, \acrshort{cnn} embedding. $w=128$, test chromosomes}   \label{fig:results:cGAN128_pearson}
\end{figure}
\begin{figure}[p] %cgan CNN, 128, matrices
    \begin{subfigure}{\textwidth}
        \centering
        \resizebox{0.8\textwidth}{!}{
        \scriptsize
        \import{figures/GAN_128/}{pred00090_chr21_030-040.pdf_tex}}
        \caption{Example  region 1} \label{fig:results:cGAN128_r1}
    \end{subfigure}\\[2mm]
    \centering{\scriptsize{see \cref{sec:methods:matrix_plots} for track explanations}}\\[3mm]
    \begin{subfigure}{\textwidth}
        \centering
        \resizebox{0.8\textwidth}{!}{
        \scriptsize
        \import{figures/GAN_128/}{pred00090_chr19_030-040.pdf_tex}}
        \caption{Example region 2} \label{fig:results:cGAN128_r2}
    \end{subfigure}\\[2mm]
    \centering{\scriptsize{see \cref{sec:methods:matrix_plots} for track explanations}}\\[3mm]
    \begin{subfigure}{\textwidth}
        \centering
        \resizebox{0.8\textwidth}{!}{
        \scriptsize
        \import{figures/GAN_128/}{pred00090_chr3_030-040.pdf_tex}}
        \caption{Example region 3} \label{fig:results:cGAN128_r3}
    \end{subfigure}
    \caption{Example predictions GM12878 $\rightarrow$ K562, \acrshort{cgan}, \acrshort{cnn} embedding, $w=128$, 90~epochs} \label{fig:results:cGAN128_matrices}
\end{figure}

\begin{figure}[p] %cGAN CNN 256, pearson and progress
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/GAN_256/}{pearson_chr03.pdf_tex}}
        \caption{chr3}
    \end{subfigure} \hfill
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/GAN_256/}{pearson_chr05.pdf_tex}}
        \caption{chr5}
    \end{subfigure}\\[5mm]
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/GAN_256/}{pearson_chr10.pdf_tex}}
        \caption{chr10}
    \end{subfigure}\hfill
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/GAN_256/}{pearson_chr19.pdf_tex}}
        \caption{chr19}
    \end{subfigure}\\[3mm]
    \centering
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/GAN_256/}{pearson_chr21.pdf_tex}}
        \caption{chr21}
    \end{subfigure} \hfill
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/GAN_256/}{lossOverEpochs.pdf_tex}}
        \caption{Learning progress} \label{fig:results:cGAN256_lossEpochs}
    \end{subfigure}
    \caption{Results\,/\,metrics \acrshort{cgan}, \acrshort{cnn} embedding, $w=256$, test chromosomes}   \label{fig:results:cGAN256_pearson}
\end{figure}
\begin{figure}[p] %cgan CNN, 256k, matrices
    \begin{subfigure}{\textwidth}
        \centering
        \scriptsize
        \import{figures/GAN_256/}{pred00100_chr21_030-040.pdf_tex}
        \caption{Example  region 1} \label{fig:results:cGAN256_r1}
    \end{subfigure}\\[2mm]
    \centering{\scriptsize{see \cref{sec:methods:matrix_plots} for track explanations}}\\[3mm]
    \begin{subfigure}{\textwidth}
        \centering
        \scriptsize
        \import{figures/GAN_256/}{pred00100_chr19_030-040.pdf_tex}
        \caption{Example region 2} \label{fig:results:cGAN256_r2}
    \end{subfigure}
    \caption{Example predictions GM12878 $\rightarrow$ K562, \acrshort{cgan}, \acrshort{cnn} embedding, $w=256$, 100~epochs}
\end{figure}
\begin{figure}\ContinuedFloat
    \begin{subfigure}{\textwidth}
        \centering
        \scriptsize
        \import{figures/GAN_256/}{pred00100_chr3_030-040.pdf_tex}
        \caption{Example region 3} \label{fig:results:cGAN256_r3}
    \end{subfigure}
    \caption{Example predictions GM12878 $\rightarrow$ K562, \acrshort{cgan}, \acrshort{cnn} embedding, $w=256$, 100~epochs} \label{fig:results:cGAN256_matrices}
\end{figure}

\subsubsection{cGAN with mixed DNN / CNN embedding} \label{sec:results:cgan-mixed}
Using a \acrshort{cgan} with mixed embedding, i.\,e. \acrshort{dnn}-embedding for the generator
and \acrshort{cnn}-embedding for the discriminator, did not make for a stable training process, \ref{fig:results:GAN64_mixed_lossEpochs}.
Despite still acceptable Pearson correlations compared to the other approaches shown above,
the predicted matrices were useless, \cref{fig:results:GAN64_mixed_pearson} and \ref{fig:results:GAN64_mixed_matrices}.
First investigations indicated that the \acrshort{cgan} may have learned to partially ignore the feature input 
and is thus predicting submatrices that look similar to real Hi-C submatrices, but do not have sufficient correlation with the feature data.

Interestingly, pre-training the \acrshort{dnn} improved the situation in this mixed \acrshort{dnn}/\acrshort{cnn} embedding setup. 
While the training process still seemed unstable, \cref{fig:results:GAN64_pretrained_mixed_lossEpochs},
the predicted matrices looked much better, similar to the ones from the \acrshort{cgan} with \acrshort{dnn}-embedding (non-pre-trained),
compare figures \ref{fig:results:GAN64_pretrained_mixed_matrices} and \ref{fig:results:cGAN64-dnn_matrices}.
However, the results were still not better than the ones from the \acrshort{cnn} embedding alone, cf. \cref{sec:results:cgan_cnn}.

\begin{figure}[p] %cGAN mixed 64, not pretrained, pearson and progress
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/GAN_64_mixed/}{pearson_chr03.pdf_tex}}
        \caption{chr3}
    \end{subfigure} \hfill
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/GAN_64_mixed/}{pearson_chr05.pdf_tex}}
        \caption{chr5}
    \end{subfigure}\\[5mm]
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/GAN_64_mixed/}{pearson_chr10.pdf_tex}}
        \caption{chr10}
    \end{subfigure}\hfill
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/GAN_64_mixed/}{pearson_chr19.pdf_tex}}
        \caption{chr19}
    \end{subfigure}\\[3mm]
    \centering
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/GAN_64_mixed/}{pearson_chr21.pdf_tex}}
        \caption{chr21}
    \end{subfigure} \hfill
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/GAN_64_mixed/}{lossOverEpochs.pdf_tex}}
        \caption{Learning progress} \label{fig:results:GAN64_mixed_lossEpochs}
    \end{subfigure}
    \caption{Results\,/\,metrics \acrshort{cgan}, mixed embedding, no pre-training, $w=64$, test chromosomes}   \label{fig:results:GAN64_mixed_pearson}
\end{figure}
\begin{figure}[p] %cgan mixed 64, pretrained, matrices
    \begin{subfigure}{\textwidth}
        \centering
        \resizebox{0.9\textwidth}{!}{
        \scriptsize
        \import{figures/GAN_64_mixed/}{pred00020_chr21_030-040.pdf_tex}}
        \caption{Example  region 1} \label{fig:results:cGAN64_mixed_r1}
    \end{subfigure}\\[2mm]
    \centering{\scriptsize{see \cref{sec:methods:matrix_plots} for track explanations}}\\[3mm]
    \begin{subfigure}{\textwidth}
        \centering
        \resizebox{0.9\textwidth}{!}{
        \scriptsize
        \import{figures/GAN_64_mixed/}{pred00020_chr19_030-040.pdf_tex}}
        \caption{Example region 2} \label{fig:results:cGAN64_mixed_r2}
    \end{subfigure}\\[2mm]
    \centering{\scriptsize{see \cref{sec:methods:matrix_plots} for track explanations}}\\[3mm]
    \begin{subfigure}{\textwidth}
        \centering
        \resizebox{0.9\textwidth}{!}{
        \scriptsize
        \import{figures/GAN_64_mixed/}{pred00020_chr3_030-040.pdf_tex}}
        \caption{Example region 3} \label{fig:results:cGAN64_mixed_r3}
    \end{subfigure}
    \caption{Example predictions GM12878 $\rightarrow$ K562, \acrshort{cgan}, mixed embedding, no pre-training, $w=64$, 20~epochs} 
     \label{fig:results:GAN64_mixed_matrices}
\end{figure}

\begin{figure}[p] %cGAN mixed 64, PRE-TRAINED, pearson and progress
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/GAN_64_pretrain-mixed/}{pearson_chr03.pdf_tex}}
        \caption{chr3}
    \end{subfigure} \hfill
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/GAN_64_pretrain-mixed/}{pearson_chr05.pdf_tex}}
        \caption{chr5}
    \end{subfigure}\\[5mm]
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/GAN_64_pretrain-mixed/}{pearson_chr10.pdf_tex}}
        \caption{chr10}
    \end{subfigure}\hfill
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/GAN_64_pretrain-mixed/}{pearson_chr19.pdf_tex}}
        \caption{chr19}
    \end{subfigure}\\[3mm]
    \centering
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/GAN_64_pretrain-mixed/}{pearson_chr21.pdf_tex}}
        \caption{chr21}
    \end{subfigure} \hfill
    \begin{subfigure}{0.45\textwidth}
        \scriptsize
        \resizebox{\textwidth}{!}{
        \import{figures/GAN_64_pretrain-mixed/}{lossOverEpochs.pdf_tex}}
        \caption{Learning progress} \label{fig:results:GAN64_pretrained_mixed_lossEpochs}
    \end{subfigure}
    \caption{Results\,/\,metrics \acrshort{cgan}, mixed embedding, \acrshort{dnn} pre-trained, $w=64$, test chromosomes}   \label{fig:results:GAN64_pretrained_mixed_pearson}
\end{figure}
\begin{figure}[p] %cgan mixed 64, pretrained, matrices
    \begin{subfigure}{\textwidth}
        \centering
        \resizebox{0.9\textwidth}{!}{
        \scriptsize
        \import{figures/GAN_64_pretrain-mixed/}{pred00040_chr21_030-040.pdf_tex}}
        \caption{Example  region 1} \label{fig:results:cGAN64_pretrained_mixed_r1}
    \end{subfigure}\\[2mm]
    \centering{\scriptsize{see \cref{sec:methods:matrix_plots} for track explanations}}\\[3mm]
    \begin{subfigure}{\textwidth}
        \centering
        \resizebox{0.9\textwidth}{!}{
        \scriptsize
        \import{figures/GAN_64_pretrain-mixed/}{pred00040_chr19_030-040.pdf_tex}}
        \caption{Example region 2} \label{fig:results:cGAN64_pretrained_mixed_r2}
    \end{subfigure}\\[2mm]
    \centering{\scriptsize{see \cref{sec:methods:matrix_plots} for track explanations}}\\[3mm]
    \begin{subfigure}{\textwidth}
        \centering
        \resizebox{0.9\textwidth}{!}{
        \scriptsize
        \import{figures/GAN_64_pretrain-mixed/}{pred00040_chr3_030-040.pdf_tex}}
        \caption{Example region 3} \label{fig:results:cGAN64_pretrained_mixed_r3}
    \end{subfigure}
    \caption{Example predictions GM12878 $\rightarrow$ K562, \acrshort{cgan}, mixed embedding, \acrshort{dnn} pre-trained, $w=64$, 40~epochs} 
     \label{fig:results:GAN64_pretrained_mixed_matrices}
\end{figure}

\clearpage
\subsection{Comparison with other approaches}\label{sec:results:comparison}
When comparing the random-forest-based method by Zhang et al., \emph{HiC-Reg} \cite{Zhang2019}, 
to the \acrshort{cgan} model trained on the typical training chromosome set, cf. \ref{sec:methods:sample_gen},
the \acrshort{cgan} approach seemed superior for distances up to about \SI{200}{\kilo\bp}, 
while both \emph{HiC-Reg}-MULTICELL and -WINDOW outperformed \acrshort{dnn} and Hi-cGAN for distances between 
\SI{200}{\kilo\bp} and \SI{1}{\mega\bp}, \cref{fig:results:zhang-vs-ours_pearson}.
This was also reflected in the matrix plots, \cref{fig:results:zhang-vs-ours_matrices}. 
While the \acrshort{cgan} often predicted smaller structures up to about \SI{400}{\kilo\bp} very well and offered distinct boundaries
even among nested structures, \emph{HiC-Reg} showed better performance for interactions in the upper half of the windowsize,
see e.\,g. chromosome 17, 30...\SI{34.5}{\mega\bp}.
Note that the WINDOW data stems from a random forest trained only on data from GM12878,
while the MULTICELL data has been trained on chromatin feature data from GM12878, K562, HMEC, HUVEC and NHEK using the GM12878 matrix
as optimization target. 
Both \emph{HiC-Reg} variants have used only data from chromosome 14 or 17 at binsize \SI{5}{\kilo\bp},
while in this setting, Hi-cGAN has been trained on feature- and matrix data from GM12878, chromosomes 1, 2, 4, 7, 9, 11, 13, 14, 16, 17, 18, 20, and 22 at binsize \SI{25}{\kilo\bp}.

To get a better comparison with the HiC-Reg WINDOW approach, in a second setting, Hi-cGAN was trained on data from chromosome 14 or 17 only.
Surprisingly, despite the low amount of training samples, the training process converged with good Pearson correlations for the (training\mbox{-})chromosomes and 
produced visually good matrices, \cref{fig:results:zhang-vs-ours_pearson_14singlechrom}/\ref{fig:results:zhang-vs-ours_pearson_17singlechrom} 
and \ref{fig:results:zhang-vs-ours_matrices_ours-only-14}/\ref{fig:results:zhang-vs-ours_matrices_ours-only-17}.
In general, the obeservations from above were also valid here: Hi-cGAN was outperforming \emph{HiC-Reg}-MULTICELL and -WINDOW when predicting smaller structures,
and \emph{HiC-Reg} performed better for larger structures.
However, in this second setting, structures sized approximately 500 to \SI{1000}{\kilo\bp} appeared more clearly in Hi-cGANs predictions 
and the intersection point in the Pearson correlation graphs moved further to the right, i.\,e. the distance range for which Hi-cGAN performed
better than \emph{HiC-Reg} was slightly shifted towards larger distances.
Interestingly, interacting pairs at distances beyond approximately \SI{1.3}{\kilo\bp} were all predicted zero by Hi-cGAN, 
likely due to the small number of samples in this setting, maybe in combination with insufficient training. 
The effect was more pronounced for chromosome 17, which indeed is shorter than chromosome 14 and thus yields less training samples.

\begin{figure}[htbp]
 \begin{subfigure}{0.45\textwidth}
  \resizebox{\textwidth}{!}{
  \scriptsize
  \import{figures/randomforest/}{pearson_chr14_overview_25k.pdf_tex}}
  \caption{Overview chromosome 14}\label{fig:results:zhang-vs-ours_pearson_14overview}
 \end{subfigure}\hfill
\begin{subfigure}{0.45\textwidth}
  \resizebox{\textwidth}{!}{
  \scriptsize
  \import{figures/randomforest/}{pearson_chr17_overview_25k.pdf_tex}}
  \caption{Overview chromosome 17}\label{fig:results:zhang-vs-ours_pearson_17overview}
 \end{subfigure}\\[4mm]
 \begin{subfigure}{0.45\textwidth}
  \resizebox{\textwidth}{!}{
  \scriptsize
  \import{figures/randomforest/}{pearson_chr14_detail_25k.pdf_tex} }
  \caption{chr14, detail 0...\SI{1}{\mega\bp}}\label{fig:results:zhang-vs-ours_pearson_14detail}
 \end{subfigure}\hfill
 \begin{subfigure}{0.45\textwidth}
  \resizebox{\textwidth}{!}{
  \scriptsize
  \import{figures/randomforest/}{pearson_chr17_detail_25k.pdf_tex}}
  \caption{chr17, detail 0...\SI{1}{\mega\bp}}\label{fig:results:zhang-vs-ours_pearson_17detail}
 \end{subfigure}\\[4mm]
 \begin{subfigure}{0.45\textwidth}
    \resizebox{\textwidth}{!}{
    \scriptsize
    \import{figures/randomforest/}{pearson_chr14_detail_25k_singleChrom.pdf_tex}}
    \caption{chr14 \acrshort{cgan} single chrom}\label{fig:results:zhang-vs-ours_pearson_14singlechrom}
 \end{subfigure}\hfill
 \begin{subfigure}{0.45\textwidth}
    \resizebox{\textwidth}{!}{
    \scriptsize
    \import{figures/randomforest/}{pearson_chr17_detail_25k_singleChrom.pdf_tex}}
    \caption{chr17 \acrshort{cgan} single chrom}\label{fig:results:zhang-vs-ours_pearson_17singlechrom}
 \end{subfigure}
  \caption{Pearson correlation comparison Hi-cGAN / \acrshort{dnn} and HiC-Reg \cite{Zhang2019}} \label{fig:results:zhang-vs-ours_pearson}
\end{figure}

\begin{figure}[htbp]
\begin{subfigure}{\textwidth}
 \centering
 \resizebox{0.92\textwidth}{!}{
 \scriptsize
 \import{figures/randomforest/}{pred0014_multicell_chr14_030-040.pdf_tex}}
 \caption{Hic-Reg (random forest), MULTICELL, reconstructed from \cite{Zhang2019}} \label{fig:results:zhang-vs-ours_matrices_multicell-14}
\end{subfigure}\\[5mm]
\begin{subfigure}{\textwidth}
 \centering
 \resizebox{0.92\textwidth}{!}{
 \scriptsize
 \import{figures/randomforest/}{pred00100_restriced_chr14_030-040.pdf_tex}}
 \caption{Hi-cGAN with windowsize $w=256$; plot restricted to first \SI{1}{\mega\bp}} \label{fig:results:zhang-vs-ours_matrices_ours-full-14}
\end{subfigure}\\[5mm]
\begin{subfigure}{\textwidth}
 \centering
 \resizebox{0.92\textwidth}{!}{
 \scriptsize
 \import{figures/randomforest/}{pred00100_chr14_030-040.pdf_tex}}
 \caption{Hi-cGAN with windowsize $w=64$, trained on chr14 only} \label{fig:results:zhang-vs-ours_matrices_ours-only-14}
\end{subfigure}\\[4mm]
    \centering{\scriptsize{\Cref{fig:results:zhang-vs-ours_matrices} a-f: Predicted matrices on top, true matrices from Rao et al. \cite{Rao2014} on bottom}}\\[5mm]
    \begin{subfigure}{\textwidth}
 \centering
 \resizebox{0.92\textwidth}{!}{
 \scriptsize
 \import{figures/randomforest/}{pred0017_multicell_chr17_030-040.pdf_tex}}
 \caption{Hic-Reg (random forest), MULTICELL, chr17, reconstructed from \cite{Zhang2019}} \label{fig:results:zhang-vs-ours_matrices_multicell-17}
\end{subfigure}\\[5mm]
\begin{subfigure}{\textwidth}
 \centering
 \resizebox{0.92\textwidth}{!}{
 \scriptsize
 \import{figures/randomforest/}{pred00100_restriced_chr17_030-040.pdf_tex}}
 \caption{Hi-cGAN with windowsize $w=256$, chr17, plot restricted to first \SI{1}{\mega\bp}}\label{fig:results:zhang-vs-ours_matrices_ours-full-17}
\end{subfigure}\\[5mm]
\begin{subfigure}{\textwidth}
 \centering
 \resizebox{0.92\textwidth}{!}{
 \scriptsize
 \import{figures/randomforest/}{pred00160_chr17_030-040.pdf_tex}}
 \caption{Hi-cGAN with windowsize $w=64$, trained on chr17 only} \label{fig:results:zhang-vs-ours_matrices_ours-only-17}
\end{subfigure}
\caption{Comparison HiC-Reg \cite{Zhang2019} and Hi-cGAN (\acrshort{cnn} embedding)} \label{fig:results:zhang-vs-ours_matrices}
\end{figure}

Next, the \acrshort{dnn} and Hi-cGAN were trained on data from Drosophila Melanogaster embryos
to allow a comparison with the results from Farr\'e et al \cite{Farre2018a}.

In the Pearson correlation plots for test chromosome 3R, the similarity between the \acrshort{dnn} approach and the method by 
Farré et al was obvious, both when plotting correlation over distance and position, \cref{fig:results:farre-vs-ours_pearson-pos_3R} 
and \ref{fig:results:farre-vs-ours_pearson-dist_3R}. 
Interestingly, the method by Farré et al. showed better values when plotting Pearson correlations over positions,
while the \acrshort{dnn} approach showed better values when plotting over distances. 
However, the values generally remained on a comparatively low level.
Note that the -- comparatively good -- values reported in the paper \cite[fig.~3]{Farre2018a} could not be reproduced; 
they may have been computed in a different way.
In terms of predicted matrices, the results published in the paper could widely be reproduced using the slightly modified 
code kindly provided by Pau Farré, cf. \ref{sec:methods:comparison}, see \cref{fig:results:farre-vs-ours_matrices} and \ref{fig:appendix:farre-vs-ours_matrices} 
(especially figure \ref{fig:appendix:farre-vs-ours_farre-matrix_recreated} vs. figure \ref{fig:appendix:farre-vs-ours_farre-matrix_original}).
Although some structures were well predicted, and despite the better distance-stratified Pearson correlations, 
the matrix plots from the \acrshort{dnn} approach looked clearly worse than the ones due to Farré et al., \cref{fig:results:farre-vs-ours_matrices}.
Since the \acrlong{dnn} is sharing the network architecture with the approach from Farr\'e et al., 
this can only mean that the chosen way of data preparation laid out in \ref{sec:methods:sample_gen} 
works worse than the approach chosen by Farré et al. for this type of network.

For Hi-cGAN, on the other hand, the chosen way of data preparation seemed to work well again.
Here, reasonable results for the test chromosomes 3R and X, \cref{fig:results:farre-vs-ours_cgan-matrix} were obtained,
both in terms of Pearson correlations and matrix plots, 
\cref{fig:results:farre-vs-ours_pearson-pos_3R}, \ref{fig:results:farre-vs-ours_pearson-dist_3R} and \ref{fig:results:farre-vs-ours_matrices}.

\begin{figure}[htbp]
 \begin{subfigure}{0.45\textwidth}
  \resizebox{\textwidth}{!}{
  \scriptsize
  \import{figures/GAN_64_drosophila/}{ppp_chr3R.pdf_tex}}
  \caption{Full chromosome 3R}
 \end{subfigure}\hfill
 \begin{subfigure}{0.45\textwidth}
  \resizebox{\textwidth}{!}{
  \scriptsize
  \import{figures/GAN_64_drosophila/}{ppp_chr3R_restricted.pdf_tex}}
  \caption{Cutout as in the paper \cite{Farre2018a}}
 \end{subfigure}
\caption{Comparison Pearson correlations \emph{per position}\\DNN / Hi-cGAN / Farr\'e et al. \cite{Farre2018a}}\label{fig:results:farre-vs-ours_pearson-pos_3R}
\end{figure}
\begin{figure}[htbp]
 \centering
 \resizebox{0.45\textwidth}{!}{
 \scriptsize
 \import{figures/FarreEtAl_drosophila/}{pearson_chr3R_comparison_DNN-Hi-cGAN-Farre.pdf_tex}}
 \caption{Comparison Pearson correlations \emph{per distance}\\DNN / Hi-cGAN / Farr\'e et al.}\label{fig:results:farre-vs-ours_pearson-dist_3R}
\end{figure}

\begin{figure}[htbp]
\centering{\scriptsize{predicted matrix on top, matrix from Schuettengruber et al. \cite{Schuettengruber2014} on bottom}}\\
\begin{subfigure}{\textwidth}
 \centering
 \scriptsize
 \import{figures/GAN_64_drosophila/}{pred00d_chr3R_1675-2175.pdf_tex}
 \caption{Matrix from Hi-cGAN, 100 epochs} \label{fig:results:farre-vs-ours_cgan-matrix}
\end{subfigure}\\[5mm]
\centering{\scriptsize{predicted matrix on top, ICE-normalized matrix derived from Schuettengruber et al. \cite{Schuettengruber2014} on bottom}}\\
\begin{subfigure}{\textwidth}
 \centering
 \scriptsize
 \import{figures/DNN_drosophila2/}{pred00d_chr3R_1675-2175.pdf_tex}
 \caption{Matrix from DNN, 40 epochs} \label{fig:results:farre-vs-ours_dnn-matrix}
\end{subfigure}\\[5mm]
\centering{\scriptsize{predicted matrix on top, ICE-normalized matrix derived from Schuettengruber et al. \cite{Schuettengruber2014} on bottom}}\\
\begin{subfigure}{\textwidth}
 \centering
 \scriptsize
 \import{figures/FarreEtAl_drosophila/}{pred00d_chr3R_1675-2175.pdf_tex}
 \caption{Matrix from Farr\'e et al. \cite{Farre2018a}, 70 epochs} \label{fig:results:farre-vs-ours_farre-matrix}
\end{subfigure}\\[5mm]
\caption{Comparison between matrix plots from DNN / Hi-cGAN / Farr\'e et al. \cite{Farre2018a}} \label{fig:results:farre-vs-ours_matrices}
\end{figure}

\clearpage
